
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>ampal package &#8212; AMPAL 1.4.0 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="ampal" href="modules.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="ampal-package">
<h1>ampal package<a class="headerlink" href="#ampal-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-ampal.align">
<span id="ampal-align-module"></span><h2>ampal.align module<a class="headerlink" href="#module-ampal.align" title="Permalink to this headline">¶</a></h2>
<p>A module containing classes for aligning structure.</p>
<dl class="class">
<dt id="ampal.align.MMCAlign">
<em class="property">class </em><code class="descclassname">ampal.align.</code><code class="descname">MMCAlign</code><span class="sig-paren">(</span><em>eval_fn, eval_args: Optional[list], polypeptide: ampal.protein.Polypeptide</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.align.MMCAlign" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A alignment protocol that uses Metropolis Monte Carlo.</p>
<p class="rubric">Notes</p>
<p>THIS IS CURRENTLY SUPER INEFFICIENT DUE TO THE DEEPCOPIES.
I plan to improve this by aligning arrays of atoms and only
recording the rotation and translation that led to that alignment.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>eval_fn</strong> (<em>Polypeptide -&gt; float</em>) – A function to evaluate the quality of your fit.</li>
<li><strong>eval_args</strong> (<em>list</em>) – A list of static args to be used in the <cite>eval_fn</cite>, these will
be unpacked into the evaluation function _i.e._
<a href="#id1"><span class="problematic" id="id2">`</span></a>eval_fn(polypeptide, <a href="#id3"><span class="problematic" id="id4">*</span></a>eval_args).</li>
<li><strong>polypeptide</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – An ampal polypeptide containing the model to be aligned.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="staticmethod">
<dt id="ampal.align.MMCAlign.check_move">
<em class="property">static </em><code class="descname">check_move</code><span class="sig-paren">(</span><em>new</em>, <em>old</em>, <em>t</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.check_move"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.align.MMCAlign.check_move" title="Permalink to this definition">¶</a></dt>
<dd><p>Determines if a model will be accepted.</p>
</dd></dl>

<dl class="staticmethod">
<dt id="ampal.align.MMCAlign.float_f">
<em class="property">static </em><code class="descname">float_f</code><span class="sig-paren">(</span><em>f</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.float_f"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.align.MMCAlign.float_f" title="Permalink to this definition">¶</a></dt>
<dd><p>Formats a float for printing to std out.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.align.MMCAlign.start_optimisation">
<code class="descname">start_optimisation</code><span class="sig-paren">(</span><em>rounds: int</em>, <em>max_angle: float</em>, <em>max_distance: float</em>, <em>temp: float = 298.15</em>, <em>stop_when=None</em>, <em>verbose=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.start_optimisation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.align.MMCAlign.start_optimisation" title="Permalink to this definition">¶</a></dt>
<dd><p>Starts the loop fitting protocol.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>rounds</strong> (<em>int</em>) – The number of Monte Carlo moves to be evaluated.</li>
<li><strong>max_angle</strong> (<em>float</em>) – The maximum variation in rotation that can moved per
step.</li>
<li><strong>max_distance</strong> (<em>float</em>) – The maximum distance the can be moved per step.</li>
<li><strong>temp</strong> (<em>float</em><em>, </em><em>optional</em>) – Temperature used during fitting process.</li>
<li><strong>stop_when</strong> (<em>float</em><em>, </em><em>optional</em>) – Stops fitting when energy is less than or equal to this value.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.align.align_backbones">
<code class="descclassname">ampal.align.</code><code class="descname">align_backbones</code><span class="sig-paren">(</span><em>reference</em>, <em>mobile</em>, <em>stop_when=None</em>, <em>verbose=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#align_backbones"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.align.align_backbones" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-ampal.amino_acids">
<span id="ampal-amino-acids-module"></span><h2>ampal.amino_acids module<a class="headerlink" href="#module-ampal.amino_acids" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="ampal.amino_acids.add_amino_acid_to_json">
<code class="descclassname">ampal.amino_acids.</code><code class="descname">add_amino_acid_to_json</code><span class="sig-paren">(</span><em>code</em>, <em>description</em>, <em>letter='X'</em>, <em>modified=None</em>, <em>force_add=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#add_amino_acid_to_json"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.amino_acids.add_amino_acid_to_json" title="Permalink to this definition">¶</a></dt>
<dd><p>Add an amino acid to the amino_acids.json file used to populate the amino_acid table.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>code</strong> (<em>str</em>) – New code to be added to amino acid table.</li>
<li><strong>description</strong> (<em>str</em>) – Description of the amino acid, e.g. ‘amidated terminal carboxy group’.</li>
<li><strong>letter</strong> (<em>str</em><em>, </em><em>optional</em>) – One letter code for the amino acid.
Defaults to ‘X’</li>
<li><strong>modified</strong> (<em>str</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Code of modified amino acid, e.g. ‘ALA’, or None.
Defaults to None</li>
<li><strong>force_add</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, will over-write existing dictionary value for code if already in amino_acids.json.
If False, then an IOError is raised if code is already in amino_acids.json.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first"><code class="xref py py-exc docutils literal notranslate"><span class="pre">IOError</span></code> – If code is already in amino_acids.json and force_add is False.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">None</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.amino_acids.get_aa_code">
<code class="descclassname">ampal.amino_acids.</code><code class="descname">get_aa_code</code><span class="sig-paren">(</span><em>aa_letter</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_code"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Get three-letter aa code if possible. If not, return None.</p>
<p>If three-letter code is None, will have to find this later from the filesystem.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aa_letter</strong> (<em>str</em>) – One-letter amino acid code.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>aa_code</strong> – Three-letter aa code.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">str, or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.amino_acids.get_aa_info">
<code class="descclassname">ampal.amino_acids.</code><code class="descname">get_aa_info</code><span class="sig-paren">(</span><em>code</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_info"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_info" title="Permalink to this definition">¶</a></dt>
<dd><p>Get dictionary of information relating to a new amino acid code not currently in the database.</p>
<p class="rubric">Notes</p>
<p>Use this function to get a dictionary that is then to be sent to the function add_amino_acid_to_json().
use to fill in rows of amino_acid table for new amino acid code.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>code</strong> (<em>str</em>) – Three-letter amino acid code.</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">IOError</span></code> – If unable to locate the page associated with the amino acid name on the PDBE site.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>aa_dict</strong> – Keys are AminoAcidDB field names.
Values are the str values for the new amino acid, scraped from the PDBE if possible. None if not found.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.amino_acids.get_aa_letter">
<code class="descclassname">ampal.amino_acids.</code><code class="descname">get_aa_letter</code><span class="sig-paren">(</span><em>aa_code</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_letter"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_letter" title="Permalink to this definition">¶</a></dt>
<dd><p>Get one-letter version of aa_code if possible. If not, return ‘X’.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aa_code</strong> (<em>str</em>) – Three-letter amino acid code.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>aa_letter</strong> – One-letter aa code.
Default value is ‘X’.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.ampal_warnings">
<span id="ampal-ampal-warnings-module"></span><h2>ampal.ampal_warnings module<a class="headerlink" href="#module-ampal.ampal_warnings" title="Permalink to this headline">¶</a></h2>
<dl class="exception">
<dt id="ampal.ampal_warnings.DependencyNotFoundWarning">
<em class="property">exception </em><code class="descclassname">ampal.ampal_warnings.</code><code class="descname">DependencyNotFoundWarning</code><a class="reference internal" href="_modules/ampal/ampal_warnings.html#DependencyNotFoundWarning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ampal_warnings.DependencyNotFoundWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="exception">
<dt id="ampal.ampal_warnings.MalformedPDBWarning">
<em class="property">exception </em><code class="descclassname">ampal.ampal_warnings.</code><code class="descname">MalformedPDBWarning</code><a class="reference internal" href="_modules/ampal/ampal_warnings.html#MalformedPDBWarning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ampal_warnings.MalformedPDBWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="exception">
<dt id="ampal.ampal_warnings.NoncanonicalWarning">
<em class="property">exception </em><code class="descclassname">ampal.ampal_warnings.</code><code class="descname">NoncanonicalWarning</code><a class="reference internal" href="_modules/ampal/ampal_warnings.html#NoncanonicalWarning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ampal_warnings.NoncanonicalWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="exception">
<dt id="ampal.ampal_warnings.NotParameterisedWarning">
<em class="property">exception </em><code class="descclassname">ampal.ampal_warnings.</code><code class="descname">NotParameterisedWarning</code><a class="reference internal" href="_modules/ampal/ampal_warnings.html#NotParameterisedWarning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ampal_warnings.NotParameterisedWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="function">
<dt id="ampal.ampal_warnings.check_availability">
<code class="descclassname">ampal.ampal_warnings.</code><code class="descname">check_availability</code><span class="sig-paren">(</span><em>program</em>, <em>test_func</em>, <em>global_settings</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#check_availability"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ampal_warnings.check_availability" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-ampal.analyse_protein">
<span id="ampal-analyse-protein-module"></span><h2>ampal.analyse_protein module<a class="headerlink" href="#module-ampal.analyse_protein" title="Permalink to this headline">¶</a></h2>
<p>Contains various tools for analysing protein structure.</p>
<dl class="function">
<dt id="ampal.analyse_protein.alpha_angles">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">alpha_angles</code><span class="sig-paren">(</span><em>p</em>, <em>reference_axis</em>, <em>tag=True</em>, <em>reference_axis_name='ref_axis'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#alpha_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.alpha_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Alpha angle calculated using points on the primitive of helix and axis.</p>
<p class="rubric">Notes</p>
<p>The final value is None, since the angle calculation requires pairs
of points along the primitive and axis. This is a generalisation
of the calculation used to measure the tilt of a helix in a
coiled-coil with respect to the central axis of the coiled coil.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</li>
<li><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, tags the Chain with the reference axis coordinates
and each Residue with its alpha angle. Alpha angles are stored
at the Residue level, but are calculated using the CA atom.</li>
<li><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>alphas</strong> – The alpha angle for the Polymer at each point of its primitive,
in degrees.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">list of float</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the Polymer and the reference_axis have unequal length.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.cc_to_local_params">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">cc_to_local_params</code><span class="sig-paren">(</span><em>pitch</em>, <em>radius</em>, <em>oligo</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#cc_to_local_params"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.cc_to_local_params" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns local parameters for an oligomeric assembly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pitch</strong> (<em>float</em>) – Pitch of assembly</li>
<li><strong>radius</strong> (<em>float</em>) – Radius of assembly</li>
<li><strong>oligo</strong> (<em>int</em>) – Oligomeric state of assembly</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>pitchloc</strong> (<em>float</em>) – Local pitch of assembly (between 2 adjacent component helices)</li>
<li><strong>rloc</strong> (<em>float</em>) – Local radius of assembly</li>
<li><strong>alphaloc</strong> (<em>float</em>) – Local pitch-angle of assembly</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.charge_series">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">charge_series</code><span class="sig-paren">(</span><em>seq</em>, <em>granularity=0.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#charge_series"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.charge_series" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the charge for pH 1-13.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</li>
<li><strong>granularity</strong> (<em>float</em><em>, </em><em>optional</em>) – Granularity of pH values i.e. if 0.1 pH = [1.0, 1.1, 1.2…]</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.crick_angles">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">crick_angles</code><span class="sig-paren">(</span><em>p</em>, <em>reference_axis</em>, <em>tag=True</em>, <em>reference_axis_name='ref_axis'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#crick_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.crick_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the Crick angle for each CA atom in the <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>The final value is in the returned list is <cite>None</cite>, since the angle
calculation requires pairs of points on both the primitive and
reference_axis.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</li>
<li><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, tags the <cite>Polymer</cite> with the reference axis coordinates
and each Residue with its Crick angle. Crick angles are stored
at the Residue level, but are calculated using the CA atom.</li>
<li><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>cr_angles</strong> – The crick angles in degrees for each CA atom of the Polymer.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">list(float)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the Polymer and the reference_axis have unequal length.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.flip_reference_axis_if_antiparallel">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">flip_reference_axis_if_antiparallel</code><span class="sig-paren">(</span><em>p</em>, <em>reference_axis</em>, <em>start_index=0</em>, <em>end_index=-1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#flip_reference_axis_if_antiparallel"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.flip_reference_axis_if_antiparallel" title="Permalink to this definition">¶</a></dt>
<dd><p>Flips reference axis if direction opposes the direction of the <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>If the angle between the vector for the Polymer and the vector
for the reference_axis is &gt; 90 degrees, then the reference axis
is reversed. This is useful to run before running
polymer_to_reference_axis_distances, crick_angles, or alpha_angles.
For more information on the start and end indices, see chain_vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</li>
<li><strong>start_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is 0 (start at the N-terminus of the Polymer)</li>
<li><strong>end_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is -1 (start at the C-terminus of the Polymer)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>reference_axis</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list(numpy.array or tuple or list)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.make_primitive">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">make_primitive</code><span class="sig-paren">(</span><em>cas_coords</em>, <em>window_length=3</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates running average of cas_coords with a fixed averaging window_length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</li>
<li><strong>window_length</strong> (<em>int</em><em>, </em><em>optional</em>) – The number of coordinate sets to average each time.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>s_primitive</strong> – Each array has length 3.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">list(numpy.array)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the length of cas_coords is smaller than the window_length.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.make_primitive_extrapolate_ends">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">make_primitive_extrapolate_ends</code><span class="sig-paren">(</span><em>cas_coords</em>, <em>smoothing_level=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive_extrapolate_ends"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive_extrapolate_ends" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates smoothed helix primitives and extrapolates lost ends.</p>
<p class="rubric">Notes</p>
<p>From an input list of CA coordinates, the running average is
calculated to form a primitive. The smoothing_level dictates how
many times to calculate the running average. A higher
smoothing_level generates a ‘smoother’ primitive - i.e. the
points on the primitive more closely fit a smooth curve in R^3.
Each time the smoothing level is increased by 1, a point is lost
from either end of the primitive. To correct for this, the primitive
is extrapolated at the ends to approximate the lost values. There
is a trade-off then between the smoothness of the primitive and
its accuracy at the ends.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</li>
<li><strong>smoothing_level</strong> (<em>int</em>) – Number of times to run the averaging.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>final_primitive</strong> – Each array has length 3.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list(numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.make_primitive_smoothed">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">make_primitive_smoothed</code><span class="sig-paren">(</span><em>cas_coords</em>, <em>smoothing_level=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive_smoothed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive_smoothed" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates smoothed primitive from a list of coordinates.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</li>
<li><strong>smoothing_level</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of times to run the averaging.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>s_primitive</strong> – Each array has length 3.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">list(numpy.array)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the smoothing level is too great compared to the length
of cas_coords.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.measure_sidechain_torsion_angles">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">measure_sidechain_torsion_angles</code><span class="sig-paren">(</span><em>residue</em>, <em>verbose=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#measure_sidechain_torsion_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.measure_sidechain_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates sidechain dihedral angles for a residue</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>residue</strong> (<em>[</em><em>ampal.Residue</em><em>]</em>) – <cite>Residue</cite> object.</li>
<li><strong>verbose</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, tells you when a residue does not have any known
dihedral angles to measure.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><p><strong>chi_angles</strong> – Length depends on residue type, in range [-pi, pi]</p>
<p>[0] = chi1 [if applicable]
[1] = chi2 [if applicable]
[2] = chi3 [if applicable]
[3] = chi4 [if applicable]</p>
</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">[float]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.measure_torsion_angles">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">measure_torsion_angles</code><span class="sig-paren">(</span><em>residues</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#measure_torsion_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.measure_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the dihedral angles for a list of backbone atoms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>residues</strong> (<em>[</em><em>ampal.Residue</em><em>]</em>) – List of <cite>Residue</cite> objects.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>torsion_angles</strong> – One triple for each residue, containing torsion angles in
the range [-pi, pi].<blockquote>
<div>[0] omega
[1] phi
[2] psi</div></blockquote>
<p>For the first residue, omega and phi are not defined. For
the final residue, psi is not defined.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">(float, float, float)</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the number of input residues is less than 2.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.partial_charge">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">partial_charge</code><span class="sig-paren">(</span><em>aa</em>, <em>pH</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#partial_charge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.partial_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the partial charge of the amino acid.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>aa</strong> (<em>str</em>) – Amino acid single-letter code.</li>
<li><strong>pH</strong> (<em>float</em>) – pH of interest.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.polymer_to_reference_axis_distances">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">polymer_to_reference_axis_distances</code><span class="sig-paren">(</span><em>p</em>, <em>reference_axis</em>, <em>tag=True</em>, <em>reference_axis_name='ref_axis'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#polymer_to_reference_axis_distances"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.polymer_to_reference_axis_distances" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns distances between the primitive of a Polymer and a reference_axis.</p>
<p class="rubric">Notes</p>
<p>Distances are calculated between each point of the Polymer primitive
and the corresponding point in reference_axis. In the special case
of the helical barrel, if the Polymer is a helix and the reference_axis
represents the centre of the barrel, then this function returns the
radius of the barrel at each point on the helix primitive. The points
of the primitive and the reference_axis are run through in the same
order, so take care with the relative orientation of the reference
axis when defining it.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>ampal.Polymer</em>) – </li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</li>
<li><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, tags the Chain with the reference axis coordinates
and each Residue with its distance to the ref axis.
Distances are stored at the Residue level, but refer to
distances from the CA atom.</li>
<li><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>distances</strong> – Distance values between corresponding points on the
reference axis and the <cite>Polymer</cite> <cite>Primitive</cite>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">list(float)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – If the Polymer and the reference_axis have unequal length.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.polypeptide_vector">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">polypeptide_vector</code><span class="sig-paren">(</span><em>p</em>, <em>start_index=0</em>, <em>end_index=-1</em>, <em>unit=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#polypeptide_vector"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.polypeptide_vector" title="Permalink to this definition">¶</a></dt>
<dd><p>Vector along the Chain primitive (default is from N-terminus to C-terminus).</p>
<p class="rubric">Notes</p>
<p><cite>start_index</cite> and <cite>end_index</cite> can be changed to examine smaller
sections of the Chain, or reversed to change the direction of
the vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</li>
<li><strong>start_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is 0 (start at the N-terminus of the Chain)</li>
<li><strong>end_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is -1 (start at the C-terminus of the Chain)</li>
<li><strong>unit</strong> (<em>bool</em>) – If True, the vector returned has a magnitude of 1.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>vector</strong> – vector has shape (1, 3)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">a numpy.array</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.reference_axis_from_chains">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">reference_axis_from_chains</code><span class="sig-paren">(</span><em>chains</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#reference_axis_from_chains"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.reference_axis_from_chains" title="Permalink to this definition">¶</a></dt>
<dd><p>Average coordinates from a set of primitives calculated from Chains.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>chains</strong> (<em>list</em><em>(</em><em>Chain</em><em>)</em>) – </td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>reference_axis</strong> – The averaged (x, y, z) coordinates of the primitives for
the list of Chains. In the case of a coiled coil barrel,
this would give the central axis for calculating e.g. Crick
angles.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">numpy.array</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><em>ValueError :</em> – If the Chains are not all of the same length.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.residues_per_turn">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">residues_per_turn</code><span class="sig-paren">(</span><em>p</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#residues_per_turn"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.residues_per_turn" title="Permalink to this definition">¶</a></dt>
<dd><p>The number of residues per turn at each Monomer in the Polymer.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the number of residues
per turn, at a point on the Polymer primitive. Calculated using
the relative positions of the CA atoms and the primitive of the
Polymer. Element i is the calculated from the dihedral angle using
the CA atoms of the Monomers with indices [i, i+1] and the
corresponding atoms of the primitive. The final value is None.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>p</strong> (<em>ampal.Polypeptide</em>) – <cite>Polypeptide</cite> from which residues per turn will be calculated.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>rpts</strong> – Residue per turn values.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">[float]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.sequence_charge">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">sequence_charge</code><span class="sig-paren">(</span><em>seq</em>, <em>pH=7.4</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_charge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the total charge of the input polypeptide sequence.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</li>
<li><strong>pH</strong> (<em>float</em>) – pH of interest.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.sequence_isoelectric_point">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">sequence_isoelectric_point</code><span class="sig-paren">(</span><em>seq</em>, <em>granularity=0.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_isoelectric_point"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the isoelectric point of the sequence for ph 1-13.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</li>
<li><strong>granularity</strong> (<em>float</em><em>, </em><em>optional</em>) – Granularity of pH values i.e. if 0.1 pH = [1.0, 1.1, 1.2…]</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.sequence_molar_extinction_280">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">sequence_molar_extinction_280</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_molar_extinction_280"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molar extinction coefficient of the sequence at 280 nm.</p>
<p class="rubric">Notes</p>
<p>Units = M/cm</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.analyse_protein.sequence_molecular_weight">
<code class="descclassname">ampal.analyse_protein.</code><code class="descname">sequence_molecular_weight</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_molecular_weight"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the polypeptide sequence.</p>
<p class="rubric">Notes</p>
<p>Units = Daltons</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.assembly">
<span id="ampal-assembly-module"></span><h2>ampal.assembly module<a class="headerlink" href="#module-ampal.assembly" title="Permalink to this headline">¶</a></h2>
<p>Defines various containers for AMPAL objects.</p>
<dl class="class">
<dt id="ampal.assembly.AmpalContainer">
<em class="property">class </em><code class="descclassname">ampal.assembly.</code><code class="descname">AmpalContainer</code><span class="sig-paren">(</span><em>ampal_objects=None</em>, <em>id=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Custom list type class that holds multiple model states.</p>
<p class="rubric">Notes</p>
<p>In this case, a state is defined as a set of coordinates that
represents a protein model and an associated score or set of scores.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>ampal_objects</strong> (<em>[</em><em>AMPAL</em><em>]</em><em>, </em><em>optional</em>) – A list of AMPAL objects with which to initialise the
AMPAL container. This can be an <cite>Assembly</cite>, <cite>Polymer</cite>
or <cite>Monomer</cite>.</li>
<li><strong>id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the AMPAL container.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.assembly.AmpalContainer.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.assembly.AmpalContainer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for the AMPAL container.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.AmpalContainer.append">
<code class="descname">append</code><span class="sig-paren">(</span><em>item</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.append"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds an AMPAL object to the <cite>AmpalContainer</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.AmpalContainer.extend">
<code class="descname">extend</code><span class="sig-paren">(</span><em>ampal_container</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.extend"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends an <cite>AmpalContainer</cite> with another <cite>AmpalContainer</cite>.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.AmpalContainer.pdb">
<code class="descname">pdb</code><a class="headerlink" href="#ampal.assembly.AmpalContainer.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Compiles the PDB strings for each state into a single file.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.AmpalContainer.sort_by_tag">
<code class="descname">sort_by_tag</code><span class="sig-paren">(</span><em>tag</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.sort_by_tag"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.sort_by_tag" title="Permalink to this definition">¶</a></dt>
<dd><p>Sorts the <cite>AmpalContainer</cite> by a tag on the component objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>tag</strong> (<em>str</em>) – Key of tag used for sorting.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.assembly.Assembly">
<em class="property">class </em><code class="descclassname">ampal.assembly.</code><code class="descname">Assembly</code><span class="sig-paren">(</span><em>molecules=None</em>, <em>assembly_id=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.BaseAmpal</span></code></a></p>
<p>A container that holds <cite>Polymer</cite> type objects.</p>
<p class="rubric">Notes</p>
<p>Has a simple hierarchy: <cite>Assembly</cite> contains one or more <cite>Polymer</cite>,
which in turn contains one or more <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>molecules</strong> (<a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Polymer</cite> or list containing <cite>Polymer</cite> objects to be assembled.</li>
<li><strong>assembly_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Assembly</cite>. This
is used when generating a pdb file using <cite>Assembly().pdb</cite>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – <cite>Assembly</cite> objects can only be initialised empty, using a <cite>Polymer</cite>
or a list of <cite>Polymers</cite>.</p>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="ampal.assembly.Assembly.append">
<code class="descname">append</code><span class="sig-paren">(</span><em>item</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.append"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds a <cite>Polymer</cite> to the <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – Raised if other is any type other than <cite>Polymer</cite>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.backbone">
<code class="descname">backbone</code><a class="headerlink" href="#ampal.assembly.Assembly.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a new <cite>Assembly</cite> containing only the backbone atoms.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.
Sequence data is retained, but only the main chain atoms are
retained.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>bb_assembly</strong> – <cite>Assembly</cite> containing only the backbone atoms of the original
<cite>Assembly</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">ampal.Protein</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.extend">
<code class="descname">extend</code><span class="sig-paren">(</span><em>assembly</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.extend"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends the <cite>Assembly</cite> with the contents of another <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – Raised if other is any type other than <cite>Assembly</cite>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.fasta">
<code class="descname">fasta</code><a class="headerlink" href="#ampal.assembly.Assembly.fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a FASTA string for the <cite>Assembly</cite>.</p>
<p class="rubric">Notes</p>
<p>Explanation of FASTA format: <a class="reference external" href="https://en.wikipedia.org/wiki/FASTA_format">https://en.wikipedia.org/wiki/FASTA_format</a>
Recommendation that all lines of text be shorter than 80
characters is adhered to. Format of PDBID|CHAIN|SEQUENCE is
consistent with files downloaded from the PDB. Uppercase
PDBID used for consistency with files downloaded from the PDB.
Useful for feeding into cdhit and then running sequence clustering.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>fasta_str</strong> – String of the fasta file for the <cite>Assembly</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.filter_mol_types">
<code class="descname">filter_mol_types</code><span class="sig-paren">(</span><em>mol_types</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.filter_mol_types"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.filter_mol_types" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.get_atoms">
<code class="descname">get_atoms</code><span class="sig-paren">(</span><em>ligands=True</em>, <em>pseudo_group=False</em>, <em>inc_alt_states=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Flat list of all the <cite>Atoms</cite> in the <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include ligand <cite>Atoms</cite>.</li>
<li><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include pseudo_group <cite>Atoms</cite>.</li>
<li><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include alternate sidechain conformations.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>atoms</strong> – All the <cite>Atoms</cite> as a iterator.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">itertools.chain</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.get_ligands">
<code class="descname">get_ligands</code><span class="sig-paren">(</span><em>solvent=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_ligands"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all ligands from the <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>solvent</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, solvent molecules will be included.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.get_monomers">
<code class="descname">get_monomers</code><span class="sig-paren">(</span><em>ligands=True</em>, <em>pseudo_group=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all the <cite>Monomers</cite> from the <cite>Assembly</cite> object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, will include ligand <cite>Monomers</cite>.</li>
<li><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include pseudo atoms.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.is_within">
<code class="descname">is_within</code><span class="sig-paren">(</span><em>cutoff_dist</em>, <em>point</em>, <em>ligands=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.is_within"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.is_within" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in AMPAL object within <cite>cut-off</cite> distance from the <cite>point</cite>.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.isoelectric_point">
<code class="descname">isoelectric_point</code><a class="headerlink" href="#ampal.assembly.Assembly.isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the isoelectric point of the <cite>Assembly</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.make_pdb">
<code class="descname">make_pdb</code><span class="sig-paren">(</span><em>ligands=True</em>, <em>alt_states=False</em>, <em>pseudo_group=False</em>, <em>header=True</em>, <em>footer=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.make_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the Assembly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include ligands in the output.</li>
<li><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include alternate conformations in the output.</li>
<li><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include pseudo atoms in the output.</li>
<li><strong>header</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> will write a header for output.</li>
<li><strong>footer</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> will write a footer for output.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>pdb_str</strong> – String of the pdb for the Assembly. Generated by collating
Polymer().pdb calls for the component Polymers.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">str</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.molar_extinction_280">
<code class="descname">molar_extinction_280</code><a class="headerlink" href="#ampal.assembly.Assembly.molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the extinction co-efficient of the <cite>Assembly</cite> at 280 nm.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.molecular_weight">
<code class="descname">molecular_weight</code><a class="headerlink" href="#ampal.assembly.Assembly.molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the <cite>Assembly</cite> in Daltons.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.pdb">
<code class="descname">pdb</code><a class="headerlink" href="#ampal.assembly.Assembly.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs make_pdb in default mode.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.primitives">
<code class="descname">primitives</code><a class="headerlink" href="#ampal.assembly.Assembly.primitives" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a new <cite>Assembly</cite> containing the primitives of each Polymer.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>prim_assembly</strong> – <cite>Assembly</cite> containing only the primitives of the <cite>Polymers</cite>
in the original <cite>Assembly</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">ampal.Protein</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.relabel_all">
<code class="descname">relabel_all</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_all"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Polymers, Monomers and Atoms with default labeling.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.relabel_atoms">
<code class="descname">relabel_atoms</code><span class="sig-paren">(</span><em>start=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Atoms in numerical order, offset by the start parameter.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>start</strong> (<em>int</em><em>, </em><em>optional</em>) – Defines an offset for the labelling.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.relabel_monomers">
<code class="descname">relabel_monomers</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Monomers in the component Polymers in numerical order.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.relabel_polymers">
<code class="descname">relabel_polymers</code><span class="sig-paren">(</span><em>labels=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_polymers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_polymers" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Relabels the component Polymers either in alphabetical order or</dt>
<dd>using a list of labels.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>labels</strong> (<em>list</em><em>, </em><em>optional</em>) – A list of new labels.</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if the number of labels does not match the number of
component Polymer objects.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.assembly.Assembly.sequences">
<code class="descname">sequences</code><a class="headerlink" href="#ampal.assembly.Assembly.sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of each <cite>Polymer</cite> in the <cite>Assembly</cite> as a list.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>sequences</strong> – List of sequences.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">[str]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.tag_atoms_unique_ids">
<code class="descname">tag_atoms_unique_ids</code><span class="sig-paren">(</span><em>force=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.tag_atoms_unique_ids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.tag_atoms_unique_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each Atom in the Assembly with its unique_id.</p>
<p class="rubric">Notes</p>
<p>The unique_id for each atom is a tuple (a double). <cite>unique_id[0]</cite>
is the unique_id for its parent <cite>Monomer</cite> (see <cite>Monomer.unique_id</cite>
for more information). <cite>unique_id[1]</cite> is the atom_type in the
<cite>Assembly</cite> as a string, e.g. ‘CA’, ‘CD2’.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True the tag will be run even if Atoms are already tagged.
If False, only runs if at least one Atom is not tagged.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.tag_ca_geometry">
<code class="descname">tag_ca_geometry</code><span class="sig-paren">(</span><em>force=False</em>, <em>reference_axis=None</em>, <em>reference_axis_name='ref_axis'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.tag_ca_geometry"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.tag_ca_geometry" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Monomer</cite> in the <cite>Assembly</cite> with its helical geometry.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True the tag will be run even if <cite>Monomers</cite> are already tagged.</li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em><em>, </em><em>optional</em>) – Coordinates to feed to geometry functions that depend on
having a reference axis.</li>
<li><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at <cite>Chain</cite> and <cite>Residue</cite> level.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.assembly.Assembly.tag_torsion_angles">
<code class="descname">tag_torsion_angles</code><span class="sig-paren">(</span><em>force=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.tag_torsion_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.assembly.Assembly.tag_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Monomer</cite> in the <cite>Assembly</cite> with its torsion angles.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, the tag will be run even if <cite>Monomers</cite> are already
tagged.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-ampal.base_ampal">
<span id="ampal-base-ampal-module"></span><h2>ampal.base_ampal module<a class="headerlink" href="#module-ampal.base_ampal" title="Permalink to this headline">¶</a></h2>
<p>Contains the base and common classes for all AMPAL objects.</p>
<dl class="class">
<dt id="ampal.base_ampal.Atom">
<em class="property">class </em><code class="descclassname">ampal.base_ampal.</code><code class="descname">Atom</code><span class="sig-paren">(</span><em>coordinates</em>, <em>element</em>, <em>atom_id=' '</em>, <em>res_label=None</em>, <em>occupancy=1.0</em>, <em>bfactor=1.0</em>, <em>charge=' '</em>, <em>state='A'</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Object containing atomic coordinates and element information.</p>
<p class="rubric">Notes</p>
<p><cite>Atom</cite> is an AMPAL object, but it does not inherit from <cite>BaseAmpal</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>coordinates</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Position of <cite>Atom</cite> in 3D space.</li>
<li><strong>element</strong> (<em>str</em>) – Element of <cite>Atom</cite>.</li>
<li><strong>atom_id</strong> (<em>str</em>) – Identifier for <cite>Atom</cite>, usually a number.</li>
<li><strong>res_label</strong> (<em>str</em><em>, </em><em>optional</em>) – Label used in <cite>Monomer</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</li>
<li><strong>occupancy</strong> (<em>float</em><em>, </em><em>optional</em>) – The occupancy of the <cite>Atom</cite>.</li>
<li><strong>bfactor</strong> (<em>float</em><em>, </em><em>optional</em>) – The bfactor of the <cite>Atom</cite>.</li>
<li><strong>charge</strong> (<em>str</em><em>, </em><em>optional</em>) – The point charge of the <cite>Atom</cite>.</li>
<li><strong>state</strong> (<em>str</em>) – The state of this <cite>Atom</cite>. Used to identify <cite>Atoms</cite> with a
number of conformations.</li>
<li><strong>parent</strong> (<em>ampal.Monomer</em><em>, </em><em>optional</em>) – A reference to the <cite>Monomer</cite> containing this <cite>Atom</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.base_ampal.Atom.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.base_ampal.Atom.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for <cite>Atom</cite>, usually a number.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.res_label">
<code class="descname">res_label</code><a class="headerlink" href="#ampal.base_ampal.Atom.res_label" title="Permalink to this definition">¶</a></dt>
<dd><p>Label used in <cite>Monomer</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.element">
<code class="descname">element</code><a class="headerlink" href="#ampal.base_ampal.Atom.element" title="Permalink to this definition">¶</a></dt>
<dd><p>Element of <cite>Atom</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.base_ampal.Atom.parent" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>A reference to the <cite>Monomer</cite> containing this <cite>Atom</cite>.</dt>
<dd>number of conformations.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Monomer</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.base_ampal.Atom.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.array">
<code class="descname">array</code><a class="headerlink" href="#ampal.base_ampal.Atom.array" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts the atomic coordinate to a <cite>numpy.array</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Atom.rotate">
<code class="descname">rotate</code><span class="sig-paren">(</span><em>angle</em>, <em>axis</em>, <em>point=None</em>, <em>radians=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom.rotate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Atom.rotate" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotates <cite>Atom</cite> by <cite>angle</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>angle</strong> (<em>float</em>) – Angle that <cite>Atom</cite> will be rotated.</li>
<li><strong>axis</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Axis about which the <cite>Atom</cite> will be rotated.</li>
<li><strong>point</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em><em>, </em><em>optional</em>) – Point that the <cite>axis</cite> lies upon. If <cite>None</cite> then the origin is used.</li>
<li><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – True is <cite>angle</cite> is define in radians, False is degrees.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Atom.translate">
<code class="descname">translate</code><span class="sig-paren">(</span><em>vector</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom.translate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Atom.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates <cite>Atom</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>vector</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Vector used for translation.</li>
<li><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.unique_id">
<code class="descname">unique_id</code><a class="headerlink" href="#ampal.base_ampal.Atom.unique_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a unique ID for the <cite>Atom</cite> based on its parents.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>unique_id</strong> – (polymer.id, residue.id, atom.id)</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">(str, str, str)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.x">
<code class="descname">x</code><a class="headerlink" href="#ampal.base_ampal.Atom.x" title="Permalink to this definition">¶</a></dt>
<dd><p>The x coordinate.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.y">
<code class="descname">y</code><a class="headerlink" href="#ampal.base_ampal.Atom.y" title="Permalink to this definition">¶</a></dt>
<dd><p>The y coordinate.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Atom.z">
<code class="descname">z</code><a class="headerlink" href="#ampal.base_ampal.Atom.z" title="Permalink to this definition">¶</a></dt>
<dd><p>The z coordinate.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.base_ampal.BaseAmpal">
<em class="property">class </em><code class="descclassname">ampal.base_ampal.</code><code class="descname">BaseAmpal</code><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Base class for all AMPAL objects except <cite>ampal.atom</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">NotImplementedError</span></code> – <cite>BaseAmpal</cite> is an abstract base class and is not intended to
be instanciated. A <cite>NotImplementedError</cite> is raised if a
method is called that is required on a child class but is
not implemented in <cite>BaseAmpal</cite>.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.base_ampal.BaseAmpal.centre_of_mass">
<code class="descname">centre_of_mass</code><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.centre_of_mass" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the centre of mass of AMPAL object.</p>
<p class="rubric">Notes</p>
<p>All atoms are included in calculation, call <cite>centre_of_mass</cite>
manually if another selection is require.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>centre_of_mass</strong> – 3D coordinate for the centre of mass.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">numpy.array</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.get_atoms">
<code class="descname">get_atoms</code><span class="sig-paren">(</span><em>ligands=True</em>, <em>inc_alt_states=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.get_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.is_within">
<code class="descname">is_within</code><span class="sig-paren">(</span><em>cutoff_dist</em>, <em>point</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.is_within"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.is_within" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in ampal object within <cite>cut-off</cite> distance from the <cite>point</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.make_pdb">
<code class="descname">make_pdb</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.make_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.BaseAmpal.pdb">
<code class="descname">pdb</code><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs make_pdb in default mode.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.rmsd">
<code class="descname">rmsd</code><span class="sig-paren">(</span><em>other</em>, <em>backbone=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.rmsd"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.rmsd" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the RMSD between two AMPAL objects.</p>
<p class="rubric">Notes</p>
<p>No fitting operation is performs and both AMPAL objects must
have the same number of atoms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>other</strong> (<em>AMPAL Object</em>) – Any AMPAL object with <cite>get_atoms</cite> method.</li>
<li><strong>backbone</strong> (<em>bool</em><em>, </em><em>optional</em>) – Calculates RMSD of backbone only.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.rotate">
<code class="descname">rotate</code><span class="sig-paren">(</span><em>angle</em>, <em>axis</em>, <em>point=None</em>, <em>radians=False</em>, <em>inc_alt_states=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.rotate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.rotate" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotates every atom in the AMPAL object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>angle</strong> (<em>float</em>) – Angle that AMPAL object will be rotated.</li>
<li><strong>axis</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Axis about which the AMPAL object will be rotated.</li>
<li><strong>point</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em><em>, </em><em>optional</em>) – Point that the axis lies upon. If <cite>None</cite> then the origin is used.</li>
<li><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – True is <cite>angle</cite> is define in radians, False is degrees.</li>
<li><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.BaseAmpal.translate">
<code class="descname">translate</code><span class="sig-paren">(</span><em>vector</em>, <em>inc_alt_states=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.translate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates every atom in the AMPAL object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>vector</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Vector used for translation.</li>
<li><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.base_ampal.Monomer">
<em class="property">class </em><code class="descclassname">ampal.base_ampal.</code><code class="descname">Monomer</code><span class="sig-paren">(</span><em>atoms=None</em>, <em>monomer_id=' '</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.BaseAmpal</span></code></a></p>
<p>Groups of <cite>Atoms</cite> that form <cite>Polymers</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atoms</strong> (<em>OrderedDict</em><em> or </em><em>{OrderedDict}</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the dictionary.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</li>
<li><strong>parent</strong> (<a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em>, </em><em>optional</em>) – A reference to the <cite>Polymer</cite> containing this <cite>Monomer</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.states">
<code class="descname">states</code><a class="headerlink" href="#ampal.base_ampal.Monomer.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.base_ampal.Monomer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.base_ampal.Monomer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polymer</cite> containing this <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.base_ampal.Monomer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.active_state">
<code class="descname">active_state</code><a class="headerlink" href="#ampal.base_ampal.Monomer.active_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Defines which state dictionary should be used currently.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Monomer.atoms">
<code class="descname">atoms</code><a class="headerlink" href="#ampal.base_ampal.Monomer.atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Atoms in the currently active state.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Monomer.close_monomers">
<code class="descname">close_monomers</code><span class="sig-paren">(</span><em>group</em>, <em>cutoff=4.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.close_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.close_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of Monomers from within a cut off distance of the Monomer</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>group</strong> (<a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><em>BaseAmpal</em></a><em> or </em><em>Subclass</em>) – Group to be search for Monomers that are close to this Monomer.</li>
<li><strong>cutoff</strong> (<em>float</em>) – Distance cut off.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>nearby_residues</strong> – List of Monomers within cut off distance.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">[Monomers]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Monomer.get_atoms">
<code class="descname">get_atoms</code><span class="sig-paren">(</span><em>inc_alt_states=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.get_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in the <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will return <cite>Atoms</cite> for alternate states.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Monomer.get_monomers">
<code class="descname">get_monomers</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.get_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the this <cite>Monomer</cite>.</p>
<p class="rubric">Notes</p>
<p>This function is only present for consistency in the interface.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Monomer.make_pdb">
<code class="descname">make_pdb</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.make_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>Monomer</cite>.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.base_ampal.Polymer">
<em class="property">class </em><code class="descclassname">ampal.base_ampal.</code><code class="descname">Polymer</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>ligands=None</em>, <em>polymer_id=' '</em>, <em>molecule_type=''</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.BaseAmpal</span></code></a></p>
<p>A container that holds <cite>Monomer</cite> type objects.</p>
<p class="rubric">Notes</p>
<p><cite>Polymer</cite> has a simple hierarchy: A <cite>Polymer</cite> contains one or
more <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomers</strong> (<a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em>]</em><em>, </em><em>optional</em>) – Monomer or list containing Monomer objects to form the Polymer().</li>
<li><strong>ligands</strong> (<a class="reference internal" href="#ampal.ligands.LigandGroup" title="ampal.ligands.LigandGroup"><em>LigandGroup</em></a><em>, </em><em>optional</em>) – <cite>Ligands</cite> associated with the <cite>Polymer</cite>.</li>
<li><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polymer</cite>. This is
used when generating a pdb file using <cite>Polymer().pdb</cite>.</li>
<li><strong>molecule_type</strong> (<em>str</em><em>, </em><em>optional</em>) – A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</li>
<li><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.base_ampal.Polymer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Polymer ID</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.base_ampal.Polymer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Assembly or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.molecule_type">
<code class="descname">molecule_type</code><a class="headerlink" href="#ampal.base_ampal.Polymer.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.ligands">
<code class="descname">ligands</code><a class="headerlink" href="#ampal.base_ampal.Polymer.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polymer</cite> chain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.LigandGroup</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.base_ampal.Polymer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.base_ampal.Polymer.sl">
<code class="descname">sl</code><a class="headerlink" href="#ampal.base_ampal.Polymer.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – Polymer objects can only be initialised empty, using a Monomer
or a list of Monomers.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="ampal.base_ampal.Polymer.append">
<code class="descname">append</code><span class="sig-paren">(</span><em>item</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.append"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Appends a <cite>Monomer to the `Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>Does not update labelling.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.extend">
<code class="descname">extend</code><span class="sig-paren">(</span><em>polymer</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.extend"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends the <cite>Polymer</cite> with the contents of another <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>Does not update labelling.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.get_atoms">
<code class="descname">get_atoms</code><span class="sig-paren">(</span><em>ligands=True</em>, <em>inc_alt_states=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Flat list of all the Atoms in the Polymer.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>inc_alt_states</strong> (<em>bool</em>) – If true atoms from alternate conformations are included rather
than only the “active” states.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>atoms</strong> – Returns an iterator of all the atoms. Convert to list if you
require indexing.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">itertools.chain</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.get_monomers">
<code class="descname">get_monomers</code><span class="sig-paren">(</span><em>ligands=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all the <cite>Monomers</cite> from the AMPAL object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will include ligand <cite>Monomers</cite>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.get_reference_coords">
<code class="descname">get_reference_coords</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_reference_coords"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_reference_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets list of coordinates of all reference atoms in the <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>ref_coords</strong> – List has the same length as the <cite>Polymer</cite>.
The first, second and third elements of array i contain the
x, y and z coordinates of the i-th reference atom.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">[numpy.array]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.make_pdb">
<code class="descname">make_pdb</code><span class="sig-paren">(</span><em>alt_states=False</em>, <em>inc_ligands=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.make_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include alternate conformations for <cite>Monomers</cite> in PDB.</li>
<li><strong>inc_ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – Includes <cite>Ligands</cite> in PDB.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>pdb_str</strong> – String of the pdb for the <cite>Polymer</cite>. Generated using information
from the component <cite>Monomers</cite>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">str</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.relabel_all">
<code class="descname">relabel_all</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_all"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all <cite>Monomers</cite> and <cite>Atoms</cite> using default labeling.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.relabel_atoms">
<code class="descname">relabel_atoms</code><span class="sig-paren">(</span><em>start=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all <cite>Atoms</cite> in numerical order.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>start</strong> (<em>int</em><em>, </em><em>optional</em>) – Offset the labelling by <cite>start</cite> residues.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.base_ampal.Polymer.relabel_monomers">
<code class="descname">relabel_monomers</code><span class="sig-paren">(</span><em>labels=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_monomers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels the either in numerically or using a list of labels.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>labels</strong> (<em>list</em><em>, </em><em>optional</em>) – A list of new labels.</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if the number of labels does not match the number of
component Monoer objects.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.base_ampal.cap">
<code class="descclassname">ampal.base_ampal.</code><code class="descname">cap</code><span class="sig-paren">(</span><em>v</em>, <em>l</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#cap"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.cap" title="Permalink to this definition">¶</a></dt>
<dd><p>Shortens string is above certain length.</p>
</dd></dl>

<dl class="function">
<dt id="ampal.base_ampal.centre_of_atoms">
<code class="descclassname">ampal.base_ampal.</code><code class="descname">centre_of_atoms</code><span class="sig-paren">(</span><em>atoms</em>, <em>mass_weighted=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#centre_of_atoms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.centre_of_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns centre point of any list of atoms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atoms</strong> (<em>list</em>) – List of AMPAL atom objects.</li>
<li><strong>mass_weighted</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True returns centre of mass, otherwise just geometric centre of points.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>centre_of_mass</strong> – 3D coordinate for the centre of mass.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">numpy.array</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.base_ampal.find_atoms_within_distance">
<code class="descclassname">ampal.base_ampal.</code><code class="descname">find_atoms_within_distance</code><span class="sig-paren">(</span><em>atoms</em>, <em>cutoff_distance</em>, <em>point</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#find_atoms_within_distance"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.find_atoms_within_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns atoms within the distance from the point.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atoms</strong> (<em>[</em><em>ampal.atom</em><em>]</em>) – A list of <cite>ampal.atoms</cite>.</li>
<li><strong>cutoff_distance</strong> (<em>float</em>) – Maximum distance from point.</li>
<li><strong>point</strong> (<em>(</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>)</em>) – Reference point, 3D coordinate.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>filtered_atoms</strong> – <cite>atoms</cite> list filtered by distance.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">[ampal.atoms]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.base_ampal.write_pdb">
<code class="descclassname">ampal.base_ampal.</code><code class="descname">write_pdb</code><span class="sig-paren">(</span><em>residues</em>, <em>chain_id=' '</em>, <em>alt_states=False</em>, <em>strip_states=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#write_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.base_ampal.write_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes a pdb file for a list of residues.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>residues</strong> (<em>list</em>) – List of Residue objects.</li>
<li><strong>chain_id</strong> (<em>str</em>) – String of the chain id, defaults to ‘ ‘.</li>
<li><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, include all occupancy states of residues, else outputs primary state only.</li>
<li><strong>strip_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, remove all state labels from residues. Only use with alt_states false.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>pdb_str</strong> – String of the PDB file.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">str</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.data">
<span id="ampal-data-module"></span><h2>ampal.data module<a class="headerlink" href="#module-ampal.data" title="Permalink to this headline">¶</a></h2>
<p>Loads data required for running ISAMBARD.</p>
</div>
<div class="section" id="module-ampal.dssp">
<span id="ampal-dssp-module"></span><h2>ampal.dssp module<a class="headerlink" href="#module-ampal.dssp" title="Permalink to this headline">¶</a></h2>
<p>This module provides an interface to the program DSSP.</p>
<p>For more information on DSSP see <a class="footnote-reference" href="#id6" id="id5">[4]</a>.</p>
<p class="rubric">References</p>
<table class="docutils footnote" frame="void" id="id6" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id5">[4]</a></td><td>Kabsch W, Sander C (1983) “Dictionary of protein
secondary structure: pattern recognition of hydrogen-bonded
and geometrical features”, Biopolymers, 22, 2577-637.</td></tr>
</tbody>
</table>
<dl class="function">
<dt id="ampal.dssp.dssp_available">
<code class="descclassname">ampal.dssp.</code><code class="descname">dssp_available</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#dssp_available"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.dssp_available" title="Permalink to this definition">¶</a></dt>
<dd><p>True if mkdssp is available on the path.</p>
</dd></dl>

<dl class="function">
<dt id="ampal.dssp.extract_all_ss_dssp">
<code class="descclassname">ampal.dssp.</code><code class="descname">extract_all_ss_dssp</code><span class="sig-paren">(</span><em>in_dssp</em>, <em>path=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#extract_all_ss_dssp"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.extract_all_ss_dssp" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses DSSP to extract secondary structure information on every residue.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>in_dssp</strong> (<em>str</em>) – Path to DSSP file.</li>
<li><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – Indicates if pdb is a path or a string.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><p><strong>dssp_residues</strong> –</p>
<dl class="docutils">
<dt>Each internal list contains:</dt>
<dd><p class="first last">[0] int Residue number
[1] str Secondary structure type
[2] str Chain identifier
[3] str Residue type
[4] float Phi torsion angle
[5] float Psi torsion angle
[6] int dssp solvent accessibility</p>
</dd>
</dl>
</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">[tuple]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.dssp.find_ss_regions">
<code class="descclassname">ampal.dssp.</code><code class="descname">find_ss_regions</code><span class="sig-paren">(</span><em>dssp_residues</em>, <em>loop_assignments=(' '</em>, <em>'B'</em>, <em>'S'</em>, <em>'T')</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#find_ss_regions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.find_ss_regions" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates parsed DSSP data into groups of secondary structure.</p>
<p class="rubric">Notes</p>
<p>Example: all residues in a single helix/loop/strand will be gathered
into a list, then the next secondary structure element will be
gathered into a separate list, and so on.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>dssp_residues</strong> (<em>[</em><em>tuple</em><em>]</em>) – <dl class="docutils">
<dt>Each internal list contains:</dt>
<dd>[0] int Residue number
[1] str Secondary structure type
[2] str Chain identifier
[3] str Residue type
[4] float Phi torsion angle
[5] float Psi torsion angle
[6] int dssp solvent accessibility</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>fragments</strong> – Lists grouped in continuous regions of secondary structure.
Innermost list has the same format as above.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">[[list]]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.dssp.get_ss_regions">
<code class="descclassname">ampal.dssp.</code><code class="descname">get_ss_regions</code><span class="sig-paren">(</span><em>assembly</em>, <em>ss_types</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#get_ss_regions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.get_ss_regions" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an Assembly containing Polymers for each region of structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>assembly</strong> (<em>ampal.Assembly</em>) – <cite>Assembly</cite> object to be searched secondary structure regions.</li>
<li><strong>ss_types</strong> (<em>list</em>) – List of secondary structure tags to be separate i.e. [‘H’]
would return helices, [‘H’, ‘E’] would return helices
and strands.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>fragments</strong> – <cite>Assembly</cite> containing a <cite>Polymer</cite> for each region of specified
secondary structure.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#ampal.assembly.Assembly" title="ampal.assembly.Assembly">Assembly</a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.dssp.run_dssp">
<code class="descclassname">ampal.dssp.</code><code class="descname">run_dssp</code><span class="sig-paren">(</span><em>pdb</em>, <em>path=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#run_dssp"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.run_dssp" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses DSSP to find helices and extracts helices from a pdb file or string.
:param pdb: Path to pdb file or string.
:type pdb: str
:param path: Indicates if pdb is a path or a string.
:type path: bool, optional</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>dssp_out</strong> – Std out from DSSP.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.dssp.tag_dssp_data">
<code class="descclassname">ampal.dssp.</code><code class="descname">tag_dssp_data</code><span class="sig-paren">(</span><em>assembly</em>, <em>loop_assignments=(' '</em>, <em>'B'</em>, <em>'S'</em>, <em>'T')</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#tag_dssp_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.dssp.tag_dssp_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds output data from DSSP to an Assembly.</p>
<p>A dictionary will be added to the <cite>tags</cite> dictionary of each
residue called <cite>dssp_data</cite>, which contains the secondary
structure definition, solvent accessibility phi and psi values
from DSSP. A list of regions of continuous secondary assignments
will also be added to each <cite>Polypeptide</cite>.</p>
<p>The tags are added in place, so nothing is returned from this
function.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>assembly</strong> (<em>ampal.Assembly</em>) – An Assembly containing some protein.</li>
<li><strong>loop_assignments</strong> (<em>tuple</em><em> or </em><em>list</em>) – A tuple containing the DSSP secondary structure identifiers to
that are classed as loop regions.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.geometry">
<span id="ampal-geometry-module"></span><h2>ampal.geometry module<a class="headerlink" href="#module-ampal.geometry" title="Permalink to this headline">¶</a></h2>
<p>Basic geometric operations that AMPAL requires.</p>
<p>Functions that have names preceded by an underscore indicate that they are not
to be used by users. However, all of these functions will generally be defined
using cdef and so will not be imported anyway.</p>
<dl class="class">
<dt id="ampal.geometry.Axis">
<em class="property">class </em><code class="descclassname">ampal.geometry.</code><code class="descname">Axis</code><a class="headerlink" href="#ampal.geometry.Axis" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.geometry.Curve" title="ampal.geometry.Curve"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.geometry.Curve</span></code></a></p>
<p>Axis: a straight line curve in 3-D space, defined by its start and end points (3-D vectors).</p>
<dl class="attribute">
<dt id="ampal.geometry.Axis.arc_length">
<code class="descname">arc_length</code><a class="headerlink" href="#ampal.geometry.Axis.arc_length" title="Permalink to this definition">¶</a></dt>
<dd><p>arc length between point t and point t0</p>
<p class="rubric">Notes</p>
<p>This is a trivial function in this particular case. Implemented for compatability with Curve class.
Returns arc length along the curve between the points specified by t and t_ref.
If t_ref &gt; t then the returned value is negative.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>t</strong> (<em>float</em>) – </li>
<li><strong>t_ref</strong> (<em>float</em>) – t and t_ref define points on curve (See self.point()).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.direction_vector">
<code class="descname">direction_vector</code><a class="headerlink" href="#ampal.geometry.Axis.direction_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.distance_to_point">
<code class="descname">distance_to_point</code><a class="headerlink" href="#ampal.geometry.Axis.distance_to_point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.get_coords">
<code class="descname">get_coords</code><a class="headerlink" href="#ampal.geometry.Axis.get_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>List of evenly spaced points along the curve of specified total length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>n_points</strong> (<em>int</em>) – The number of points to be returned.</li>
<li><strong>spacing</strong> (<em>float</em>) – arc-length spacing of points to be returned.</li>
<li><strong>t_ref</strong> (<em>The value of t for specifying the first point in the coordinates list.</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>coords</strong> – 3D coordinates of evenly spaced points along the curve.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list of numpy.arrays</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.length">
<code class="descname">length</code><a class="headerlink" href="#ampal.geometry.Axis.length" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.midpoint">
<code class="descname">midpoint</code><a class="headerlink" href="#ampal.geometry.Axis.midpoint" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.point">
<code class="descname">point</code><a class="headerlink" href="#ampal.geometry.Axis.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.t_from_arc_length">
<code class="descname">t_from_arc_length</code><a class="headerlink" href="#ampal.geometry.Axis.t_from_arc_length" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Returns parameter t that corresponds to a point at distance=arc_length along the curve from the curve</dt>
<dd>at t_ref.</dd>
</dl>
<p class="rubric">Notes</p>
<p>This is a trivial function in this particular case. Implemented for compatability with Curve class.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>arc_length</strong> (<em>float</em>) – distance along curve</li>
<li><strong>t_ref</strong> (<em>float</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Value of parameter t that corresponds to a point at distance=arc_length along the curve from the point
defined by parameter t_ref.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.unit_binormal">
<code class="descname">unit_binormal</code><a class="headerlink" href="#ampal.geometry.Axis.unit_binormal" title="Permalink to this definition">¶</a></dt>
<dd><p>Binormal defined as (Tangent X Normal).</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.unit_normal">
<code class="descname">unit_normal</code><a class="headerlink" href="#ampal.geometry.Axis.unit_normal" title="Permalink to this definition">¶</a></dt>
<dd><p>Normal is perpendicular to tangent. See notes.</p>
<p class="rubric">Notes</p>
<p>Since the curvature of a straight line is zero, the normal and binormal are not well-defined.
<a class="reference external" href="https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures">https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures</a>
Here we have chosen to use the following convention:</p>
<blockquote>
<div><dl class="docutils">
<dt>If the Axis is parallel to the z-axis (i.e. unit_tangent = [0, 0, 1]),</dt>
<dd>then the normal is [1, 0, 0] (x-axis) and the binormal is [0, 1, 0] (y-axis).</dd>
</dl>
<p>If the Axis is not parallel to the z-axis, then we can rotate the z-axis so it aligns with our Axis.
The normal is the same rotation appled to the x-axis.
It’s helpful here to think of the right-hand rule.
<a class="reference external" href="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/Right_hand_rule_cross_product.svg/2000px-Right_hand_rule_cross_product.svg.png">https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/Right_hand_rule_cross_product.svg/2000px-Right_hand_rule_cross_product.svg.png</a>
<a class="reference external" href="https://en.wikipedia.org/wiki/Cross_product#Definition">https://en.wikipedia.org/wiki/Cross_product#Definition</a>
Using the right-hand rule, the Axis direction vector is your index finger,
the normal vector is the middle finger and the binormal is the thumb.</p>
</div></blockquote>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Axis.unit_tangent">
<code class="descname">unit_tangent</code><a class="headerlink" href="#ampal.geometry.Axis.unit_tangent" title="Permalink to this definition">¶</a></dt>
<dd><p>Tangent points along the direction of the line.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.geometry.Curve">
<em class="property">class </em><code class="descclassname">ampal.geometry.</code><code class="descname">Curve</code><a class="headerlink" href="#ampal.geometry.Curve" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Not intended to be instantiated. This is a base class for other Curve classes to inherit from.</p>
<p class="rubric">Notes</p>
<p>Any curve must have a point method, which returns the coordinates of a point, given a parameter t.</p>
<dl class="attribute">
<dt id="ampal.geometry.Curve.point">
<code class="descname">point</code><a class="headerlink" href="#ampal.geometry.Curve.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.geometry.HelicalCurve">
<em class="property">class </em><code class="descclassname">ampal.geometry.</code><code class="descname">HelicalCurve</code><a class="headerlink" href="#ampal.geometry.HelicalCurve" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.geometry.Curve" title="ampal.geometry.Curve"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.geometry.Curve</span></code></a></p>
<p>HelicalCurve, parameterised by a and b.</p>
<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.alpha">
<code class="descname">alpha</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between tangent to helix and the helix axis, in degrees.</p>
</dd></dl>

<dl class="classmethod">
<dt id="ampal.geometry.HelicalCurve.alpha_and_pitch">
<em class="property">classmethod </em><code class="descname">alpha_and_pitch</code><span class="sig-paren">(</span><em>alpha</em>, <em>pitch</em>, <em>handedness='r'</em>, <em>t0=0</em>, <em>radians=False</em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha_and_pitch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="ampal.geometry.HelicalCurve.alpha_and_radius">
<em class="property">classmethod </em><code class="descname">alpha_and_radius</code><span class="sig-paren">(</span><em>alpha</em>, <em>radius</em>, <em>handedness='r'</em>, <em>t0=0</em>, <em>radians=False</em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha_and_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.arc_length">
<code class="descname">arc_length</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.arc_length" title="Permalink to this definition">¶</a></dt>
<dd><p>arc length between point t and point t0</p>
<p class="rubric">Notes</p>
<p>Returns arc length along the curve between the points specified by t and t_ref.
If t_ref &gt; t then the returned value is negative.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>t</strong> (<em>float</em>) – </li>
<li><strong>t_ref</strong> (<em>float</em>) – t and t_ref define points on curve (See self.point()).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.axis">
<code class="descname">axis</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.axis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.curvature">
<code class="descname">curvature</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.curvature" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.get_coords">
<code class="descname">get_coords</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.get_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>List of evenly spaced points along the curve of specified total length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>n_points</strong> (<em>int</em>) – The number of points to be returned.</li>
<li><strong>spacing</strong> (<em>float</em>) – arc-length spacing of points to be returned.</li>
<li><strong>axis_start</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array of length 3</em>) – </li>
<li><strong>axis_end</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array of length 3</em>) – Coordinates returned will be for a helical path winding around an axis specified by axis_end - axis_start.</li>
<li><strong>t_ref</strong> (<em>The value of t for specifying the first point in the coordinates list.</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>coords</strong> – 3D coordinates of evenly spaced points along the curve.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list of numpy.arrays</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.pitch">
<code class="descname">pitch</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.pitch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="ampal.geometry.HelicalCurve.pitch_and_radius">
<em class="property">classmethod </em><code class="descname">pitch_and_radius</code><span class="sig-paren">(</span><em>pitch</em>, <em>radius</em>, <em>handedness='r'</em>, <em>t0=0</em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.pitch_and_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.point">
<code class="descname">point</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.radius">
<code class="descname">radius</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.t_from_arc_length">
<code class="descname">t_from_arc_length</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.t_from_arc_length" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Returns parameter t that corresponds to a point at distance=arc_length along the curve from the curve</dt>
<dd>at t_ref.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>arc_length</strong> (<em>float</em>) – distance along curve</li>
<li><strong>t_ref</strong> (<em>float</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Value of parameter t that corresponds to a point at distance=arc_length along the curve from the point
defined by parameter t_ref.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.torsion">
<code class="descname">torsion</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.torsion" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.unit_binormal">
<code class="descname">unit_binormal</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_binormal" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.unit_normal">
<code class="descname">unit_normal</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_normal" title="Permalink to this definition">¶</a></dt>
<dd><p>Unit vector normal to the curve. Together with unit_tangent, defined the osculating plane at t</p>
<p class="rubric">Notes</p>
<p>See this article for more information on the tangent, normal and binormal vectors.
<a class="reference external" href="https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures">https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures</a></p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.unit_tangent">
<code class="descname">unit_tangent</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_tangent" title="Permalink to this definition">¶</a></dt>
<dd><p>Unit vector pointing along the direction of the curve.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.x">
<code class="descname">x</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.x" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.y">
<code class="descname">y</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.y" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.HelicalCurve.z">
<code class="descname">z</code><a class="headerlink" href="#ampal.geometry.HelicalCurve.z" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.geometry.Quaternion">
<em class="property">class </em><code class="descclassname">ampal.geometry.</code><code class="descname">Quaternion</code><a class="headerlink" href="#ampal.geometry.Quaternion" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="classmethod">
<dt id="ampal.geometry.Quaternion.angle_and_axis">
<em class="property">classmethod </em><code class="descname">angle_and_axis</code><span class="sig-paren">(</span><em>angle</em>, <em>axis</em>, <em>radians=False</em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.angle_and_axis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.apply_conjugation">
<code class="descname">apply_conjugation</code><a class="headerlink" href="#ampal.geometry.Quaternion.apply_conjugation" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.as_array">
<code class="descname">as_array</code><a class="headerlink" href="#ampal.geometry.Quaternion.as_array" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.as_rotation">
<code class="descname">as_rotation</code><a class="headerlink" href="#ampal.geometry.Quaternion.as_rotation" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.conjugate">
<code class="descname">conjugate</code><a class="headerlink" href="#ampal.geometry.Quaternion.conjugate" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.distance">
<code class="descname">distance</code><a class="headerlink" href="#ampal.geometry.Quaternion.distance" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.extract_angle_and_axis">
<code class="descname">extract_angle_and_axis</code><a class="headerlink" href="#ampal.geometry.Quaternion.extract_angle_and_axis" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the angle and axis of the rotation that the Quaternion represents.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.norm">
<code class="descname">norm</code><a class="headerlink" href="#ampal.geometry.Quaternion.norm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="ampal.geometry.Quaternion.real_and_vector">
<em class="property">classmethod </em><code class="descname">real_and_vector</code><span class="sig-paren">(</span><em>r</em>, <em>vector</em>, <em>is_rotation=False</em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.real_and_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.rotate_vector">
<code class="descname">rotate_vector</code><a class="headerlink" href="#ampal.geometry.Quaternion.rotate_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="ampal.geometry.Quaternion.vector">
<code class="descname">vector</code><a class="headerlink" href="#ampal.geometry.Quaternion.vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.geometry.angle_between_vectors">
<code class="descclassname">ampal.geometry.</code><code class="descname">angle_between_vectors</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.angle_between_vectors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate angle between two vectors using the dot product.</p>
<p class="rubric">Notes</p>
<p>All angle-based functions in tools_geometry will return values in degrees by default.
The radians flag can be used to return values in radians.
If a or b is very close to the origin (norm &lt; 1e-7), returns 0.0.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array.</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array.</em>) – </li>
<li><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, returns angle in radians.
If False, returns angle in degrees.
Default value is False.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>angle</strong> – The angle between vectors v1 and v2.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">double</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.apply_unit_quaternion">
<code class="descclassname">ampal.geometry.</code><code class="descname">apply_unit_quaternion</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.apply_unit_quaternion" title="Permalink to this definition">¶</a></dt>
<dd><p>Applies the rotation defined by the unit quaternion to a 3D vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>vector</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate.</li>
<li><strong>u_in</strong> (<em>double</em><em>[</em><em>3</em><em>]</em>) – Imaginary component of the unit quaternion.</li>
<li><strong>s</strong> (<em>double</em>) – Real component of the quaternion.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>r</strong> – 3D coordinate.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">double[3]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.cartesian_to_cylindrical">
<code class="descclassname">ampal.geometry.</code><code class="descname">cartesian_to_cylindrical</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cartesian_to_cylindrical" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cartesian to cylindrical coordinates system</p>
<p class="rubric">Notes</p>
<p>See cylindrical_to_cartesian for more information.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>x</strong> (<em>float</em>) – </li>
<li><strong>y</strong> (<em>float</em>) – </li>
<li><strong>z</strong> (<em>float</em>) – </li>
<li><strong>radians</strong> (<em>bool</em>) – True if returned azimuth is in radians instead of degrees.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>radius</strong> (<em>float</em>)</li>
<li><strong>azimuth</strong> (<em>float</em>)</li>
<li><strong>z</strong> (<em>float</em>) – Cylindrical coordinate parameters. See cylindrical_to_cartesian for more information.</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.cartesian_to_spherical">
<code class="descclassname">ampal.geometry.</code><code class="descname">cartesian_to_spherical</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cartesian_to_spherical" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cartesian to spherical coordinates system</p>
<p class="rubric">Notes</p>
<p>See spherical_to_cartesian for more information.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>x</strong> (<em>float</em>) – </li>
<li><strong>y</strong> (<em>float</em>) – </li>
<li><strong>z</strong> (<em>float</em>) – </li>
<li><strong>radians</strong> (<em>bool</em>) – True if returned azimuth and zenith are in radians instead of degrees.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>radius</strong> (<em>float</em>)</li>
<li><strong>azimuth</strong> (<em>float</em>)</li>
<li><strong>zenith</strong> (<em>float</em>) – Spherical coordinate parameters. See spherical_to_cartesian for more information.</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.centre_of_mass">
<code class="descclassname">ampal.geometry.</code><code class="descname">centre_of_mass</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.centre_of_mass" title="Permalink to this definition">¶</a></dt>
<dd><p>Find centre of mass of list of points.</p>
<p class="rubric">Notes</p>
<p>Uniform weighting applied if masses not supplied.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>points</strong> (<em>list</em><em>(</em><em>tuple</em><em>, </em><em>list</em><em> or </em><em>numpy.array</em><em>)</em>) – </li>
<li><strong>masses</strong> (<em>list</em><em>(</em><em>float</em><em>)</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>com</strong> – Centre of mass of points.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">numpy.array</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – if len(masses) != len(points)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.closest_distance">
<code class="descclassname">ampal.geometry.</code><code class="descname">closest_distance</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.closest_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Minimun distance between two lists of points in R^3.</p>
<p>Distances are calculated pairwise between points in points1 and points2.
The minimum of all these distance values is returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>points1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array of triples.</em>) – A number of points in R^3.</li>
<li><strong>points2</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array of triples.</em>) – A number of points in R^3.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>min_dist</strong> – The minimum distance between any point from points1 and any point from point2.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.closest_points_on_lines">
<code class="descclassname">ampal.geometry.</code><code class="descname">closest_points_on_lines</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.closest_points_on_lines" title="Permalink to this definition">¶</a></dt>
<dd><p class="rubric">Notes</p>
<p>p0, p1, q0, q1 describe two lines, L1 (p0 -&gt; p1) and L2 (q0 -&gt; q1).
L1 (2) is the infinite extension of the line that passes through a (q0) and has direction p0-&gt;p1 (q0-&gt;q1).
Two points point_1, and point_2 on L1 and L2 respectively, are returned.
The distance between point_1 and point_2 is smaller than any other points on L1 and L2.
Code adapted from <a class="reference external" href="http://geomalgorithms.com/a07-_distance.html">http://geomalgorithms.com/a07-_distance.html</a>, where a more detailed explanation is available.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>p1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>q0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>q1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>segments</strong> (<em>bool</em>) – If True, the points returned must lie on line segments p0-&gt;p1 and q0-&gt;q1, rather than their infinite extensions.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>point_1</strong> (<em>numpy.array</em>)</li>
<li><strong>point_2</strong> (<em>numpy.array</em>)</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.cylindrical_to_cartesian">
<code class="descclassname">ampal.geometry.</code><code class="descname">cylindrical_to_cartesian</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cylindrical_to_cartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cylindrical to cartesian coordinates system</p>
<p class="rubric">Notes</p>
<p>Cylindrical coordinates are here defined by a radius, an azimuth angle and a z value.
The azimuth is the angle in the x-y plane, with counter-clockwise direction being positive.
So: azimuth = 0 =&gt; lies on x-axis, azimuth = 90 =&gt; lies in y axis, zenith = 180 =&gt; lies antiparallel to x-axis.
The radius and the z value are the radius and height of the cylinder, respectively.
See <a class="reference external" href="https://en.wikipedia.org/wiki/Cylindrical_coordinate_system">https://en.wikipedia.org/wiki/Cylindrical_coordinate_system</a>
and/or <a class="reference external" href="http://mathworld.wolfram.com/CylindricalCoordinates.html">http://mathworld.wolfram.com/CylindricalCoordinates.html</a>
for more information.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>radius</strong> (<em>float</em>) – </li>
<li><strong>azimuth</strong> (<em>float</em>) – </li>
<li><strong>z</strong> (<em>float</em>) – </li>
<li><strong>radians</strong> (<em>bool</em>) – True if azimuth is given in radians instead of degrees.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">cartesian coordinates corresponding to the input cylindrical coordinates.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">numpy.array([x, y, z])</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.dihedral">
<code class="descclassname">ampal.geometry.</code><code class="descname">dihedral</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.dihedral" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the dihedral angle defined by four points in R^3.</p>
<p>The points a, b and c form a plane in R^3.
The points b, c and d form a plane in R^3.
The dihedral angle is the angle between these two planes.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://en.wikipedia.org/wiki/Dihedral_angle">http://en.wikipedia.org/wiki/Dihedral_angle</a> and <a class="reference external" href="https://en.wikipedia.org/wiki/Atan2">https://en.wikipedia.org/wiki/Atan2</a> for more information.
Dihedral between 4 points.
a, b, c, d are all 3-vectors (x,y,z) coordinates.
(a,b,c) and (b,c,d) must both be non-colinear.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>c</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>d</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, returns angle in radians.
If False, returns angle in degrees.
Default value is False.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>angle</strong> – The dihedral angle formed between the points a, b, c and d.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">double</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.distance">
<code class="descclassname">ampal.geometry.</code><code class="descname">distance</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the Euclidian distance between two points in R^3.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>dist</strong> – Euclidean distance between a and b.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">double</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.find_foot">
<code class="descclassname">ampal.geometry.</code><code class="descname">find_foot</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_foot" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the perpendicular foot of a point (p) and a line (ab).</p>
<p>The points a and b define a line in R^3.
There is a point on this line, q, that has the shortest Euclidian distance from the point p.
This point is the perpendicular foot, and is returned by this function.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html">http://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html</a> for a useful work-through of the maths.
Parameterise the line ab by t, so that a point on the line is of the form: a + (b-a)*t.
Define f(t) = [distance(t, p)]^2. Want t such that f(t) is at a minimum.
Solve f’(t) = 0 for t.
Substituting this value of t into a + (b-a)*t yields the desired point q.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Together, the points a and b define a line in R^3.</li>
<li><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – A point in R^3.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>q</strong> – The point on the line between a and b that is closest to q.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list or tuple or numpy.array</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.find_foot_on_plane">
<code class="descclassname">ampal.geometry.</code><code class="descname">find_foot_on_plane</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_foot_on_plane" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the perpendicular foot of a point (p) and a plane defined by points a, b and c.</p>
<p>The points a, b and c must define a plane in R^3 (i.e. they cannot be collinear).
There is a point on this plane, q, that has the shortest Euclidian distance from the point p.
This point is the perpendicular foot, and is returned by this function.</p>
<p class="rubric">Notes</p>
<dl class="docutils">
<dt>This page:</dt>
<dd><a class="reference external" href="http://math.stackexchange.com/questions/723937/find-the-point-on-a-plane-3x-4y-z-1-that-is-closest-to-1-0-1">http://math.stackexchange.com/questions/723937/find-the-point-on-a-plane-3x-4y-z-1-that-is-closest-to-1-0-1</a>
gives a work-through of this kind of problem for a specific example.</dd>
</dl>
<p>The solution uses the Lagrangian multiplier: <a class="reference external" href="https://en.wikipedia.org/wiki/Lagrange_multiplier">https://en.wikipedia.org/wiki/Lagrange_multiplier</a>
We look for the minimum values of the distance function f (squared to make it easier!)</p>
<blockquote>
<div>f(x, y, z) = (x - p[0])^2 + (y - p[1])^2 + (z - p[2])^2</div></blockquote>
<p>subject to the constraint g(x, y, z) = 0, where g is the equation of the plane.
The Langrangian multiplier L = f(x, y, z) - lambda*(g(x, y, z)).
We calculate the partial deriviatves of this (wrt x, y, z and lambda).
Solve first for lambda gives the equation for the lagrangian_lambda value in this function.
Then substitute lambda back in to get the solution for (x, y, z) = q.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>c</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Together, the points a, b and c define a plane in R^3.</li>
<li><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – A point in R^3.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>s</strong> – The point on the plane that is closest to q.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">numpy.array</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.find_limits">
<code class="descclassname">ampal.geometry.</code><code class="descname">find_limits</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_limits" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="ampal.geometry.find_transformations">
<code class="descclassname">ampal.geometry.</code><code class="descname">find_transformations</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_transformations" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns geometrical transformations required to align the vector from s1 -&gt; e1 with the vector from s2 -&gt; e2</p>
<p class="rubric">Notes</p>
<p>Applying the transformations returned to the line from s1 -&gt; e2, will align it with the line from s2 -&gt; e2.
After transformation, s1 will have been moved onto s2.
Apply the rotation (given by angle, axis and point) first and then the translation (given by translation).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>s1</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. Start of line 1</li>
<li><strong>e1</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. End of line 1</li>
<li><strong>s2</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. Start of line 2</li>
<li><strong>e2</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. End of line 2</li>
<li><strong>radians</strong> (<em>bool</em>) – If True, angle will be returned in radians.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>translation</strong> (<em>numpy.array(3)</em>) – 3D vector. Translation to be applied to s1 to move it onto s2.</li>
<li><strong>angle</strong> (<em>float</em>) – angle for rotation (default: degrees).</li>
<li><strong>axis</strong> (<em>numpy.array(3)</em>) – axis about which to rotate line 1 to align it with line 2.</li>
<li><strong>point</strong> (<em>numpy.array(3)</em>) – a point lying on the rotation axis.</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.gen_sectors">
<code class="descclassname">ampal.geometry.</code><code class="descname">gen_sectors</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.gen_sectors" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates the coordinates into overlapping sectors.</p>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.intersection_start_and_end">
<code class="descclassname">ampal.geometry.</code><code class="descname">intersection_start_and_end</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.intersection_start_and_end" title="Permalink to this definition">¶</a></dt>
<dd><p>Find start and end indices of the part of the list of points ‘subject’ that will intersect with ‘reference’</p>
<p class="rubric">Notes</p>
<p>This is a function designed to find assemblies from within bundles of helices.
The idea here is to take two lists of points, a subject and a reference. We want to find the part of the subject
(this could be all of it, or none of it) that would intersect the line joining the points in the reference, if
the line joining the points of subject was the prinicipal axis of a cylinder of infinite radius.
If subject was a list of ten points joining (0, 0 , 0) and (0, 0, 10), and reference was a list of 1000 points
joining (10, 0, 7) and (10, 0, 12) this function would return (7, 10).
If the subject and reference for the above example were swapped around, however, the function would return (0, 599).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>subject</strong> (<em>list of numpy.array</em>) – A list of points in R^3.</li>
<li><strong>reference</strong> (<em>list of numpy.array</em>) – A list of points in R^3.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><ul class="simple">
<li><strong>start_index</strong> (<em>int</em>) – The index of the first point in subject that is part of the intersection described above.</li>
<li><strong>end_index</strong> (<em>int</em>) – The index of the final point in subject that is part of the intersection described above.</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.is_acute">
<code class="descclassname">ampal.geometry.</code><code class="descname">is_acute</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.is_acute" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if angle between two vectors is less than 90 degrees.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">True if angle between v1 and v2 is less than 90 degrees.
False otherwise.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">bool</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.minimal_distance_between_lines">
<code class="descclassname">ampal.geometry.</code><code class="descname">minimal_distance_between_lines</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.minimal_distance_between_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the minimal distance between two the two lines joining p0 to p1 and q0 to q1.</p>
<p>The points p0, p1, q0 and d are used to construct two lines in R^3.
The first line L1 is from p0 to p1, the second line L2 is from q0 to q1.
The minimal distance between these lines is the length of the line that
connects them and is perpendicular to both L1 and L2.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="https://en.wikipedia.org/wiki/Skew_lines#Distance">https://en.wikipedia.org/wiki/Skew_lines#Distance</a> for more information.
Also <a class="reference external" href="http://geomalgorithms.com/a07-_distance.html">http://geomalgorithms.com/a07-_distance.html</a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>p1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>q0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>q1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>segments</strong> (<em>bool</em>) – If True, the points returned must lie on line segments p0-&gt;p1 and q0-&gt;q1, rather than their infinite extensions.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>dist</strong> – The distance between the two lines.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.points_on_a_circle">
<code class="descclassname">ampal.geometry.</code><code class="descname">points_on_a_circle</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.points_on_a_circle" title="Permalink to this definition">¶</a></dt>
<dd><p>List of n uniformly distributed (x, y) coordinates on the circumference of a circle.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>n</strong> (<em>int</em>) – Number of points to return.</li>
<li><strong>radius</strong> (<em>float</em>) – </li>
<li><strong>centre</strong> (<em>tuple</em><em> or </em><em>list</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>rotation</strong> (<em>float</em>) – Angle in degrees by which all points will be rotated.
rotation = 0 means that the line from the centre to the first point is parallel to the x-axis.
rotation &gt; 0 =&gt; anti-clockwise rotation</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>points</strong> – (x, y) coordinates of uniformly distributed points on the circumference of the circle</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">list(2-tuples)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.radius_of_circumcircle">
<code class="descclassname">ampal.geometry.</code><code class="descname">radius_of_circumcircle</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.radius_of_circumcircle" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the radius the circumcircle of a triangle, when supplied with three points (p, q, r).</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://www.analyzemath.com/Geometry/Circumcircle/CircumcircleRadius.html">http://www.analyzemath.com/Geometry/Circumcircle/CircumcircleRadius.html</a>
for more information.
Three (non-colinear) points define a traingle.
The circumcircle is the circle that passes though these points.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>q</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
<li><strong>r</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>radius</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.rmsd">
<code class="descclassname">ampal.geometry.</code><code class="descname">rmsd</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.rmsd" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns average distance between points in points1 and their counterparts in points2.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>points1</strong> (<em>list of</em><em> [</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – </li>
<li><strong>points2</strong> (<em>list of</em><em> [</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – </li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>rmsd</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first">float</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last">ValueError : if len(points1) != len(points2)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.rotation_matrix">
<code class="descclassname">ampal.geometry.</code><code class="descname">rotation_matrix</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.rotation_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Find rotation matrix for rotating about unit_vect by angle.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="https://en.wikipedia.org/wiki/Rotation_matrix">https://en.wikipedia.org/wiki/Rotation_matrix</a> for more information.
In particular, the section headed ‘Rotation matrix from axis and angle’.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>angle</strong> (<em>float</em>) – Angle about which to rotate.</li>
<li><strong>unit_vect</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Vector about which to rotate.</li>
<li><strong>point</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em>, </em><em>optional.</em>) – Specify a point about which to rotate.
If point is None, then the rotation will be about the origin.
Default value is None.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>m</strong> – A (4 x 4) rotation matrix.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">numpy.array</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.spherical_to_cartesian">
<code class="descclassname">ampal.geometry.</code><code class="descname">spherical_to_cartesian</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.spherical_to_cartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from spherical to cartesian coordinates system</p>
<p class="rubric">Notes</p>
<p>Spherical coordinates are here defined by a radius, an azimuth angle and a zenith angle.
The zenith is the angle between the point and the z-axis.
So: zenith = 0 =&gt; lies on z-axis, zenith = 90 =&gt; lies in x-y plane, zenith = 180 =&gt; lies antiparallel to z-axis.
The azimuth is the angle in the x-y plane, with counter-clockwise direction being positive.
So: azimuth = 0 =&gt; lies on x-axis, azimuth = 90 =&gt; lies in y axis, zenith = 180 =&gt; lies antiparallel to x-axis.
See <a class="reference external" href="https://en.wikipedia.org/wiki/Spherical_coordinate_system">https://en.wikipedia.org/wiki/Spherical_coordinate_system</a>
and/or <a class="reference external" href="http://mathworld.wolfram.com/SphericalCoordinates.html">http://mathworld.wolfram.com/SphericalCoordinates.html</a>
for more information.
If zenith = 0, then x and y are both 0 regardless of azimuth value.
If azimuth = 0, then y is 0 regardless of zenith value.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>radius</strong> (<em>float</em>) – </li>
<li><strong>azimuth</strong> (<em>float</em>) – </li>
<li><strong>zenith</strong> (<em>float</em>) – </li>
<li><strong>radians</strong> (<em>bool</em>) – True if azimuth and zenith are given in radians instead of degrees.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">cartesian coordinates corresponding to the input spherical coordinates.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">numpy.array([x, y, z])</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.geometry.unit_vector">
<code class="descclassname">ampal.geometry.</code><code class="descname">unit_vector</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.unit_vector" title="Permalink to this definition">¶</a></dt>
<dd><p>Return vector normalised by length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – The vector to be normalised.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>v_out</strong> – A numpy.array that is a unit vector.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">numpy.array</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ZeroDivisionError</span></code> – If the vector has length 0.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.interactions">
<span id="ampal-interactions-module"></span><h2>ampal.interactions module<a class="headerlink" href="#module-ampal.interactions" title="Permalink to this headline">¶</a></h2>
<p>Contains code for analysing chemical interactions in AMPAL objects.</p>
<dl class="class">
<dt id="ampal.interactions.CovalentBond">
<em class="property">class </em><code class="descclassname">ampal.interactions.</code><code class="descname">CovalentBond</code><span class="sig-paren">(</span><em>a</em>, <em>b</em>, <em>dist</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#CovalentBond"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.CovalentBond" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.Interaction" title="ampal.interactions.Interaction"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.interactions.Interaction</span></code></a></p>
<p>Defines a covalent bond.</p>
<dl class="attribute">
<dt id="ampal.interactions.CovalentBond.a">
<code class="descname">a</code><a class="headerlink" href="#ampal.interactions.CovalentBond.a" title="Permalink to this definition">¶</a></dt>
<dd><p>One <cite>Atom</cite> involved in the covalent bond.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.CovalentBond.b">
<code class="descname">b</code><a class="headerlink" href="#ampal.interactions.CovalentBond.b" title="Permalink to this definition">¶</a></dt>
<dd><p>One <cite>Atom</cite> involved in the covalent bond.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.interactions.HydrogenBond">
<em class="property">class </em><code class="descclassname">ampal.interactions.</code><code class="descname">HydrogenBond</code><span class="sig-paren">(</span><em>donor</em>, <em>acceptor</em>, <em>dist</em>, <em>ang_d</em>, <em>ang_a</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#HydrogenBond"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.HydrogenBond" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.NonCovalentInteraction" title="ampal.interactions.NonCovalentInteraction"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.interactions.NonCovalentInteraction</span></code></a></p>
<p>Defines a hydrogen bond in terms of a donor and an acceptor.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>donor</strong> (<em>ampal.Atom</em>) – The donor <cite>Atom</cite> in the interaction.</li>
<li><strong>acceptor</strong> (<em>ampal.Atom</em>) – The acceptor atom in the interaction.</li>
<li><strong>dist</strong> (<em>float</em>) – The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</li>
<li><strong>ang_a</strong> (<em>float</em>) – Angle between the acceptor and the interaction vector.</li>
<li><strong>ang_d</strong> (<em>float</em>) – Angle between the donor and the interaction vector.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.donor">
<code class="descname">donor</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.donor" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.acceptor">
<code class="descname">acceptor</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.acceptor" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor atom in the interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.dist">
<code class="descname">dist</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.ang_a">
<code class="descname">ang_a</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.ang_a" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between the acceptor and the interaction vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.ang_d">
<code class="descname">ang_d</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.ang_d" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between the donor and the interaction vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.acceptor_monomer">
<code class="descname">acceptor_monomer</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.acceptor_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor <cite>Monomer</cite> in the interaction.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.HydrogenBond.donor_monomer">
<code class="descname">donor_monomer</code><a class="headerlink" href="#ampal.interactions.HydrogenBond.donor_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Monomer</cite> in the interaction.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.interactions.Interaction">
<em class="property">class </em><code class="descclassname">ampal.interactions.</code><code class="descname">Interaction</code><span class="sig-paren">(</span><em>a</em>, <em>b</em>, <em>dist</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#Interaction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.Interaction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A container for all types of interaction with donor and acceptor.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>a</strong> (<em>ampal.Atom</em>) – A member of a pairwise interaction.</li>
<li><strong>b</strong> (<em>ampal.Atom</em>) – A member of a pairwise interaction.</li>
<li><strong>dist</strong> (<em>float</em>) – The distance between <cite>a</cite> and <cite>b</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.interactions.Interaction.a">
<code class="descname">a</code><a class="headerlink" href="#ampal.interactions.Interaction.a" title="Permalink to this definition">¶</a></dt>
<dd><p>A member of a pairwise interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.Interaction.b">
<code class="descname">b</code><a class="headerlink" href="#ampal.interactions.Interaction.b" title="Permalink to this definition">¶</a></dt>
<dd><p>A member of a pairwise interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.Interaction.dist">
<code class="descname">dist</code><a class="headerlink" href="#ampal.interactions.Interaction.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.interactions.NonCovalentInteraction">
<em class="property">class </em><code class="descclassname">ampal.interactions.</code><code class="descname">NonCovalentInteraction</code><span class="sig-paren">(</span><em>donor</em>, <em>acceptor</em>, <em>dist</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#NonCovalentInteraction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.Interaction" title="ampal.interactions.Interaction"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.interactions.Interaction</span></code></a></p>
<p>A container for all non-covalent interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>donor</strong> (<em>ampal.Atom</em>) – The donor <cite>Atom</cite> in the interaction.</li>
<li><strong>acceptor</strong> (<em>ampal.Atom</em>) – The acceptor atom in the interaction.</li>
<li><strong>dist</strong> (<em>float</em>) – The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.interactions.NonCovalentInteraction.donor">
<code class="descname">donor</code><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.donor" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.NonCovalentInteraction.acceptor">
<code class="descname">acceptor</code><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.acceptor" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor atom in the interaction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Atom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.interactions.NonCovalentInteraction.dist">
<code class="descname">dist</code><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">acceptor</code></dt>
<dd><p>The acceptor in the interaction.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">donor</code></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.interactions.covalent_bonds">
<code class="descclassname">ampal.interactions.</code><code class="descname">covalent_bonds</code><span class="sig-paren">(</span><em>atoms</em>, <em>threshold=1.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#covalent_bonds"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.covalent_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all the covalent bonds in a list of <cite>Atom</cite> pairs.</p>
<p class="rubric">Notes</p>
<p>Uses information <cite>ELEMENT_DATA</cite>, which can be accessed directly
through this module i.e. <cite>isambard.ampal.interactions.ELEMENT_DATA</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atoms</strong> ([(<cite>Atom</cite>, <cite>Atom</cite>)]) – List of pairs of <cite>Atoms</cite>.</li>
<li><strong>threshold</strong> (<em>float</em><em>, </em><em>optional</em>) – Allows deviation from ideal covalent bond distance to be included.
For example, a value of 1.1 would allow interactions up to 10% further
from the ideal distance to be included.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.interactions.find_covalent_bonds">
<code class="descclassname">ampal.interactions.</code><code class="descname">find_covalent_bonds</code><span class="sig-paren">(</span><em>ampal</em>, <em>max_range=2.2</em>, <em>threshold=1.1</em>, <em>tag=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#find_covalent_bonds"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.find_covalent_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds all covalent bonds in the AMPAL object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>ampal</strong> (<em>AMPAL Object</em>) – Any AMPAL object with a <cite>get_atoms</cite> method.</li>
<li><strong>max_range</strong> (<em>float</em><em>, </em><em>optional</em>) – Used to define the sector size, so interactions at longer ranges
will not be found.</li>
<li><strong>threshold</strong> (<em>float</em><em>, </em><em>optional</em>) – Allows deviation from ideal covalent bond distance to be included.
For example, a value of 1.1 would allow interactions up to 10% further
from the ideal distance to be included.</li>
<li><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will add the covalent bond to the tags dictionary of
each <cite>Atom</cite> involved in the interaction under the <cite>covalent_bonds</cite>
key.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.interactions.generate_bond_subgraphs_from_break">
<code class="descclassname">ampal.interactions.</code><code class="descname">generate_bond_subgraphs_from_break</code><span class="sig-paren">(</span><em>bond_graph</em>, <em>atom1</em>, <em>atom2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#generate_bond_subgraphs_from_break"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.generate_bond_subgraphs_from_break" title="Permalink to this definition">¶</a></dt>
<dd><p>Splits the bond graph between two atoms to producing subgraphs.</p>
<p class="rubric">Notes</p>
<p>This will not work if there are cycles in the bond graph.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>bond_graph</strong> (<em>networkx.Graph</em>) – Graph of covalent bond network</li>
<li><strong>atom1</strong> (<em>isambard.ampal.Atom</em>) – First atom in the bond.</li>
<li><strong>atom2</strong> (<em>isambard.ampal.Atom</em>) – Second atom in the bond.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>subgraphs</strong> – A list of subgraphs generated when a bond is broken in the covalent
bond network.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">[networkx.Graph]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.interactions.generate_covalent_bond_graph">
<code class="descclassname">ampal.interactions.</code><code class="descname">generate_covalent_bond_graph</code><span class="sig-paren">(</span><em>covalent_bonds</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#generate_covalent_bond_graph"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.interactions.generate_covalent_bond_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a graph of the covalent bond network described by the interactions.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>covalent_bonds</strong> (<em>[</em><a class="reference internal" href="#ampal.interactions.CovalentBond" title="ampal.interactions.CovalentBond"><em>CovalentBond</em></a><em>]</em>) – List of <cite>CovalentBond</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>bond_graph</strong> – A graph of the covalent bond network.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">networkx.Graph</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.ligands">
<span id="ampal-ligands-module"></span><h2>ampal.ligands module<a class="headerlink" href="#module-ampal.ligands" title="Permalink to this headline">¶</a></h2>
<p>AMPAL objects that represent ligands.</p>
<dl class="class">
<dt id="ampal.ligands.Ligand">
<em class="property">class </em><code class="descclassname">ampal.ligands.</code><code class="descname">Ligand</code><span class="sig-paren">(</span><em>mol_code</em>, <em>atoms=None</em>, <em>monomer_id=' '</em>, <em>insertion_code=' '</em>, <em>is_hetero=False</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ligands.html#Ligand"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ligands.Ligand" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Monomer</span></code></a></p>
<p><cite>Monomer</cite> that represents a <cite>Ligand</cite>.</p>
<p class="rubric">Notes</p>
<p>All <cite>Monomers</cite> that do not have dedicated classes are
represented using the <cite>Ligand</cite> class.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>mol_code</strong> (<em>str</em>) – PDB molecule code that represents the monomer.</li>
<li><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</li>
<li><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of monomer, used if reading from pdb.</li>
<li><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.ligands.Ligand.atoms">
<code class="descname">atoms</code><a class="headerlink" href="#ampal.ligands.Ligand.atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">OrderedDict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.mol_code">
<code class="descname">mol_code</code><a class="headerlink" href="#ampal.ligands.Ligand.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Ligand</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.insertion_code">
<code class="descname">insertion_code</code><a class="headerlink" href="#ampal.ligands.Ligand.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Ligand</cite>, used if reading from pdb.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.is_hetero">
<code class="descname">is_hetero</code><a class="headerlink" href="#ampal.ligands.Ligand.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.self.states">
<code class="descclassname">self.</code><code class="descname">states</code><a class="headerlink" href="#ampal.ligands.Ligand.self.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Ligand</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.ligands.Ligand.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.ligands.Ligand.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>LigandGroup</cite> containing this <cite>Ligand</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.ligands.Ligand.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.ligands.Ligand.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.ligands.LigandGroup">
<em class="property">class </em><code class="descclassname">ampal.ligands.</code><code class="descname">LigandGroup</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>polymer_id=' '</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ligands.html#LigandGroup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.ligands.LigandGroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Polymer</span></code></a></p>
<p>A container for <cite>Ligand</cite> <cite>Monomers</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomers</strong> (<a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em>]</em><em>, </em><em>optional</em>) – Monomer or list containing Monomer objects to form the Polymer().</li>
<li><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polymer</cite>. This is
used when generating a pdb file using <cite>Polymer().pdb</cite>.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</li>
<li><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.nucleic_acid">
<span id="ampal-nucleic-acid-module"></span><h2>ampal.nucleic_acid module<a class="headerlink" href="#module-ampal.nucleic_acid" title="Permalink to this headline">¶</a></h2>
<p>Contains AMPAL objects representing nucleic acids.</p>
<dl class="class">
<dt id="ampal.nucleic_acid.Nucleotide">
<em class="property">class </em><code class="descclassname">ampal.nucleic_acid.</code><code class="descname">Nucleotide</code><span class="sig-paren">(</span><em>atoms=None</em>, <em>mol_code='UNK'</em>, <em>monomer_id=' '</em>, <em>insertion_code=' '</em>, <em>is_hetero=False</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/nucleic_acid.html#Nucleotide"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Monomer</span></code></a></p>
<p>Represents a nucleotide base.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>Nucleotide</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</li>
<li><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>Nucleotide</cite>.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>Nucleotide</cite>.</li>
<li><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</li>
<li><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</li>
<li><strong>parent</strong> (<em>ampal.Polynucleotide</em><em>, </em><em>optional</em>) – Reference to <cite>Polynucleotide</cite> containing the <cite>Nucleotide</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.mol_code">
<code class="descname">mol_code</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.insertion_code">
<code class="descname">insertion_code</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.is_hetero">
<code class="descname">is_hetero</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.states">
<code class="descname">states</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.reference_atom">
<code class="descname">reference_atom</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Nucleotide.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if <cite>mol_code</cite> is not length 1 or 3.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="ampal.nucleic_acid.Polynucleotide">
<em class="property">class </em><code class="descclassname">ampal.nucleic_acid.</code><code class="descname">Polynucleotide</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>polymer_id=' '</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/nucleic_acid.html#Polynucleotide"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Polymer</span></code></a></p>
<p><cite>Polymer</cite> type object that represents a <cite>Polynucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomers</strong> (<a class="reference internal" href="#ampal.nucleic_acid.Nucleotide" title="ampal.nucleic_acid.Nucleotide"><em>Nucleotide</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.nucleic_acid.Nucleotide" title="ampal.nucleic_acid.Nucleotide"><em>Nucleotide</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Nucleotide</cite> or list containing <cite>Nucleotide</cite> objects to form the
<cite>Polynucleotide</cite>.</li>
<li><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polynucleotide</cite>. This is
used when generating a pdb file using <cite>Polynucleotide().pdb</cite>.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polynucleotide</cite>.</li>
<li><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>Polynucleotide</cite> ID</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polynucleotide</cite></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Assembly or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.molecule_type">
<code class="descname">molecule_type</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.ligands">
<code class="descname">ligands</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polynucleotide</cite> chain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.LigandGroup</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.sl">
<code class="descname">sl</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – <cite>Polymer</cite> type objects can only be initialised empty or using
a <cite>Monomer</cite>.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.nucleic_acid.Polynucleotide.sequence">
<code class="descname">sequence</code><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of the <cite>Polynucleotide</cite> as a string.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>sequence</strong> – String of the monomer sequence of the <cite>Polynucleotide</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-ampal.pdb_parser">
<span id="ampal-pdb-parser-module"></span><h2>ampal.pdb_parser module<a class="headerlink" href="#module-ampal.pdb_parser" title="Permalink to this headline">¶</a></h2>
<p>Contains code for parsing PDB files.</p>
<dl class="class">
<dt id="ampal.pdb_parser.PdbParser">
<em class="property">class </em><code class="descclassname">ampal.pdb_parser.</code><code class="descname">PdbParser</code><span class="sig-paren">(</span><em>pdb</em>, <em>path=True</em>, <em>pdb_id=''</em>, <em>ignore_end=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Parses a PDB file and produces and AMPAL Assembly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>pdb</strong> (<em>str</em>) – Either a path to a PDB file or a string containing PDB
format structural data.</li>
<li><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, flags <cite>pdb</cite> as a path and not a PDB string.</li>
<li><strong>pdb_id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the <cite>Assembly</cite>.</li>
<li><strong>ignore_end</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.proc_functions">
<code class="descname">proc_functions</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_functions" title="Permalink to this definition">¶</a></dt>
<dd><p>Keys are PDB labels i.e. “ATOM” or “END”, values are the
function that processes that specific line.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for the <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.pdb_lines">
<code class="descname">pdb_lines</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.pdb_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Input PDB split into line.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">[str]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.new_labels">
<code class="descname">new_labels</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.new_labels" title="Permalink to this definition">¶</a></dt>
<dd><p>Indicates if new atom or residue labels have been found while
parsing the PDB file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.state">
<code class="descname">state</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.state" title="Permalink to this definition">¶</a></dt>
<dd><p>Current state being appended to. This is used on multi-state
files like NMR structures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.pdb_parse_tree">
<code class="descname">pdb_parse_tree</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.pdb_parse_tree" title="Permalink to this definition">¶</a></dt>
<dd><p>This dictionary represents the parse tree of the PDB file.
Each line of the structure is broken down into a key, the
entry label, and a value, the data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.current_line">
<code class="descname">current_line</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.current_line" title="Permalink to this definition">¶</a></dt>
<dd><p>The line that is currently being parsed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pdb_parser.PdbParser.ignore_end">
<code class="descname">ignore_end</code><a class="headerlink" href="#ampal.pdb_parser.PdbParser.ignore_end" title="Permalink to this definition">¶</a></dt>
<dd><p>If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool, optional</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.change_state">
<code class="descname">change_state</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.change_state"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.change_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Increments current state and adds a new dict to the parse tree.</p>
</dd></dl>

<dl class="staticmethod">
<dt id="ampal.pdb_parser.PdbParser.check_for_non_canonical">
<em class="property">static </em><code class="descname">check_for_non_canonical</code><span class="sig-paren">(</span><em>residue</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.check_for_non_canonical"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.check_for_non_canonical" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks to see if the residue is non-canonical.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.end">
<code class="descname">end</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.end"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.end" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes an “END” record.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.gen_states">
<code class="descname">gen_states</code><span class="sig-paren">(</span><em>monomer_data</em>, <em>parent</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.gen_states"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.gen_states" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates the <cite>states</cite> dictionary for a <cite>Monomer</cite>.</p>
<dl class="docutils">
<dt>monomer_data <span class="classifier-delimiter">:</span> <span class="classifier">list</span></dt>
<dd>A list of atom data parsed from the input PDB.</dd>
<dt>parent <span class="classifier-delimiter">:</span> <span class="classifier">ampal.Monomer</span></dt>
<dd><cite>Monomer</cite> used to assign <cite>parent</cite> on created
<cite>Atoms</cite>.</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.make_ampal">
<code class="descname">make_ampal</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.make_ampal"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.make_ampal" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates an AMPAL object from the parse tree.</p>
<p class="rubric">Notes</p>
<p>Will create an <cite>Assembly</cite> if there is a single state in the
parese tree or an <cite>AmpalContainer</cite> if there is more than one.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.parse_pdb_file">
<code class="descname">parse_pdb_file</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.parse_pdb_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.parse_pdb_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs the PDB parser.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.proc_atom">
<code class="descname">proc_atom</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_atom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes an “ATOM” or “HETATM” record.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.proc_chain">
<code class="descname">proc_chain</code><span class="sig-paren">(</span><em>chain_info</em>, <em>parent</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_chain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a chain into a <cite>Polymer</cite> type object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>chain_info</strong> (<em>(</em><em>set</em><em>, </em><em>OrderedDict</em><em>)</em>) – Contains a set of chain labels and atom records.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em>) – <cite>Assembly</cite> used to assign <cite>parent</cite> on created
<cite>Polymer</cite>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><ul class="first last simple">
<li><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if multiple or unknown atom types found
within the same chain.</li>
<li><code class="xref py py-exc docutils literal notranslate"><span class="pre">AttributeError</span></code> – Raised if unknown <cite>Monomer</cite> type encountered.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.proc_line_coordinate">
<code class="descname">proc_line_coordinate</code><span class="sig-paren">(</span><em>line</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_line_coordinate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_line_coordinate" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts data from columns in ATOM/HETATM record.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.proc_monomer">
<code class="descname">proc_monomer</code><span class="sig-paren">(</span><em>monomer_info</em>, <em>parent</em>, <em>mon_cls=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_monomer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes a records into a <cite>Monomer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomer_info</strong> (<em>(</em><em>set</em><em>, </em><em>OrderedDict</em><em>)</em>) – Labels and data for a monomer.</li>
<li><strong>parent</strong> (<em>ampal.Polymer</em>) – <cite>Polymer</cite> used to assign <cite>parent</cite> on created
<cite>Monomer</cite>.</li>
<li><strong>mon_cls</strong> (<cite>Monomer class or subclass</cite>, optional) – A <cite>Monomer</cite> class can be defined explicitly.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.pdb_parser.PdbParser.proc_state">
<code class="descname">proc_state</code><span class="sig-paren">(</span><em>state_data</em>, <em>state_id</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_state"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes a state into an <cite>Assembly</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>state_data</strong> (<em>dict</em>) – Contains information about the state, including all
the per line structural data.</li>
<li><strong>state_id</strong> (<em>str</em>) – ID given to <cite>Assembly</cite> that represents the state.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.pdb_parser.load_pdb">
<code class="descclassname">ampal.pdb_parser.</code><code class="descname">load_pdb</code><span class="sig-paren">(</span><em>pdb</em>, <em>path=True</em>, <em>pdb_id=''</em>, <em>ignore_end=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#load_pdb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pdb_parser.load_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a PDB file into an AMPAL object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pdb</strong> (<em>str</em>) – Either a path to a PDB file or a string containing PDB
format structural data.</li>
<li><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, flags <cite>pdb</cite> as a path and not a PDB string.</li>
<li><strong>pdb_id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the <cite>Assembly</cite>.</li>
<li><strong>ignore_end</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>ampal</strong> – AMPAL object that contains the structural information from
the PDB file provided. If the PDB file has a single state
then an <cite>Assembly</cite> will be returned, otherwise an
<cite>AmpalContainer</cite> will be returned.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">ampal.Assembly or ampal.AmpalContainer</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.protein">
<span id="ampal-protein-module"></span><h2>ampal.protein module<a class="headerlink" href="#module-ampal.protein" title="Permalink to this headline">¶</a></h2>
<p>AMPAL objects that represent protein.</p>
<dl class="class">
<dt id="ampal.protein.Polypeptide">
<em class="property">class </em><code class="descclassname">ampal.protein.</code><code class="descname">Polypeptide</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>polymer_id=' '</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Polymer</span></code></a></p>
<p>Container for <cite>Residues</cite>, inherits from <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomers</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Residue</cite> or list containing <cite>Residue</cite> objects to form the
<cite>Polypeptide</cite>.</li>
<li><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polypeptide</cite>. This is
used when generating a pdb file using <cite>Polypeptide().pdb</cite>.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</li>
<li><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.protein.Polypeptide.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.protein.Polypeptide.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>Polypeptide</cite> ID</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.protein.Polypeptide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polypeptide</cite></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Assembly or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.molecule_type">
<code class="descname">molecule_type</code><a class="headerlink" href="#ampal.protein.Polypeptide.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.ligands">
<code class="descname">ligands</code><a class="headerlink" href="#ampal.protein.Polypeptide.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polypeptide</cite> chain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.LigandGroup</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.protein.Polypeptide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.sl">
<code class="descname">sl</code><a class="headerlink" href="#ampal.protein.Polypeptide.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – <cite>Polymer</cite> type objects can only be initialised empty or using
a <cite>Monomer</cite>.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.protein.Polypeptide.backbone">
<code class="descname">backbone</code><a class="headerlink" href="#ampal.protein.Polypeptide.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Polymer</cite> containing only the backbone atoms.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.
Sequence data is retained, but only the main chain atoms are retained.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>bb_poly</strong> – Polymer containing only the backbone atoms of the original
Polymer.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.backbone_bond_angles">
<code class="descname">backbone_bond_angles</code><a class="headerlink" href="#ampal.protein.Polypeptide.backbone_bond_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary containing backbone bond angles as lists of floats.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>bond_angles</strong> – Keys are <cite>n_ca_c</cite>, <cite>ca_c_o</cite>, <cite>ca_c_n</cite> and <cite>c_n_ca</cite>, referring
to the N-CA-C, CA-C=O, CA-C-N and C-N-CA angles respectively.
Values are lists of floats : the bond angles in degrees.
The lists of n_ca_c, ca_c_o are of length k for a <cite>Polypeptide</cite>
containing k <cite>Residues</cite>. The list of ca_c_n and c_n_ca are of
length k-1 for a <cite>Polypeptide</cite> containing k <cite>Residues</cite> (These
angles are across the peptide bond, and are therefore formed
between successive <cite>Residue</cite> pairs).</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.backbone_bond_lengths">
<code class="descname">backbone_bond_lengths</code><a class="headerlink" href="#ampal.protein.Polypeptide.backbone_bond_lengths" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary containing backbone bond lengths as lists of floats.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>bond_lengths</strong> – Keys are <cite>n_ca</cite>, <cite>ca_c</cite>, <cite>c_o</cite> and <cite>c_n</cite>, referring to the
N-CA, CA-C, C=O and C-N bonds respectively. Values are
lists of floats : the bond lengths in Angstroms.
The lists of n_ca, ca_c and c_o are of length k for
a Polypeptide containing k Residues. The list of c_n bonds
is of length k-1 for a Polypeptide containing k Residues
(C-N formed between successive <cite>Residue</cite> pairs).</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.c_join">
<code class="descname">c_join</code><span class="sig-paren">(</span><em>other</em>, <em>psi=-40.76</em>, <em>omega=-178.25</em>, <em>phi=-65.07</em>, <em>o_c_n_angle=None</em>, <em>c_n_ca_angle=None</em>, <em>c_n_length=None</em>, <em>relabel=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.c_join"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.c_join" title="Permalink to this definition">¶</a></dt>
<dd><p>Joins other to self at the C-terminus via a peptide bond.</p>
<p class="rubric">Notes</p>
<p>This function directly modifies self. It does not return a new object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>other</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – </li>
<li><strong>psi</strong> (<em>float</em><em>, </em><em>optional</em>) – Psi torsion angle (degrees) between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other.</li>
<li><strong>omega</strong> (<em>float</em><em>, </em><em>optional</em>) – Omega torsion angle (degrees) between final <cite>Residue</cite> of
self and first <cite>Residue</cite> of other.</li>
<li><strong>phi</strong> (<em>float</em><em>, </em><em>optional</em>) – Phi torsion angle (degrees) between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other.</li>
<li><strong>o_c_n_angle</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired angle between O, C (final <cite>Residue</cite> of self) and N
(first <cite>Residue</cite> of other) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</li>
<li><strong>c_n_ca_angle</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired angle between C (final <cite>Residue</cite> of self) and N, CA
(first <cite>Residue</cite> of other) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</li>
<li><strong>c_n_length</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired peptide bond length between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other. If <cite>None</cite>, default value is taken
from <cite>ideal_backbone_bond_lengths</cite>.</li>
<li><strong>relabel</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, <cite>relabel_all</cite> is run on self before returning.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last">TypeError: – If other is not a <cite>Residue</cite> or a Polypeptide.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.fasta">
<code class="descname">fasta</code><a class="headerlink" href="#ampal.protein.Polypeptide.fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates sequence data for the protein in FASTA format.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.get_slice_from_res_id">
<code class="descname">get_slice_from_res_id</code><span class="sig-paren">(</span><em>start</em>, <em>end</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.get_slice_from_res_id"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.get_slice_from_res_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Polypeptide</cite> containing the <cite>Residues</cite> in start/end range.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>start</strong> (<em>str</em>) – string representing start residue id (PDB numbering)</li>
<li><strong>end</strong> (<em>str</em>) – string representing end residue id (PDB numbering)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>slice_polymer</strong> – Polymer containing the residue range specified by start-end</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.isoelectric_point">
<code class="descname">isoelectric_point</code><a class="headerlink" href="#ampal.protein.Polypeptide.isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the isoelectric point of the <cite>Assembly</cite>.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.molar_extinction_280">
<code class="descname">molar_extinction_280</code><a class="headerlink" href="#ampal.protein.Polypeptide.molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the extinction co-efficient of the <cite>Assembly</cite> at 280 nm.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.molecular_weight">
<code class="descname">molecular_weight</code><a class="headerlink" href="#ampal.protein.Polypeptide.molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the <cite>Assembly</cite> in Daltons.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.n_join">
<code class="descname">n_join</code><span class="sig-paren">(</span><em>other</em>, <em>psi=-40.76</em>, <em>omega=-178.25</em>, <em>phi=-65.07</em>, <em>o_c_n_angle=None</em>, <em>c_n_ca_angle=None</em>, <em>c_n_length=None</em>, <em>relabel=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.n_join"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.n_join" title="Permalink to this definition">¶</a></dt>
<dd><p>Joins other to self at the N-terminus via a peptide bond.</p>
<p class="rubric">Notes</p>
<p>This function directly modifies self. It does not return a new object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>other</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – </li>
<li><strong>psi</strong> (<em>float</em>) – Psi torsion angle (degrees) between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self.</li>
<li><strong>omega</strong> (<em>float</em>) – Omega torsion angle (degrees) between final <cite>Residue</cite> of
other and first <cite>Residue</cite> of self.</li>
<li><strong>phi</strong> (<em>float</em>) – Phi torsion angle (degrees) between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self.</li>
<li><strong>o_c_n_angle</strong> (<em>float</em><em> or </em><em>None</em>) – Desired angle between O, C (final <cite>Residue</cite> of other) and N
(first <cite>Residue</cite> of self) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</li>
<li><strong>c_n_ca_angle</strong> (<em>float</em><em> or </em><em>None</em>) – Desired angle between C (final <cite>Residue</cite> of other) and N, CA
(first <cite>Residue</cite> of self) atoms. If <cite>None</cite>, default value is taken
from <cite>ideal_backbone_bond_angles</cite>.</li>
<li><strong>c_n_length</strong> (<em>float</em><em> or </em><em>None</em>) – Desired peptide bond length between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self. If None, default value is taken
from ideal_backbone_bond_lengths.</li>
<li><strong>relabel</strong> (<em>bool</em>) – If True, relabel_all is run on self before returning.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last">TypeError: – If other is not a <cite>Residue</cite> or a <cite>Polypeptide</cite></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.primitive">
<code class="descname">primitive</code><a class="headerlink" href="#ampal.protein.Polypeptide.primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Primitive of the backbone.</p>
<p class="rubric">Notes</p>
<p>This is the average of the positions of all the CAs in frames
of <cite>sl</cite> <cite>Residues</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.radii_of_curvature">
<code class="descname">radii_of_curvature</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.radii_of_curvature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.radii_of_curvature" title="Permalink to this definition">¶</a></dt>
<dd><p>List of radius of curvature values along the <cite>Polypeptide</cite>.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.rise_per_residue">
<code class="descname">rise_per_residue</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.rise_per_residue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.rise_per_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>List of rise per residue values along the <cite>Polypeptide</cite>.
.. rubric:: Notes</p>
<p>Calculated from <cite>Polypeptide.primitive</cite>.</p>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Polypeptide.sequence">
<code class="descname">sequence</code><a class="headerlink" href="#ampal.protein.Polypeptide.sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of the <cite>Polymer</cite> as a string.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>sequence</strong> – String of the <cite>Residue</cite> sequence of the <cite>Polypeptide</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.tag_ca_geometry">
<code class="descname">tag_ca_geometry</code><span class="sig-paren">(</span><em>force=False</em>, <em>reference_axis=None</em>, <em>reference_axis_name='ref_axis'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_ca_geometry"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_ca_geometry" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Residue</cite> with rise_per_residue, radius_of_curvature and residues_per_turn.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are already
tagged.</li>
<li><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em><em>, </em><em>optional</em>) – Coordinates to feed to geometry functions that depend on
having a reference axis.</li>
<li><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at <cite>Polypeptide</cite> and <cite>Residue</cite> level.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.tag_sidechain_dihedrals">
<code class="descname">tag_sidechain_dihedrals</code><span class="sig-paren">(</span><em>force=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_sidechain_dihedrals"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_sidechain_dihedrals" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each monomer with side-chain dihedral angles</p>
<dl class="docutils">
<dt>force: bool, optional</dt>
<dd>If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are
already tagged.</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.tag_torsion_angles">
<code class="descname">tag_torsion_angles</code><span class="sig-paren">(</span><em>force=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_torsion_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each Monomer of the Polymer with its omega, phi and psi torsion angle.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are
already tagged.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.valid_backbone_bond_angles">
<code class="descname">valid_backbone_bond_angles</code><span class="sig-paren">(</span><em>atol=20</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.valid_backbone_bond_angles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.valid_backbone_bond_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>True if all backbone bond angles are within atol degrees of their expected values.</p>
<p class="rubric">Notes</p>
<p>Ideal bond angles taken from [1].</p>
<p class="rubric">References</p>
<table class="docutils footnote" frame="void" id="id7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[1]</td><td>Schulz, G. E, and R. Heiner Schirmer. Principles Of
Protein Structure. New York: Springer-Verlag, 1979.</td></tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>atol</strong> (<em>float</em><em>, </em><em>optional</em>) – Tolerance value in degrees for the absolute deviation
away from ideal backbone bond angles.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="ampal.protein.Polypeptide.valid_backbone_bond_lengths">
<code class="descname">valid_backbone_bond_lengths</code><span class="sig-paren">(</span><em>atol=0.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.valid_backbone_bond_lengths"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Polypeptide.valid_backbone_bond_lengths" title="Permalink to this definition">¶</a></dt>
<dd><p>True if all backbone bonds are within atol Angstroms of the expected distance.</p>
<p class="rubric">Notes</p>
<p>Ideal bond lengths taken from [1].</p>
<p class="rubric">References</p>
<table class="docutils footnote" frame="void" id="id8" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[1]</td><td>Schulz, G. E, and R. Heiner Schirmer. Principles Of
Protein Structure. New York: Springer-Verlag, 1979.</td></tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>atol</strong> (<em>float</em><em>, </em><em>optional</em>) – Tolerance value in Angstoms for the absolute deviation
away from ideal backbone bond lengths.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.protein.Residue">
<em class="property">class </em><code class="descclassname">ampal.protein.</code><code class="descname">Residue</code><span class="sig-paren">(</span><em>atoms=None</em>, <em>mol_code='UNK'</em>, <em>monomer_id=' '</em>, <em>insertion_code=' '</em>, <em>is_hetero=False</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Residue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.Residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Monomer</span></code></a></p>
<p>Represents a amino acid <cite>Residue</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</li>
<li><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the monomer.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</li>
<li><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of monomer, used if reading from pdb.</li>
<li><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</li>
<li><strong>parent</strong> (<em>ampal.Polypeptide</em><em>, </em><em>optional</em>) – Reference to <cite>Polypeptide</cite> containing the <cite>Residue</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.protein.Residue.mol_code">
<code class="descname">mol_code</code><a class="headerlink" href="#ampal.protein.Residue.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Residue</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.insertion_code">
<code class="descname">insertion_code</code><a class="headerlink" href="#ampal.protein.Residue.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Residue</cite>, used if reading from pdb.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.is_hetero">
<code class="descname">is_hetero</code><a class="headerlink" href="#ampal.protein.Residue.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.states">
<code class="descname">states</code><a class="headerlink" href="#ampal.protein.Residue.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Residue</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.protein.Residue.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.reference_atom">
<code class="descname">reference_atom</code><a class="headerlink" href="#ampal.protein.Residue.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference atom. This is used
by various functions, for example backbone primitives are
calculated using the atom defined using this key.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.protein.Residue.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polypeptide</cite> containing this <cite>Residue</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.protein.Residue.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if <cite>mol_code</cite> is not length 1 or 3.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.protein.Residue.backbone">
<code class="descname">backbone</code><a class="headerlink" href="#ampal.protein.Residue.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Residue</cite> containing only the backbone atoms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>bb_monomer</strong> – <cite>Residue</cite> containing only the backbone atoms of the original
<cite>Monomer</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue">Residue</a></td>
</tr>
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">IndexError</span></code> – Raise if the <cite>atoms</cite> dict does not contain the backbone
atoms (N, CA, C, O).</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.centroid">
<code class="descname">centroid</code><a class="headerlink" href="#ampal.protein.Residue.centroid" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the centroid of the residue.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>centroid</strong> – Returns a 3D coordinate for the residue unless a CB
atom is not available, in which case <cite>None</cite> is
returned.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">numpy.array or None</td>
</tr>
</tbody>
</table>
<p class="rubric">Notes</p>
<p>Uses the definition of the centroid from Huang <em>et al</em> <a class="footnote-reference" href="#id10" id="id9">[2]</a>.</p>
<p class="rubric">References</p>
<table class="docutils footnote" frame="void" id="id10" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id9">[2]</a></td><td>Huang ES, Subbiah S and Levitt M (1995) Recognizing Native
Folds by the Arrangement of Hydrophobic and Polar Residues, J. Mol.
Biol return., <strong>252</strong>, 709-720.</td></tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.side_chain">
<code class="descname">side_chain</code><a class="headerlink" href="#ampal.protein.Residue.side_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>List of the side-chain atoms (R-group).</p>
<p class="rubric">Notes</p>
<p>Returns empty list for glycine.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>side_chain_atoms</strong></td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">list(<cite>Atoms</cite>)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.protein.Residue.unique_id">
<code class="descname">unique_id</code><a class="headerlink" href="#ampal.protein.Residue.unique_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a tuple that uniquely identifies a <cite>Monomer</cite> in an <cite>Assembly</cite>.</p>
<p class="rubric">Notes</p>
<p>The unique_id will uniquely identify each monomer within a polymer.
If each polymer in an assembly has a distinct id, it will uniquely
identify each monomer within the assembly.</p>
<p>The hetero-flag is defined as in Biopython as a string that is
either a single whitespace in the case of a non-hetero atom,
or ‘<a href="#id11"><span class="problematic" id="id12">H_</span></a>’ plus the name of the hetero-residue (e.g. ‘H_GLC’ in
the case of a glucose molecule), or ‘W’ in the case of a water
molecule.</p>
<p>For more information, see the Biopython documentation or this
Biopython wiki page:
<a class="reference external" href="http://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ">http://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ</a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><strong>unique_id</strong> – unique_id[0] is the polymer_id unique_id[1] is a triple
of the hetero-flag, the monomer id (residue number) and the
insertion code.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">tuple</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="ampal.protein.align">
<code class="descclassname">ampal.protein.</code><code class="descname">align</code><span class="sig-paren">(</span><em>target</em>, <em>mobile</em>, <em>target_i=0</em>, <em>mobile_i=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#align"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.align" title="Permalink to this definition">¶</a></dt>
<dd><p>Aligns one Polypeptide (mobile) to another (target).</p>
<p class="rubric">Notes</p>
<p>This function directly modifies atoms of the mobile Polypeptide!
It does not return a new object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>target</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – Polypeptide to be aligned to.</li>
<li><strong>mobile</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – Polypeptide to be moved during alignment.</li>
<li><strong>target_i</strong> (<em>int</em><em>, </em><em>optional</em>) – Index of <cite>Residue</cite> in target to align to.</li>
<li><strong>mobile_i</strong> (<em>int</em><em>, </em><em>optional</em>) – Index of <cite>Residue</cite> in mobile to be aligned.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.protein.flat_list_to_dummy_chain">
<code class="descclassname">ampal.protein.</code><code class="descname">flat_list_to_dummy_chain</code><span class="sig-paren">(</span><em>atom_list</em>, <em>atom_group_s=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#flat_list_to_dummy_chain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.flat_list_to_dummy_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts flat list of coordinates into dummy C-alpha carbons</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atom_list</strong> (<em>[</em><a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><em>Atom</em></a><em>]</em>) – Flat list of co-ordinates.</li>
<li><strong>atom_group_s</strong> (<em>int</em><em>, </em><em>optional</em>) – Size of atom groups.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>polymer</strong> – <cite>Polymer</cite> object containing atom coord converted <cite>Monomers</cite>
with ‘DUM’ atom name.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last"><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="ampal.protein.flat_list_to_polymer">
<code class="descclassname">ampal.protein.</code><code class="descname">flat_list_to_polymer</code><span class="sig-paren">(</span><em>atom_list</em>, <em>atom_group_s=4</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#flat_list_to_polymer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.protein.flat_list_to_polymer" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes a flat list of atomic coordinates and converts it to a <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atom_list</strong> (<em>[</em><a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><em>Atom</em></a><em>]</em>) – Flat list of coordinates.</li>
<li><strong>atom_group_s</strong> (<em>int</em><em>, </em><em>optional</em>) – Size of atom groups.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first"><strong>polymer</strong> – <cite>Polymer</cite> object containing atom coords converted <cite>Monomers</cite>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first"><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><p class="first last"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if <cite>atom_group_s</cite> != 4 or 5</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-ampal.pseudo_atoms">
<span id="ampal-pseudo-atoms-module"></span><h2>ampal.pseudo_atoms module<a class="headerlink" href="#module-ampal.pseudo_atoms" title="Permalink to this headline">¶</a></h2>
<p>Contains AMPAL objects representing pseudo atoms.</p>
<dl class="class">
<dt id="ampal.pseudo_atoms.Primitive">
<em class="property">class </em><code class="descclassname">ampal.pseudo_atoms.</code><code class="descname">Primitive</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>polymer_id=' '</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.pseudo_atoms.PseudoGroup" title="ampal.pseudo_atoms.PseudoGroup"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.pseudo_atoms.PseudoGroup</span></code></a></p>
<p>A backbone path composed of <cite>PseudoAtoms</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>pseudo_atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>PsuedoMonomer</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</li>
<li><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>PsuedoMonomer</cite>.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>PsuedoMonomer</cite>.</li>
<li><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>PsuedoMonomer</cite>, used if reading from pdb.</li>
<li><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</li>
<li><strong>parent</strong> (<em>ampal.PseudoGroup</em><em>, </em><em>optional</em>) – Reference to <cite>PseudoGroup</cite> containing the <cite>PsuedoMonomer</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.mol_code">
<code class="descname">mol_code</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.insertion_code">
<code class="descname">insertion_code</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.is_hetero">
<code class="descname">is_hetero</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.states">
<code class="descname">states</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.reference_atom">
<code class="descname">reference_atom</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if <cite>mol_code</cite> is not length 1 or 3.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.Primitive.coordinates">
<code class="descname">coordinates</code><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the backbone coordinates for the <cite>Primitive</cite>.</p>
</dd></dl>

<dl class="classmethod">
<dt id="ampal.pseudo_atoms.Primitive.from_coordinates">
<em class="property">classmethod </em><code class="descname">from_coordinates</code><span class="sig-paren">(</span><em>coordinates</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.from_coordinates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.from_coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a <cite>Primitive</cite> from a list of coordinates.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pseudo_atoms.Primitive.radii_of_curvature">
<code class="descname">radii_of_curvature</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.radii_of_curvature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.radii_of_curvature" title="Permalink to this definition">¶</a></dt>
<dd><p>The radius of curvature at each point on the Polymer primitive.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the radius of curvature,
at a point on the Polymer primitive. Element i is the radius
of the circumcircle formed from indices [i-1, i, i+1] of the
primitve. The first and final values are None.</p>
</dd></dl>

<dl class="method">
<dt id="ampal.pseudo_atoms.Primitive.rise_per_residue">
<code class="descname">rise_per_residue</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.rise_per_residue"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.rise_per_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>The rise per residue at each point on the Primitive.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the rise per residue,
at a point on the Primitive. Element i is the distance
between primitive[i] and primitive[i + 1]. The final value
is None.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.pseudo_atoms.PseudoAtom">
<em class="property">class </em><code class="descclassname">ampal.pseudo_atoms.</code><code class="descname">PseudoAtom</code><span class="sig-paren">(</span><em>coordinates</em>, <em>name=''</em>, <em>occupancy=1.0</em>, <em>bfactor=1.0</em>, <em>charge=' '</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoAtom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Atom</span></code></a></p>
<p>Object containing 3D coordinates and name.</p>
<p class="rubric">Notes</p>
<p>Used to represent pseudo atoms (e.g. centre_of_mass) in ISAMBARD.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>coordinates</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Position of <cite>PseudoAtom</cite> in 3D space.</li>
<li><strong>element</strong> (<em>str</em>) – Element of <cite>PseudoAtom</cite>.</li>
<li><strong>atom_id</strong> (<em>str</em>) – Identifier for <cite>PseudoAtom</cite>, usually a number.</li>
<li><strong>res_label</strong> (<em>str</em><em>, </em><em>optional</em>) – Label used in <cite>Monomer</cite> to refer to the <cite>PseudoAtom</cite> type i.e.
“CA” or “OD1”.</li>
<li><strong>occupancy</strong> (<em>float</em><em>, </em><em>optional</em>) – The occupancy of the <cite>PseudoAtom</cite>.</li>
<li><strong>bfactor</strong> (<em>float</em><em>, </em><em>optional</em>) – The bfactor of the <cite>PseudoAtom</cite>.</li>
<li><strong>charge</strong> (<em>str</em><em>, </em><em>optional</em>) – The point charge of the <cite>PseudoAtom</cite>.</li>
<li><strong>state</strong> (<em>str</em>) – The state of this <cite>PseudoAtom</cite>. Used to identify <cite>PseudoAtoms</cite>
with a number of conformations.</li>
<li><strong>parent</strong> (<em>ampal.Monomer</em><em>, </em><em>optional</em>) – A reference to the <cite>Monomer</cite> containing this <cite>PseudoAtom</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoAtom.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for <cite>PseudoAtom</cite>, usually a number.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoAtom.res_label">
<code class="descname">res_label</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.res_label" title="Permalink to this definition">¶</a></dt>
<dd><p>Label used in <cite>PseudoGroup</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoAtom.element">
<code class="descname">element</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.element" title="Permalink to this definition">¶</a></dt>
<dd><p>Element of <cite>Atom</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoAtom.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.parent" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>A reference to the <cite>PseudoGroup</cite> containing this <cite>PseudoAtom</cite>.</dt>
<dd>number of conformations.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.PseudoAtom</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoAtom.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ampal.pseudo_atoms.PseudoGroup">
<em class="property">class </em><code class="descclassname">ampal.pseudo_atoms.</code><code class="descname">PseudoGroup</code><span class="sig-paren">(</span><em>monomers=None</em>, <em>polymer_id=' '</em>, <em>parent=None</em>, <em>sl=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoGroup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Polymer</span></code></a></p>
<p>Container for <cite>PseudoMonomer</cite>, inherits from <cite>Polymer</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>monomers</strong> (<a class="reference internal" href="#ampal.pseudo_atoms.PseudoAtom" title="ampal.pseudo_atoms.PseudoAtom"><em>PseudoAtom</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.pseudo_atoms.PseudoGroup" title="ampal.pseudo_atoms.PseudoGroup"><em>PseudoGroup</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>PseudoMonomer</cite> or list containing <cite>PseudoMonomer</cite> objects to form the
<cite>PseudoGroup</cite>.</li>
<li><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>PseudoGroup</cite>. This is
used when generating a pdb file using <cite>PseudoGroup().pdb</cite>.</li>
<li><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>PseudoGroup</cite>.</li>
<li><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>PseudoGroup</cite> ID</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>PseudoGroup</cite></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.Assembly or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.molecule_type">
<code class="descname">molecule_type</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.ligands">
<code class="descname">ligands</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>PseudoGroup</cite> chain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">ampal.LigandGroup</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoGroup.sl">
<code class="descname">sl</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TypeError</span></code> – <cite>Polymer</cite> type objects can only be initialised empty or using
a <cite>Monomer</cite>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="ampal.pseudo_atoms.PseudoMonomer">
<em class="property">class </em><code class="descclassname">ampal.pseudo_atoms.</code><code class="descname">PseudoMonomer</code><span class="sig-paren">(</span><em>pseudo_atoms=None</em>, <em>mol_code='UNK'</em>, <em>monomer_id=' '</em>, <em>insertion_code=' '</em>, <em>parent=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoMonomer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ampal.base_ampal.Monomer</span></code></a></p>
<p>Represents a collection of <cite>PsuedoAtoms</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>pseudo_atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>PsuedoMonomer</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</li>
<li><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>PsuedoMonomer</cite>.</li>
<li><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>PsuedoMonomer</cite>.</li>
<li><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>PsuedoMonomer</cite>, used if reading from pdb.</li>
<li><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</li>
<li><strong>parent</strong> (<em>ampal.PseudoGroup</em><em>, </em><em>optional</em>) – Reference to <cite>PseudoGroup</cite> containing the <cite>PsuedoMonomer</cite>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.mol_code">
<code class="descname">mol_code</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.insertion_code">
<code class="descname">insertion_code</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.is_hetero">
<code class="descname">is_hetero</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">bool</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.states">
<code class="descname">states</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.id">
<code class="descname">id</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.reference_atom">
<code class="descname">reference_atom</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">str</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.parent">
<code class="descname">parent</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body"><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.tags">
<code class="descname">tags</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Type:</th><td class="field-body">dict</td>
</tr>
</tbody>
</table>
</dd></dl>

<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ValueError</span></code> – Raised if <cite>mol_code</cite> is not length 1 or 3.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="ampal.pseudo_atoms.PseudoMonomer.pdb">
<code class="descname">pdb</code><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>PseudoMonomer</cite>.</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-ampal">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-ampal" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">AMPAL</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="AMPAL_Introduction_and_Selections.html">Introduction to AMPAL</a></li>
<li class="toctree-l1"><a class="reference internal" href="Advanced_Selections_and_Analysis.html">Advanced Selections</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API Reference</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">ampal package</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="modules.html">ampal</a><ul>
      <li>Previous: <a href="modules.html" title="previous chapter">ampal</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Woolfson Group, UoB.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/ampal.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>