<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>ampal package &#8212; AMPAL 1.4.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="ampal" href="modules.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="ampal-package">
<h1>ampal package<a class="headerlink" href="#ampal-package" title="Permalink to this heading">¶</a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this heading">¶</a></h2>
</section>
<section id="module-ampal.align">
<span id="ampal-align-module"></span><h2>ampal.align module<a class="headerlink" href="#module-ampal.align" title="Permalink to this heading">¶</a></h2>
<p>A module containing classes for aligning structure.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.align.MMCAlign">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.align.</span></span><span class="sig-name descname"><span class="pre">MMCAlign</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">eval_fn</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eval_args</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polypeptide</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><span class="pre">Polypeptide</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.align.MMCAlign" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A alignment protocol that uses Metropolis Monte Carlo.</p>
<p class="rubric">Notes</p>
<p>THIS IS CURRENTLY SUPER INEFFICIENT DUE TO THE DEEPCOPIES.
I plan to improve this by aligning arrays of atoms and only
recording the rotation and translation that led to that alignment.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>eval_fn</strong> (<em>Polypeptide -&gt; float</em>) – A function to evaluate the quality of your fit.</p></li>
<li><p><strong>eval_args</strong> (<em>list</em>) – A list of static args to be used in the <cite>eval_fn</cite>, these will
be unpacked into the evaluation function _i.e._
<a href="#id1"><span class="problematic" id="id2">`</span></a>eval_fn(polypeptide, <a href="#id3"><span class="problematic" id="id4">*</span></a>eval_args).</p></li>
<li><p><strong>polypeptide</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – An ampal polypeptide containing the model to be aligned.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.align.MMCAlign.check_move">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">check_move</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">new</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">old</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.check_move"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.align.MMCAlign.check_move" title="Permalink to this definition">¶</a></dt>
<dd><p>Determines if a model will be accepted.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.align.MMCAlign.float_f">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">float_f</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">f</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.float_f"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.align.MMCAlign.float_f" title="Permalink to this definition">¶</a></dt>
<dd><p>Formats a float for printing to std out.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.align.MMCAlign.start_optimisation">
<span class="sig-name descname"><span class="pre">start_optimisation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">rounds</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_angle</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_distance</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temp</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">298.15</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stop_when</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#MMCAlign.start_optimisation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.align.MMCAlign.start_optimisation" title="Permalink to this definition">¶</a></dt>
<dd><p>Starts the loop fitting protocol.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>rounds</strong> (<em>int</em>) – The number of Monte Carlo moves to be evaluated.</p></li>
<li><p><strong>max_angle</strong> (<em>float</em>) – The maximum variation in rotation that can moved per
step.</p></li>
<li><p><strong>max_distance</strong> (<em>float</em>) – The maximum distance the can be moved per step.</p></li>
<li><p><strong>temp</strong> (<em>float</em><em>, </em><em>optional</em>) – Temperature used during fitting process.</p></li>
<li><p><strong>stop_when</strong> (<em>float</em><em>, </em><em>optional</em>) – Stops fitting when energy is less than or equal to this value.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.align.align_backbones">
<span class="sig-prename descclassname"><span class="pre">ampal.align.</span></span><span class="sig-name descname"><span class="pre">align_backbones</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reference</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mobile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stop_when</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/align.html#align_backbones"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.align.align_backbones" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</section>
<section id="module-ampal.amino_acids">
<span id="ampal-amino-acids-module"></span><h2>ampal.amino_acids module<a class="headerlink" href="#module-ampal.amino_acids" title="Permalink to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ampal.amino_acids.add_amino_acid_to_json">
<span class="sig-prename descclassname"><span class="pre">ampal.amino_acids.</span></span><span class="sig-name descname"><span class="pre">add_amino_acid_to_json</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">code</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">description</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">letter</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'X'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">modified</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">force_add</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#add_amino_acid_to_json"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.amino_acids.add_amino_acid_to_json" title="Permalink to this definition">¶</a></dt>
<dd><p>Add an amino acid to the amino_acids.json file used to populate the amino_acid table.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>code</strong> (<em>str</em>) – New code to be added to amino acid table.</p></li>
<li><p><strong>description</strong> (<em>str</em>) – Description of the amino acid, e.g. ‘amidated terminal carboxy group’.</p></li>
<li><p><strong>letter</strong> (<em>str</em><em>, </em><em>optional</em>) – One letter code for the amino acid.
Defaults to ‘X’</p></li>
<li><p><strong>modified</strong> (<em>str</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Code of modified amino acid, e.g. ‘ALA’, or None.
Defaults to None</p></li>
<li><p><strong>force_add</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, will over-write existing dictionary value for code if already in amino_acids.json.
If False, then an IOError is raised if code is already in amino_acids.json.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>IOError</strong> – If code is already in amino_acids.json and force_add is False.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.amino_acids.get_aa_code">
<span class="sig-prename descclassname"><span class="pre">ampal.amino_acids.</span></span><span class="sig-name descname"><span class="pre">get_aa_code</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aa_letter</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_code"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Get three-letter aa code if possible. If not, return None.</p>
<p>If three-letter code is None, will have to find this later from the filesystem.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>aa_letter</strong> (<em>str</em>) – One-letter amino acid code.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>aa_code</strong> – Three-letter aa code.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str, or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.amino_acids.get_aa_info">
<span class="sig-prename descclassname"><span class="pre">ampal.amino_acids.</span></span><span class="sig-name descname"><span class="pre">get_aa_info</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">code</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_info"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_info" title="Permalink to this definition">¶</a></dt>
<dd><p>Get dictionary of information relating to a new amino acid code not currently in the database.</p>
<p class="rubric">Notes</p>
<p>Use this function to get a dictionary that is then to be sent to the function add_amino_acid_to_json().
use to fill in rows of amino_acid table for new amino acid code.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>code</strong> (<em>str</em>) – Three-letter amino acid code.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>IOError</strong> – If unable to locate the page associated with the amino acid name on the PDBE site.</p>
</dd>
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>aa_dict</strong> – Keys are AminoAcidDB field names.
Values are the str values for the new amino acid, scraped from the PDBE if possible. None if not found.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.amino_acids.get_aa_letter">
<span class="sig-prename descclassname"><span class="pre">ampal.amino_acids.</span></span><span class="sig-name descname"><span class="pre">get_aa_letter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aa_code</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/amino_acids.html#get_aa_letter"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.amino_acids.get_aa_letter" title="Permalink to this definition">¶</a></dt>
<dd><p>Get one-letter version of aa_code if possible. If not, return ‘X’.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>aa_code</strong> (<em>str</em>) – Three-letter amino acid code.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>aa_letter</strong> – One-letter aa code.
Default value is ‘X’.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.ampal_warnings">
<span id="ampal-ampal-warnings-module"></span><h2>ampal.ampal_warnings module<a class="headerlink" href="#module-ampal.ampal_warnings" title="Permalink to this heading">¶</a></h2>
<dl class="py exception">
<dt class="sig sig-object py" id="ampal.ampal_warnings.DependencyNotFoundWarning">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ampal_warnings.</span></span><span class="sig-name descname"><span class="pre">DependencyNotFoundWarning</span></span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#DependencyNotFoundWarning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ampal_warnings.DependencyNotFoundWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="ampal.ampal_warnings.MalformedPDBWarning">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ampal_warnings.</span></span><span class="sig-name descname"><span class="pre">MalformedPDBWarning</span></span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#MalformedPDBWarning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ampal_warnings.MalformedPDBWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="ampal.ampal_warnings.NoncanonicalWarning">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ampal_warnings.</span></span><span class="sig-name descname"><span class="pre">NoncanonicalWarning</span></span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#NoncanonicalWarning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ampal_warnings.NoncanonicalWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="ampal.ampal_warnings.NotParameterisedWarning">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ampal_warnings.</span></span><span class="sig-name descname"><span class="pre">NotParameterisedWarning</span></span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#NotParameterisedWarning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ampal_warnings.NotParameterisedWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">RuntimeWarning</span></code></p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.ampal_warnings.check_availability">
<span class="sig-prename descclassname"><span class="pre">ampal.ampal_warnings.</span></span><span class="sig-name descname"><span class="pre">check_availability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">program</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">test_func</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">global_settings</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ampal_warnings.html#check_availability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ampal_warnings.check_availability" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</section>
<section id="module-ampal.analyse_protein">
<span id="ampal-analyse-protein-module"></span><h2>ampal.analyse_protein module<a class="headerlink" href="#module-ampal.analyse_protein" title="Permalink to this heading">¶</a></h2>
<p>Contains various tools for analysing protein structure.</p>
<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.alpha_angles">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">alpha_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ref_axis'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#alpha_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.alpha_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Alpha angle calculated using points on the primitive of helix and axis.</p>
<p class="rubric">Notes</p>
<p>The final value is None, since the angle calculation requires pairs
of points along the primitive and axis. This is a generalisation
of the calculation used to measure the tilt of a helix in a
coiled-coil with respect to the central axis of the coiled coil.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</p></li>
<li><p><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, tags the Chain with the reference axis coordinates
and each Residue with its alpha angle. Alpha angles are stored
at the Residue level, but are calculated using the CA atom.</p></li>
<li><p><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>alphas</strong> – The alpha angle for the Polymer at each point of its primitive,
in degrees.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of float</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the Polymer and the reference_axis have unequal length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.cc_to_local_params">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">cc_to_local_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pitch</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">oligo</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#cc_to_local_params"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.cc_to_local_params" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns local parameters for an oligomeric assembly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pitch</strong> (<em>float</em>) – Pitch of assembly</p></li>
<li><p><strong>radius</strong> (<em>float</em>) – Radius of assembly</p></li>
<li><p><strong>oligo</strong> (<em>int</em>) – Oligomeric state of assembly</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>pitchloc</strong> (<em>float</em>) – Local pitch of assembly (between 2 adjacent component helices)</p></li>
<li><p><strong>rloc</strong> (<em>float</em>) – Local radius of assembly</p></li>
<li><p><strong>alphaloc</strong> (<em>float</em>) – Local pitch-angle of assembly</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.charge_series">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">charge_series</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#charge_series"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.charge_series" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the charge for pH 1-13.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</p></li>
<li><p><strong>granularity</strong> (<em>float</em><em>, </em><em>optional</em>) – Granularity of pH values i.e. if 0.1 pH = [1.0, 1.1, 1.2…]</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.classify_angle_as_rotamer">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">classify_angle_as_rotamer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angle</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="_modules/ampal/analyse_protein.html#classify_angle_as_rotamer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.classify_angle_as_rotamer" title="Permalink to this definition">¶</a></dt>
<dd><p>Classifies a chi angle into a rotamer.</p>
<p class="rubric">Notes</p>
<p>This function uses the convention 1:g+ 2:t 3:g- as from 10.1006/jmbi.1993.1170
R L Dunbrack Jr, M Karplus (1993) Backbone-dependent rotamer library for proteins.
Application to side-chain prediction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>angle</strong> (<em>float</em>) – Chi angle to classify.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.crick_angles">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">crick_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ref_axis'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#crick_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.crick_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the Crick angle for each CA atom in the <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>The final value is in the returned list is <cite>None</cite>, since the angle
calculation requires pairs of points on both the primitive and
reference_axis.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</p></li>
<li><p><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, tags the <cite>Polymer</cite> with the reference axis coordinates
and each Residue with its Crick angle. Crick angles are stored
at the Residue level, but are calculated using the CA atom.</p></li>
<li><p><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>cr_angles</strong> – The crick angles in degrees for each CA atom of the Polymer.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(float)</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the Polymer and the reference_axis have unequal length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.flip_reference_axis_if_antiparallel">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">flip_reference_axis_if_antiparallel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start_index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end_index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#flip_reference_axis_if_antiparallel"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.flip_reference_axis_if_antiparallel" title="Permalink to this definition">¶</a></dt>
<dd><p>Flips reference axis if direction opposes the direction of the <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>If the angle between the vector for the Polymer and the vector
for the reference_axis is &gt; 90 degrees, then the reference axis
is reversed. This is useful to run before running
polymer_to_reference_axis_distances, crick_angles, or alpha_angles.
For more information on the start and end indices, see chain_vector.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</p></li>
<li><p><strong>start_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is 0 (start at the N-terminus of the Polymer)</p></li>
<li><p><strong>end_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is -1 (start at the C-terminus of the Polymer)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>reference_axis</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(numpy.array or tuple or list)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.make_primitive">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">make_primitive</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cas_coords</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates running average of cas_coords with a fixed averaging window_length.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</p></li>
<li><p><strong>window_length</strong> (<em>int</em><em>, </em><em>optional</em>) – The number of coordinate sets to average each time.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>s_primitive</strong> – Each array has length 3.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(numpy.array)</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the length of cas_coords is smaller than the window_length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.make_primitive_extrapolate_ends">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">make_primitive_extrapolate_ends</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cas_coords</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smoothing_level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive_extrapolate_ends"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive_extrapolate_ends" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates smoothed helix primitives and extrapolates lost ends.</p>
<p class="rubric">Notes</p>
<p>From an input list of CA coordinates, the running average is
calculated to form a primitive. The smoothing_level dictates how
many times to calculate the running average. A higher
smoothing_level generates a ‘smoother’ primitive - i.e. the
points on the primitive more closely fit a smooth curve in R^3.
Each time the smoothing level is increased by 1, a point is lost
from either end of the primitive. To correct for this, the primitive
is extrapolated at the ends to approximate the lost values. There
is a trade-off then between the smoothness of the primitive and
its accuracy at the ends.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</p></li>
<li><p><strong>smoothing_level</strong> (<em>int</em>) – Number of times to run the averaging.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>final_primitive</strong> – Each array has length 3.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(numpy.array)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.make_primitive_smoothed">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">make_primitive_smoothed</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cas_coords</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">smoothing_level</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#make_primitive_smoothed"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.make_primitive_smoothed" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates smoothed primitive from a list of coordinates.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cas_coords</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>float</em><em> or </em><em>tuple</em><em>)</em>) – Each element of the list must have length 3.</p></li>
<li><p><strong>smoothing_level</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of times to run the averaging.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>s_primitive</strong> – Each array has length 3.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(numpy.array)</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the smoothing level is too great compared to the length
    of cas_coords.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.measure_sidechain_torsion_angles">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">measure_sidechain_torsion_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="_modules/ampal/analyse_protein.html#measure_sidechain_torsion_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.measure_sidechain_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates sidechain dihedral angles for a residue</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>residue</strong> (<em>[</em><em>ampal.Residue</em><em>]</em>) – <cite>Residue</cite> object.</p></li>
<li><p><strong>verbose</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, tells you when a residue does not have any known
dihedral angles to measure.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul>
<li><p><strong>chi_angles</strong> (<em>[float]</em>) – Length depends on residue type, in range [-pi, pi]</p>
<p>[0] = chi1 [if applicable]
[1] = chi2 [if applicable]
[2] = chi3 [if applicable]
[3] = chi4 [if applicable]</p>
</li>
<li><p><strong>rotamers</strong> (<em>[int]</em>) – Rotamer class according to Dunbrack 1993 {1, 2, 3}</p>
<p>[0] = chi1 [if applicable]
[1] = chi2 [if applicable]
[2] = chi3 [if applicable]
[3] = chi4 [if applicable]</p>
</li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.measure_torsion_angles">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">measure_torsion_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residues</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#measure_torsion_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.measure_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the dihedral angles for a list of backbone atoms.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>residues</strong> (<em>[</em><em>ampal.Residue</em><em>]</em>) – List of <cite>Residue</cite> objects.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><p><strong>torsion_angles</strong> – One triple for each residue, containing torsion angles in
the range [-pi, pi].</p>
<blockquote>
<div><p>[0] omega
[1] phi
[2] psi</p>
</div></blockquote>
<p>For the first residue, omega and phi are not defined. For
the final residue, psi is not defined.</p>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>(float, float, float)</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the number of input residues is less than 2.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.partial_charge">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">partial_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aa</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pH</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#partial_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.partial_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the partial charge of the amino acid.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>aa</strong> (<em>str</em>) – Amino acid single-letter code.</p></li>
<li><p><strong>pH</strong> (<em>float</em>) – pH of interest.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.polymer_to_reference_axis_distances">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">polymer_to_reference_axis_distances</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ref_axis'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#polymer_to_reference_axis_distances"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.polymer_to_reference_axis_distances" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns distances between the primitive of a Polymer and a reference_axis.</p>
<p class="rubric">Notes</p>
<p>Distances are calculated between each point of the Polymer primitive
and the corresponding point in reference_axis. In the special case
of the helical barrel, if the Polymer is a helix and the reference_axis
represents the centre of the barrel, then this function returns the
radius of the barrel at each point on the helix primitive. The points
of the primitive and the reference_axis are run through in the same
order, so take care with the relative orientation of the reference
axis when defining it.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ampal.Polymer</em>) – </p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em>) – Length of reference_axis must equal length of the Polymer.
Each element of reference_axis represents a point in R^3.</p></li>
<li><p><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, tags the Chain with the reference axis coordinates
and each Residue with its distance to the ref axis.
Distances are stored at the Residue level, but refer to
distances from the CA atom.</p></li>
<li><p><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at Chain and Residue level.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>distances</strong> – Distance values between corresponding points on the
reference axis and the <cite>Polymer</cite> <cite>Primitive</cite>.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(float)</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – If the Polymer and the reference_axis have unequal length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.polypeptide_vector">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">polypeptide_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start_index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end_index</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#polypeptide_vector"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.polypeptide_vector" title="Permalink to this definition">¶</a></dt>
<dd><p>Vector along the Chain primitive (default is from N-terminus to C-terminus).</p>
<p class="rubric">Notes</p>
<p><cite>start_index</cite> and <cite>end_index</cite> can be changed to examine smaller
sections of the Chain, or reversed to change the direction of
the vector.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ampal.Polymer</em>) – Reference <cite>Polymer</cite>.</p></li>
<li><p><strong>start_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is 0 (start at the N-terminus of the Chain)</p></li>
<li><p><strong>end_index</strong> (<em>int</em><em>, </em><em>optional</em>) – Default is -1 (start at the C-terminus of the Chain)</p></li>
<li><p><strong>unit</strong> (<em>bool</em>) – If True, the vector returned has a magnitude of 1.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>vector</strong> – vector has shape (1, 3)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>a numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.reference_axis_from_chains">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">reference_axis_from_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chains</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#reference_axis_from_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.reference_axis_from_chains" title="Permalink to this definition">¶</a></dt>
<dd><p>Average coordinates from a set of primitives calculated from Chains.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>chains</strong> (<em>list</em><em>(</em><em>Chain</em><em>)</em>) – </p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>reference_axis</strong> – The averaged (x, y, z) coordinates of the primitives for
the list of Chains. In the case of a coiled coil barrel,
this would give the central axis for calculating e.g. Crick
angles.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError :</strong> – If the Chains are not all of the same length.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.residues_per_turn">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">residues_per_turn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#residues_per_turn"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.residues_per_turn" title="Permalink to this definition">¶</a></dt>
<dd><p>The number of residues per turn at each Monomer in the Polymer.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the number of residues
per turn, at a point on the Polymer primitive. Calculated using
the relative positions of the CA atoms and the primitive of the
Polymer. Element i is the calculated from the dihedral angle using
the CA atoms of the Monomers with indices [i, i+1] and the
corresponding atoms of the primitive. The final value is None.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>p</strong> (<em>ampal.Polypeptide</em>) – <cite>Polypeptide</cite> from which residues per turn will be calculated.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>rpts</strong> – Residue per turn values.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[float]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.sequence_charge">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">sequence_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pH</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7.4</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the total charge of the input polypeptide sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</p></li>
<li><p><strong>pH</strong> (<em>float</em>) – pH of interest.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.sequence_isoelectric_point">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">sequence_isoelectric_point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_isoelectric_point"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the isoelectric point of the sequence for ph 1-13.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</p></li>
<li><p><strong>granularity</strong> (<em>float</em><em>, </em><em>optional</em>) – Granularity of pH values i.e. if 0.1 pH = [1.0, 1.1, 1.2…]</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.sequence_molar_extinction_280">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">sequence_molar_extinction_280</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_molar_extinction_280"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molar extinction coefficient of the sequence at 280 nm.</p>
<p class="rubric">Notes</p>
<p>Units = M/cm</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.analyse_protein.sequence_molecular_weight">
<span class="sig-prename descclassname"><span class="pre">ampal.analyse_protein.</span></span><span class="sig-name descname"><span class="pre">sequence_molecular_weight</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/analyse_protein.html#sequence_molecular_weight"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.analyse_protein.sequence_molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the polypeptide sequence.</p>
<p class="rubric">Notes</p>
<p>Units = Daltons</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>seq</strong> (<em>str</em>) – Sequence of amino acids.</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.assembly">
<span id="ampal-assembly-module"></span><h2>ampal.assembly module<a class="headerlink" href="#module-ampal.assembly" title="Permalink to this heading">¶</a></h2>
<p>Defines various containers for AMPAL objects.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.assembly.</span></span><span class="sig-name descname"><span class="pre">AmpalContainer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ampal_objects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Custom list type class that holds multiple model states.</p>
<p class="rubric">Notes</p>
<p>In this case, a state is defined as a set of coordinates that
represents a protein model and an associated score or set of scores.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ampal_objects</strong> (<em>[</em><em>AMPAL</em><em>]</em><em>, </em><em>optional</em>) – A list of AMPAL objects with which to initialise the
AMPAL container. This can be an <cite>Assembly</cite>, <cite>Polymer</cite>
or <cite>Monomer</cite>.</p></li>
<li><p><strong>id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the AMPAL container.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.assembly.AmpalContainer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for the AMPAL container.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer.append">
<span class="sig-name descname"><span class="pre">append</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">item</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.append"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds an AMPAL object to the <cite>AmpalContainer</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer.extend">
<span class="sig-name descname"><span class="pre">extend</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ampal_container</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.extend"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends an <cite>AmpalContainer</cite> with another <cite>AmpalContainer</cite>.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer.pdb">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pdb</span></span><a class="headerlink" href="#ampal.assembly.AmpalContainer.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Compiles the PDB strings for each state into a single file.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.AmpalContainer.sort_by_tag">
<span class="sig-name descname"><span class="pre">sort_by_tag</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tag</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#AmpalContainer.sort_by_tag"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.AmpalContainer.sort_by_tag" title="Permalink to this definition">¶</a></dt>
<dd><p>Sorts the <cite>AmpalContainer</cite> by a tag on the component objects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>tag</strong> (<em>str</em>) – Key of tag used for sorting.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.assembly.Assembly">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.assembly.</span></span><span class="sig-name descname"><span class="pre">Assembly</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">molecules</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assembly_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">BaseAmpal</span></code></a></p>
<p>A container that holds <cite>Polymer</cite> type objects.</p>
<p class="rubric">Notes</p>
<p>Has a simple hierarchy: <cite>Assembly</cite> contains one or more <cite>Polymer</cite>,
which in turn contains one or more <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>molecules</strong> (<a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Polymer</cite> or list containing <cite>Polymer</cite> objects to be assembled.</p></li>
<li><p><strong>assembly_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Assembly</cite>. This
is used when generating a pdb file using <cite>Assembly().pdb</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>TypeError</strong> – <cite>Assembly</cite> objects can only be initialised empty, using a <cite>Polymer</cite>
    or a list of <cite>Polymers</cite>.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.append">
<span class="sig-name descname"><span class="pre">append</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">item</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.append"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds a <cite>Polymer</cite> to the <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – Raised if other is any type other than <cite>Polymer</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.backbone">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">backbone</span></span><a class="headerlink" href="#ampal.assembly.Assembly.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a new <cite>Assembly</cite> containing only the backbone atoms.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.
Sequence data is retained, but only the main chain atoms are
retained.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bb_assembly</strong> – <cite>Assembly</cite> containing only the backbone atoms of the original
<cite>Assembly</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>ampal.Protein</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.extend">
<span class="sig-name descname"><span class="pre">extend</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">assembly</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.extend"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends the <cite>Assembly</cite> with the contents of another <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – Raised if other is any type other than <cite>Assembly</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.fasta">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">fasta</span></span><a class="headerlink" href="#ampal.assembly.Assembly.fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a FASTA string for the <cite>Assembly</cite>.</p>
<p class="rubric">Notes</p>
<p>Explanation of FASTA format: <a class="reference external" href="https://en.wikipedia.org/wiki/FASTA_format">https://en.wikipedia.org/wiki/FASTA_format</a>
Recommendation that all lines of text be shorter than 80
characters is adhered to. Format of PDBID|CHAIN|SEQUENCE is
consistent with files downloaded from the PDB. Uppercase
PDBID used for consistency with files downloaded from the PDB.
Useful for feeding into cdhit and then running sequence clustering.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>fasta_str</strong> – String of the fasta file for the <cite>Assembly</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.filter_mol_types">
<span class="sig-name descname"><span class="pre">filter_mol_types</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mol_types</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.filter_mol_types"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.filter_mol_types" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.get_atoms">
<span class="sig-name descname"><span class="pre">get_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudo_group</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Flat list of all the <cite>Atoms</cite> in the <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include ligand <cite>Atoms</cite>.</p></li>
<li><p><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include pseudo_group <cite>Atoms</cite>.</p></li>
<li><p><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include alternate sidechain conformations.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>atoms</strong> – All the <cite>Atoms</cite> as a iterator.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>itertools.chain</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.get_ligands">
<span class="sig-name descname"><span class="pre">get_ligands</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">solvent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_ligands"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all ligands from the <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>solvent</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, solvent molecules will be included.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.get_monomers">
<span class="sig-name descname"><span class="pre">get_monomers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudo_group</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.get_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all the <cite>Monomers</cite> from the <cite>Assembly</cite> object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, will include ligand <cite>Monomers</cite>.</p></li>
<li><p><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include pseudo atoms.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.is_within">
<span class="sig-name descname"><span class="pre">is_within</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cutoff_dist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.is_within"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.is_within" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in AMPAL object within <cite>cut-off</cite> distance from the <cite>point</cite>.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.isoelectric_point">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">isoelectric_point</span></span><a class="headerlink" href="#ampal.assembly.Assembly.isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the isoelectric point of the <cite>Assembly</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.make_pdb">
<span class="sig-name descname"><span class="pre">make_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pseudo_group</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">header</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">footer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.make_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the Assembly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include ligands in the output.</p></li>
<li><p><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include alternate conformations in the output.</p></li>
<li><p><strong>pseudo_group</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will include pseudo atoms in the output.</p></li>
<li><p><strong>header</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> will write a header for output.</p></li>
<li><p><strong>footer</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> will write a footer for output.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pdb_str</strong> – String of the pdb for the Assembly. Generated by collating
Polymer().pdb calls for the component Polymers.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.molar_extinction_280">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">molar_extinction_280</span></span><a class="headerlink" href="#ampal.assembly.Assembly.molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the extinction co-efficient of the <cite>Assembly</cite> at 280 nm.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.molecular_weight">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">molecular_weight</span></span><a class="headerlink" href="#ampal.assembly.Assembly.molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the <cite>Assembly</cite> in Daltons.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.pdb">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pdb</span></span><a class="headerlink" href="#ampal.assembly.Assembly.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs make_pdb in default mode.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.primitives">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">primitives</span></span><a class="headerlink" href="#ampal.assembly.Assembly.primitives" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a new <cite>Assembly</cite> containing the primitives of each Polymer.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>prim_assembly</strong> – <cite>Assembly</cite> containing only the primitives of the <cite>Polymers</cite>
in the original <cite>Assembly</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>ampal.Protein</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.relabel_all">
<span class="sig-name descname"><span class="pre">relabel_all</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Polymers, Monomers and Atoms with default labeling.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.relabel_atoms">
<span class="sig-name descname"><span class="pre">relabel_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Atoms in numerical order, offset by the start parameter.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>start</strong> (<em>int</em><em>, </em><em>optional</em>) – Defines an offset for the labelling.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.relabel_monomers">
<span class="sig-name descname"><span class="pre">relabel_monomers</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all Monomers in the component Polymers in numerical order.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.relabel_polymers">
<span class="sig-name descname"><span class="pre">relabel_polymers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.relabel_polymers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.relabel_polymers" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Relabels the component Polymers either in alphabetical order or</dt><dd><p>using a list of labels.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>labels</strong> (<em>list</em><em>, </em><em>optional</em>) – A list of new labels.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – Raised if the number of labels does not match the number of
    component Polymer objects.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.sequences">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">sequences</span></span><a class="headerlink" href="#ampal.assembly.Assembly.sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of each <cite>Polymer</cite> in the <cite>Assembly</cite> as a list.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>sequences</strong> – List of sequences.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.tag_ca_geometry">
<span class="sig-name descname"><span class="pre">tag_ca_geometry</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ref_axis'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.tag_ca_geometry"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.tag_ca_geometry" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Monomer</cite> in the <cite>Assembly</cite> with its helical geometry.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True the tag will be run even if <cite>Monomers</cite> are already tagged.</p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em><em>, </em><em>optional</em>) – Coordinates to feed to geometry functions that depend on
having a reference axis.</p></li>
<li><p><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at <cite>Chain</cite> and <cite>Residue</cite> level.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.assembly.Assembly.tag_torsion_angles">
<span class="sig-name descname"><span class="pre">tag_torsion_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/assembly.html#Assembly.tag_torsion_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.assembly.Assembly.tag_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Monomer</cite> in the <cite>Assembly</cite> with its torsion angles.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, the tag will be run even if <cite>Monomers</cite> are already
tagged.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>
<section id="module-ampal.base_ampal">
<span id="ampal-base-ampal-module"></span><h2>ampal.base_ampal module<a class="headerlink" href="#module-ampal.base_ampal" title="Permalink to this heading">¶</a></h2>
<p>Contains the base and common classes for all AMPAL objects.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">Atom</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coordinates</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">element</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">res_label</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">occupancy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bfactor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">charge</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">state</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'A'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Object containing atomic coordinates and element information.</p>
<p class="rubric">Notes</p>
<p><cite>Atom</cite> is an AMPAL object, but it does not inherit from <cite>BaseAmpal</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>coordinates</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Position of <cite>Atom</cite> in 3D space.</p></li>
<li><p><strong>element</strong> (<em>str</em>) – Element of <cite>Atom</cite>.</p></li>
<li><p><strong>atom_id</strong> (<em>str</em>) – Identifier for <cite>Atom</cite>, usually a number.</p></li>
<li><p><strong>res_label</strong> (<em>str</em><em>, </em><em>optional</em>) – Label used in <cite>Monomer</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</p></li>
<li><p><strong>occupancy</strong> (<em>float</em><em>, </em><em>optional</em>) – The occupancy of the <cite>Atom</cite>.</p></li>
<li><p><strong>bfactor</strong> (<em>float</em><em>, </em><em>optional</em>) – The bfactor of the <cite>Atom</cite>.</p></li>
<li><p><strong>charge</strong> (<em>str</em><em>, </em><em>optional</em>) – The point charge of the <cite>Atom</cite>.</p></li>
<li><p><strong>state</strong> (<em>str</em>) – The state of this <cite>Atom</cite>. Used to identify <cite>Atoms</cite> with a
number of conformations.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Monomer</em><em>, </em><em>optional</em>) – A reference to the <cite>Monomer</cite> containing this <cite>Atom</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for <cite>Atom</cite>, usually a number.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.res_label">
<span class="sig-name descname"><span class="pre">res_label</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.res_label" title="Permalink to this definition">¶</a></dt>
<dd><p>Label used in <cite>Monomer</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.element">
<span class="sig-name descname"><span class="pre">element</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.element" title="Permalink to this definition">¶</a></dt>
<dd><p>Element of <cite>Atom</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.parent" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>A reference to the <cite>Monomer</cite> containing this <cite>Atom</cite>.</dt><dd><p>number of conformations.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Monomer</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.array">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">array</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.array" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts the atomic coordinate to a <cite>numpy.array</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.rotate">
<span class="sig-name descname"><span class="pre">rotate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom.rotate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Atom.rotate" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotates <cite>Atom</cite> by <cite>angle</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>angle</strong> (<em>float</em>) – Angle that <cite>Atom</cite> will be rotated.</p></li>
<li><p><strong>axis</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Axis about which the <cite>Atom</cite> will be rotated.</p></li>
<li><p><strong>point</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em><em>, </em><em>optional</em>) – Point that the <cite>axis</cite> lies upon. If <cite>None</cite> then the origin is used.</p></li>
<li><p><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – True is <cite>angle</cite> is define in radians, False is degrees.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.translate">
<span class="sig-name descname"><span class="pre">translate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vector</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Atom.translate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Atom.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates <cite>Atom</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>vector</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Vector used for translation.</p></li>
<li><p><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.unique_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">unique_id</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.unique_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a unique ID for the <cite>Atom</cite> based on its parents.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>unique_id</strong> – (polymer.id, residue.id, atom.id)</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>(str, str, str)</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.x">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">x</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.x" title="Permalink to this definition">¶</a></dt>
<dd><p>The x coordinate.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.y">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">y</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.y" title="Permalink to this definition">¶</a></dt>
<dd><p>The y coordinate.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Atom.z">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">z</span></span><a class="headerlink" href="#ampal.base_ampal.Atom.z" title="Permalink to this definition">¶</a></dt>
<dd><p>The z coordinate.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">BaseAmpal</span></span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Base class for all AMPAL objects except <cite>ampal.atom</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>NotImplementedError</strong> – <cite>BaseAmpal</cite> is an abstract base class and is not intended to
    be instanciated. A <cite>NotImplementedError</cite> is raised if a
    method is called that is required on a child class but is
    not implemented in <cite>BaseAmpal</cite>.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.centre_of_mass">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">centre_of_mass</span></span><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.centre_of_mass" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the centre of mass of AMPAL object.</p>
<p class="rubric">Notes</p>
<p>All atoms are included in calculation, call <cite>centre_of_mass</cite>
manually if another selection is require.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>centre_of_mass</strong> – 3D coordinate for the centre of mass.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.get_atoms">
<span class="sig-name descname"><span class="pre">get_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.get_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.is_within">
<span class="sig-name descname"><span class="pre">is_within</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cutoff_dist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.is_within"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.is_within" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in ampal object within <cite>cut-off</cite> distance from the <cite>point</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.make_pdb">
<span class="sig-name descname"><span class="pre">make_pdb</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.make_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.pdb">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pdb</span></span><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs make_pdb in default mode.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.rmsd">
<span class="sig-name descname"><span class="pre">rmsd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">other</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">backbone</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.rmsd"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.rmsd" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the RMSD between two AMPAL objects.</p>
<p class="rubric">Notes</p>
<p>No fitting operation is performs and both AMPAL objects must
have the same number of atoms.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>other</strong> (<em>AMPAL Object</em>) – Any AMPAL object with <cite>get_atoms</cite> method.</p></li>
<li><p><strong>backbone</strong> (<em>bool</em><em>, </em><em>optional</em>) – Calculates RMSD of backbone only.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.rotate">
<span class="sig-name descname"><span class="pre">rotate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.rotate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.rotate" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotates every atom in the AMPAL object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>angle</strong> (<em>float</em>) – Angle that AMPAL object will be rotated.</p></li>
<li><p><strong>axis</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Axis about which the AMPAL object will be rotated.</p></li>
<li><p><strong>point</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em><em>, </em><em>optional</em>) – Point that the axis lies upon. If <cite>None</cite> then the origin is used.</p></li>
<li><p><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – True is <cite>angle</cite> is define in radians, False is degrees.</p></li>
<li><p><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.BaseAmpal.translate">
<span class="sig-name descname"><span class="pre">translate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">vector</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#BaseAmpal.translate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.BaseAmpal.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates every atom in the AMPAL object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>vector</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Vector used for translation.</p></li>
<li><p><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will rotate atoms in all states i.e. includes
alternate conformations for sidechains.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">Monomer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">BaseAmpal</span></code></a></p>
<p>Groups of <cite>Atoms</cite> that form <cite>Polymers</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>OrderedDict</em><em> or </em><em>{OrderedDict}</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the dictionary.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</p></li>
<li><p><strong>parent</strong> (<a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><em>Polymer</em></a><em>, </em><em>optional</em>) – A reference to the <cite>Polymer</cite> containing this <cite>Monomer</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.states">
<span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polymer</cite> containing this <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.active_state">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">active_state</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.active_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Defines which state dictionary should be used currently.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.atoms">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">atoms</span></span><a class="headerlink" href="#ampal.base_ampal.Monomer.atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Atoms in the currently active state.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.close_monomers">
<span class="sig-name descname"><span class="pre">close_monomers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">group</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.close_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.close_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of Monomers from within a cut off distance of the Monomer</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>group</strong> (<a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><em>BaseAmpal</em></a><em> or </em><em>Subclass</em>) – Group to be search for Monomers that are close to this Monomer.</p></li>
<li><p><strong>cutoff</strong> (<em>float</em>) – Distance cut off.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>nearby_residues</strong> – List of Monomers within cut off distance.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[Monomers]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.get_atoms">
<span class="sig-name descname"><span class="pre">get_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.get_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all atoms in the <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>inc_alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will return <cite>Atoms</cite> for alternate states.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.get_monomers">
<span class="sig-name descname"><span class="pre">get_monomers</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.get_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the this <cite>Monomer</cite>.</p>
<p class="rubric">Notes</p>
<p>This function is only present for consistency in the interface.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Monomer.make_pdb">
<span class="sig-name descname"><span class="pre">make_pdb</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Monomer.make_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Monomer.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>Monomer</cite>.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">Polymer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">molecule_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.BaseAmpal" title="ampal.base_ampal.BaseAmpal"><code class="xref py py-class docutils literal notranslate"><span class="pre">BaseAmpal</span></code></a></p>
<p>A container that holds <cite>Monomer</cite> type objects.</p>
<p class="rubric">Notes</p>
<p><cite>Polymer</cite> has a simple hierarchy: A <cite>Polymer</cite> contains one or
more <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomers</strong> (<a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em>]</em><em>, </em><em>optional</em>) – Monomer or list containing Monomer objects to form the Polymer().</p></li>
<li><p><strong>ligands</strong> (<a class="reference internal" href="#ampal.ligands.LigandGroup" title="ampal.ligands.LigandGroup"><em>LigandGroup</em></a><em>, </em><em>optional</em>) – <cite>Ligands</cite> associated with the <cite>Polymer</cite>.</p></li>
<li><p><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polymer</cite>. This is
used when generating a pdb file using <cite>Polymer().pdb</cite>.</p></li>
<li><p><strong>molecule_type</strong> (<em>str</em><em>, </em><em>optional</em>) – A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</p></li>
<li><p><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Polymer ID</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Assembly or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.molecule_type">
<span class="sig-name descname"><span class="pre">molecule_type</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.ligands">
<span class="sig-name descname"><span class="pre">ligands</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polymer</cite> chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.LigandGroup</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.sl">
<span class="sig-name descname"><span class="pre">sl</span></span><a class="headerlink" href="#ampal.base_ampal.Polymer.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – Polymer objects can only be initialised empty, using a Monomer
    or a list of Monomers.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.append">
<span class="sig-name descname"><span class="pre">append</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">item</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.append"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Appends a <cite>Monomer to the `Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>Does not update labelling.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.extend">
<span class="sig-name descname"><span class="pre">extend</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">polymer</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.extend"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Extends the <cite>Polymer</cite> with the contents of another <cite>Polymer</cite>.</p>
<p class="rubric">Notes</p>
<p>Does not update labelling.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.get_atoms">
<span class="sig-name descname"><span class="pre">get_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Flat list of all the Atoms in the Polymer.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>inc_alt_states</strong> (<em>bool</em>) – If true atoms from alternate conformations are included rather
than only the “active” states.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>atoms</strong> – Returns an iterator of all the atoms. Convert to list if you
require indexing.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>itertools.chain</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.get_monomers">
<span class="sig-name descname"><span class="pre">get_monomers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves all the <cite>Monomers</cite> from the AMPAL object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, will include ligand <cite>Monomers</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.get_reference_coords">
<span class="sig-name descname"><span class="pre">get_reference_coords</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.get_reference_coords"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.get_reference_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets list of coordinates of all reference atoms in the <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ref_coords</strong> – List has the same length as the <cite>Polymer</cite>.
The first, second and third elements of array i contain the
x, y and z coordinates of the i-th reference atom.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>[numpy.array]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.make_pdb">
<span class="sig-name descname"><span class="pre">make_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inc_ligands</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.make_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.make_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – Include alternate conformations for <cite>Monomers</cite> in PDB.</p></li>
<li><p><strong>inc_ligands</strong> (<em>bool</em><em>, </em><em>optional</em>) – Includes <cite>Ligands</cite> in PDB.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pdb_str</strong> – String of the pdb for the <cite>Polymer</cite>. Generated using information
from the component <cite>Monomers</cite>.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.relabel_all">
<span class="sig-name descname"><span class="pre">relabel_all</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all <cite>Monomers</cite> and <cite>Atoms</cite> using default labeling.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.relabel_atoms">
<span class="sig-name descname"><span class="pre">relabel_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels all <cite>Atoms</cite> in numerical order.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>start</strong> (<em>int</em><em>, </em><em>optional</em>) – Offset the labelling by <cite>start</cite> residues.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.base_ampal.Polymer.relabel_monomers">
<span class="sig-name descname"><span class="pre">relabel_monomers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#Polymer.relabel_monomers"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.Polymer.relabel_monomers" title="Permalink to this definition">¶</a></dt>
<dd><p>Relabels the either in numerically or using a list of labels.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>labels</strong> (<em>list</em><em>, </em><em>optional</em>) – A list of new labels.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – Raised if the number of labels does not match the number of
    component Monoer objects.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.base_ampal.cap">
<span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">cap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">v</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">l</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#cap"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.cap" title="Permalink to this definition">¶</a></dt>
<dd><p>Shortens string is above certain length.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.base_ampal.centre_of_atoms">
<span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">centre_of_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mass_weighted</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#centre_of_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.centre_of_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns centre point of any list of atoms.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>list</em>) – List of AMPAL atom objects.</p></li>
<li><p><strong>mass_weighted</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True returns centre of mass, otherwise just geometric centre of points.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>centre_of_mass</strong> – 3D coordinate for the centre of mass.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.base_ampal.find_atoms_within_distance">
<span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">find_atoms_within_distance</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutoff_distance</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#find_atoms_within_distance"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.find_atoms_within_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns atoms within the distance from the point.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>[</em><em>ampal.atom</em><em>]</em>) – A list of <cite>ampal.atoms</cite>.</p></li>
<li><p><strong>cutoff_distance</strong> (<em>float</em>) – Maximum distance from point.</p></li>
<li><p><strong>point</strong> (<em>(</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>)</em>) – Reference point, 3D coordinate.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>filtered_atoms</strong> – <cite>atoms</cite> list filtered by distance.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[ampal.atoms]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.base_ampal.write_pdb">
<span class="sig-prename descclassname"><span class="pre">ampal.base_ampal.</span></span><span class="sig-name descname"><span class="pre">write_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residues</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alt_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strip_states</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/base_ampal.html#write_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.base_ampal.write_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes a pdb file for a list of residues.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>residues</strong> (<em>list</em>) – List of Residue objects.</p></li>
<li><p><strong>chain_id</strong> (<em>str</em>) – String of the chain id, defaults to ‘ ‘.</p></li>
<li><p><strong>alt_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, include all occupancy states of residues, else outputs primary state only.</p></li>
<li><p><strong>strip_states</strong> (<em>bool</em><em>, </em><em>optional</em>) – If true, remove all state labels from residues. Only use with alt_states false.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pdb_str</strong> – String of the PDB file.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.data">
<span id="ampal-data-module"></span><h2>ampal.data module<a class="headerlink" href="#module-ampal.data" title="Permalink to this heading">¶</a></h2>
<p>Loads data required for running ISAMBARD.</p>
</section>
<section id="module-ampal.dssp">
<span id="ampal-dssp-module"></span><h2>ampal.dssp module<a class="headerlink" href="#module-ampal.dssp" title="Permalink to this heading">¶</a></h2>
<p>This module provides an interface to the program DSSP.</p>
<p>For more information on DSSP see <a class="footnote-reference brackets" href="#id6" id="id5" role="doc-noteref"><span class="fn-bracket">[</span>4<span class="fn-bracket">]</span></a>.</p>
<p class="rubric">References</p>
<aside class="footnote-list brackets">
<aside class="footnote brackets" id="id6" role="doc-footnote">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id5">4</a><span class="fn-bracket">]</span></span>
<p>Kabsch W, Sander C (1983) “Dictionary of protein
secondary structure: pattern recognition of hydrogen-bonded
and geometrical features”, Biopolymers, 22, 2577-637.</p>
</aside>
</aside>
<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.dssp_available">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">dssp_available</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#dssp_available"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.dssp_available" title="Permalink to this definition">¶</a></dt>
<dd><p>True if mkdssp is available on the path.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.extract_all_ss_dssp">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">extract_all_ss_dssp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">in_dssp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#extract_all_ss_dssp"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.extract_all_ss_dssp" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses DSSP to extract secondary structure information on every residue.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>in_dssp</strong> (<em>str</em>) – Path to DSSP file.</p></li>
<li><p><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – Indicates if pdb is a path or a string.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><p><strong>dssp_residues</strong> –</p>
<dl class="simple">
<dt>Each internal list contains:</dt><dd><p>[0] int Residue number
[1] str Secondary structure type
[2] str Chain identifier
[3] str Residue type
[4] float Phi torsion angle
[5] float Psi torsion angle
[6] int dssp solvent accessibility</p>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[tuple]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.find_ss_regions">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">find_ss_regions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dssp_residues</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loop_assignments</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">('</span> <span class="pre">',</span> <span class="pre">'B',</span> <span class="pre">'S',</span> <span class="pre">'T')</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#find_ss_regions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.find_ss_regions" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates parsed DSSP data into groups of secondary structure.</p>
<p class="rubric">Notes</p>
<p>Example: all residues in a single helix/loop/strand will be gathered
into a list, then the next secondary structure element will be
gathered into a separate list, and so on.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dssp_residues</strong> (<em>[</em><em>tuple</em><em>]</em>) – <dl class="simple">
<dt>Each internal list contains:</dt><dd><p>[0] int Residue number
[1] str Secondary structure type
[2] str Chain identifier
[3] str Residue type
[4] float Phi torsion angle
[5] float Psi torsion angle
[6] int dssp solvent accessibility</p>
</dd>
</dl>
</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>fragments</strong> – Lists grouped in continuous regions of secondary structure.
Innermost list has the same format as above.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[[list]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.get_ss_regions">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">get_ss_regions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">assembly</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ss_types</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#get_ss_regions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.get_ss_regions" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an Assembly containing Polymers for each region of structure.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>assembly</strong> (<em>ampal.Assembly</em>) – <cite>Assembly</cite> object to be searched secondary structure regions.</p></li>
<li><p><strong>ss_types</strong> (<em>list</em>) – List of secondary structure tags to be separate i.e. [‘H’]
would return helices, [‘H’, ‘E’] would return helices
and strands.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>fragments</strong> – <cite>Assembly</cite> containing a <cite>Polymer</cite> for each region of specified
secondary structure.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.assembly.Assembly" title="ampal.assembly.Assembly">Assembly</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.run_dssp">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">run_dssp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#run_dssp"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.run_dssp" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses DSSP to find helices and extracts helices from a pdb file or string.
:param pdb: Path to pdb file or string.
:type pdb: str
:param path: Indicates if pdb is a path or a string.
:type path: bool, optional</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>dssp_out</strong> – Std out from DSSP.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.dssp.tag_dssp_data">
<span class="sig-prename descclassname"><span class="pre">ampal.dssp.</span></span><span class="sig-name descname"><span class="pre">tag_dssp_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">assembly</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loop_assignments</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">('</span> <span class="pre">',</span> <span class="pre">'B',</span> <span class="pre">'S',</span> <span class="pre">'T')</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/dssp.html#tag_dssp_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.dssp.tag_dssp_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds output data from DSSP to an Assembly.</p>
<p>A dictionary will be added to the <cite>tags</cite> dictionary of each
residue called <cite>dssp_data</cite>, which contains the secondary
structure definition, solvent accessibility phi and psi values
from DSSP. A list of regions of continuous secondary assignments
will also be added to each <cite>Polypeptide</cite>.</p>
<p>The tags are added in place, so nothing is returned from this
function.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>assembly</strong> (<em>ampal.Assembly</em>) – An Assembly containing some protein.</p></li>
<li><p><strong>loop_assignments</strong> (<em>tuple</em><em> or </em><em>list</em>) – A tuple containing the DSSP secondary structure identifiers to
that are classed as loop regions.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.geometry">
<span id="ampal-geometry-module"></span><h2>ampal.geometry module<a class="headerlink" href="#module-ampal.geometry" title="Permalink to this heading">¶</a></h2>
<p>Basic geometric operations that AMPAL requires.</p>
<p>Functions that have names preceded by an underscore indicate that they are not
to be used by users. However, all of these functions will generally be defined
using cdef and so will not be imported anyway.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.geometry.Axis">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">Axis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.geometry.Curve" title="ampal.geometry.Curve"><code class="xref py py-class docutils literal notranslate"><span class="pre">Curve</span></code></a></p>
<p>Axis: a straight line curve in 3-D space, defined by its start and end points (3-D vectors).</p>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Axis.arc_length">
<span class="sig-name descname"><span class="pre">arc_length</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis.arc_length" title="Permalink to this definition">¶</a></dt>
<dd><p>arc length between point t and point t0</p>
<p class="rubric">Notes</p>
<p>This is a trivial function in this particular case. Implemented for compatability with Curve class.
Returns arc length along the curve between the points specified by t and t_ref.
If t_ref &gt; t then the returned value is negative.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>t</strong> (<em>float</em>) – </p></li>
<li><p><strong>t_ref</strong> (<em>float</em>) – t and t_ref define points on curve (See self.point()).</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.direction_vector">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">direction_vector</span></span><a class="headerlink" href="#ampal.geometry.Axis.direction_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Axis.distance_to_point">
<span class="sig-name descname"><span class="pre">distance_to_point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">point</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis.distance_to_point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Axis.get_coords">
<span class="sig-name descname"><span class="pre">get_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_points</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spacing</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.52</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis.get_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>List of evenly spaced points along the curve of specified total length.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n_points</strong> (<em>int</em>) – The number of points to be returned.</p></li>
<li><p><strong>spacing</strong> (<em>float</em>) – arc-length spacing of points to be returned.</p></li>
<li><p><strong>t_ref</strong> (<em>The value</em><em> of </em><em>t for specifying the first point in the coordinates list.</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>coords</strong> – 3D coordinates of evenly spaced points along the curve.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of numpy.arrays</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.length">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">length</span></span><a class="headerlink" href="#ampal.geometry.Axis.length" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.midpoint">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">midpoint</span></span><a class="headerlink" href="#ampal.geometry.Axis.midpoint" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Axis.point">
<span class="sig-name descname"><span class="pre">point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Axis.t_from_arc_length">
<span class="sig-name descname"><span class="pre">t_from_arc_length</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">arc_length</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Axis.t_from_arc_length" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Returns parameter t that corresponds to a point at distance=arc_length along the curve from the curve</dt><dd><p>at t_ref.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This is a trivial function in this particular case. Implemented for compatability with Curve class.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>arc_length</strong> (<em>float</em>) – distance along curve</p></li>
<li><p><strong>t_ref</strong> (<em>float</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Value of parameter t that corresponds to a point at distance=arc_length along the curve from the point
defined by parameter t_ref.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.unit_binormal">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">unit_binormal</span></span><a class="headerlink" href="#ampal.geometry.Axis.unit_binormal" title="Permalink to this definition">¶</a></dt>
<dd><p>Binormal defined as (Tangent X Normal).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.unit_normal">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">unit_normal</span></span><a class="headerlink" href="#ampal.geometry.Axis.unit_normal" title="Permalink to this definition">¶</a></dt>
<dd><p>Normal is perpendicular to tangent. See notes.</p>
<p class="rubric">Notes</p>
<p>Since the curvature of a straight line is zero, the normal and binormal are not well-defined.
<a class="reference external" href="https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures">https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures</a>
Here we have chosen to use the following convention:</p>
<blockquote>
<div><dl class="simple">
<dt>If the Axis is parallel to the z-axis (i.e. unit_tangent = [0, 0, 1]),</dt><dd><p>then the normal is [1, 0, 0] (x-axis) and the binormal is [0, 1, 0] (y-axis).</p>
</dd>
</dl>
<p>If the Axis is not parallel to the z-axis, then we can rotate the z-axis so it aligns with our Axis.
The normal is the same rotation appled to the x-axis.
It’s helpful here to think of the right-hand rule.
<a class="reference external" href="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/Right_hand_rule_cross_product.svg/2000px-Right_hand_rule_cross_product.svg.png">https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/Right_hand_rule_cross_product.svg/2000px-Right_hand_rule_cross_product.svg.png</a>
<a class="reference external" href="https://en.wikipedia.org/wiki/Cross_product#Definition">https://en.wikipedia.org/wiki/Cross_product#Definition</a>
Using the right-hand rule, the Axis direction vector is your index finger,
the normal vector is the middle finger and the binormal is the thumb.</p>
</div></blockquote>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Axis.unit_tangent">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">unit_tangent</span></span><a class="headerlink" href="#ampal.geometry.Axis.unit_tangent" title="Permalink to this definition">¶</a></dt>
<dd><p>Tangent points along the direction of the line.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.geometry.Curve">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">Curve</span></span><a class="headerlink" href="#ampal.geometry.Curve" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Not intended to be instantiated. This is a base class for other Curve classes to inherit from.</p>
<p class="rubric">Notes</p>
<p>Any curve must have a point method, which returns the coordinates of a point, given a parameter t.</p>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Curve.point">
<span class="sig-name descname"><span class="pre">point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Curve.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">HelicalCurve</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">handedness</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.geometry.Curve" title="ampal.geometry.Curve"><code class="xref py py-class docutils literal notranslate"><span class="pre">Curve</span></code></a></p>
<p>HelicalCurve, parameterised by a and b.</p>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.alpha">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">alpha</span></span><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between tangent to helix and the helix axis, in degrees.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.alpha_and_pitch">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">alpha_and_pitch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alpha</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pitch</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">handedness</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha_and_pitch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.alpha_and_radius">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">alpha_and_radius</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alpha</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">handedness</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.alpha_and_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.arc_length">
<span class="sig-name descname"><span class="pre">arc_length</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.arc_length" title="Permalink to this definition">¶</a></dt>
<dd><p>arc length between point t and point t0</p>
<p class="rubric">Notes</p>
<p>Returns arc length along the curve between the points specified by t and t_ref.
If t_ref &gt; t then the returned value is negative.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>t</strong> (<em>float</em>) – </p></li>
<li><p><strong>t_ref</strong> (<em>float</em>) – t and t_ref define points on curve (See self.point()).</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.axis">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">axis</span></span><a class="headerlink" href="#ampal.geometry.HelicalCurve.axis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.curvature">
<span class="sig-name descname"><span class="pre">curvature</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.curvature" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.get_coords">
<span class="sig-name descname"><span class="pre">get_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_points</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spacing</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.52</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.get_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>List of evenly spaced points along the curve of specified total length.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n_points</strong> (<em>int</em>) – The number of points to be returned.</p></li>
<li><p><strong>spacing</strong> (<em>float</em>) – arc-length spacing of points to be returned.</p></li>
<li><p><strong>axis_start</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em> of </em><em>length 3</em>) – </p></li>
<li><p><strong>axis_end</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em> of </em><em>length 3</em>) – Coordinates returned will be for a helical path winding around an axis specified by axis_end - axis_start.</p></li>
<li><p><strong>t_ref</strong> (<em>The value</em><em> of </em><em>t for specifying the first point in the coordinates list.</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>coords</strong> – 3D coordinates of evenly spaced points along the curve.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list of numpy.arrays</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.pitch">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pitch</span></span><a class="headerlink" href="#ampal.geometry.HelicalCurve.pitch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.pitch_and_radius">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pitch_and_radius</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pitch</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">handedness</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'r'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t0</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.pitch_and_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.point">
<span class="sig-name descname"><span class="pre">point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.point" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.radius">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">radius</span></span><a class="headerlink" href="#ampal.geometry.HelicalCurve.radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.t_from_arc_length">
<span class="sig-name descname"><span class="pre">t_from_arc_length</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">arc_length</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t_ref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.t_from_arc_length" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Returns parameter t that corresponds to a point at distance=arc_length along the curve from the curve</dt><dd><p>at t_ref.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>arc_length</strong> (<em>float</em>) – distance along curve</p></li>
<li><p><strong>t_ref</strong> (<em>float</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Value of parameter t that corresponds to a point at distance=arc_length along the curve from the point
defined by parameter t_ref.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.torsion">
<span class="sig-name descname"><span class="pre">torsion</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.torsion" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.unit_binormal">
<span class="sig-name descname"><span class="pre">unit_binormal</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_binormal" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.unit_normal">
<span class="sig-name descname"><span class="pre">unit_normal</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_normal" title="Permalink to this definition">¶</a></dt>
<dd><p>Unit vector normal to the curve. Together with unit_tangent, defined the osculating plane at t</p>
<p class="rubric">Notes</p>
<p>See this article for more information on the tangent, normal and binormal vectors.
<a class="reference external" href="https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures">https://en.wikipedia.org/wiki/Differential_geometry_of_curves#Special_Frenet_vectors_and_generalized_curvatures</a></p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.unit_tangent">
<span class="sig-name descname"><span class="pre">unit_tangent</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.unit_tangent" title="Permalink to this definition">¶</a></dt>
<dd><p>Unit vector pointing along the direction of the curve.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.x">
<span class="sig-name descname"><span class="pre">x</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.x" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.y">
<span class="sig-name descname"><span class="pre">y</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.y" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.HelicalCurve.z">
<span class="sig-name descname"><span class="pre">z</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.HelicalCurve.z" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">Quaternion</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">i</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">j</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_rotation</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.angle_and_axis">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">angle_and_axis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axis</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.angle_and_axis" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.apply_conjugation">
<span class="sig-name descname"><span class="pre">apply_conjugation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">other</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.apply_conjugation" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.as_array">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">as_array</span></span><a class="headerlink" href="#ampal.geometry.Quaternion.as_array" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.as_rotation">
<span class="sig-name descname"><span class="pre">as_rotation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.as_rotation" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.conjugate">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">conjugate</span></span><a class="headerlink" href="#ampal.geometry.Quaternion.conjugate" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.distance">
<span class="sig-name descname"><span class="pre">distance</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">other</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.distance" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.extract_angle_and_axis">
<span class="sig-name descname"><span class="pre">extract_angle_and_axis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">radians</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.extract_angle_and_axis" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the angle and axis of the rotation that the Quaternion represents.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.norm">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">norm</span></span><a class="headerlink" href="#ampal.geometry.Quaternion.norm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.real_and_vector">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">real_and_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vector</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_rotation</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.real_and_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.rotate_vector">
<span class="sig-name descname"><span class="pre">rotate_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">v</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">point</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.Quaternion.rotate_vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.geometry.Quaternion.vector">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">vector</span></span><a class="headerlink" href="#ampal.geometry.Quaternion.vector" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.angle_between_vectors">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">angle_between_vectors</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.angle_between_vectors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate angle between two vectors using the dot product.</p>
<p class="rubric">Notes</p>
<p>All angle-based functions in tools_geometry will return values in degrees by default.
The radians flag can be used to return values in radians.
If a or b is very close to the origin (norm &lt; 1e-7), returns 0.0.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array.</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array.</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, returns angle in radians.
If False, returns angle in degrees.
Default value is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>angle</strong> – The angle between vectors v1 and v2.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>double</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.apply_unit_quaternion">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">apply_unit_quaternion</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.apply_unit_quaternion" title="Permalink to this definition">¶</a></dt>
<dd><p>Applies the rotation defined by the unit quaternion to a 3D vector.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>vector</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate.</p></li>
<li><p><strong>u_in</strong> (<em>double</em><em>[</em><em>3</em><em>]</em>) – Imaginary component of the unit quaternion.</p></li>
<li><p><strong>s</strong> (<em>double</em>) – Real component of the quaternion.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>r</strong> – 3D coordinate.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>double[3]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.cartesian_to_cylindrical">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">cartesian_to_cylindrical</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cartesian_to_cylindrical" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cartesian to cylindrical coordinates system</p>
<p class="rubric">Notes</p>
<p>See cylindrical_to_cartesian for more information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>x</strong> (<em>float</em>) – </p></li>
<li><p><strong>y</strong> (<em>float</em>) – </p></li>
<li><p><strong>z</strong> (<em>float</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em>) – True if returned azimuth is in radians instead of degrees.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>radius</strong> (<em>float</em>)</p></li>
<li><p><strong>azimuth</strong> (<em>float</em>)</p></li>
<li><p><strong>z</strong> (<em>float</em>) – Cylindrical coordinate parameters. See cylindrical_to_cartesian for more information.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.cartesian_to_spherical">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">cartesian_to_spherical</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cartesian_to_spherical" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cartesian to spherical coordinates system</p>
<p class="rubric">Notes</p>
<p>See spherical_to_cartesian for more information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>x</strong> (<em>float</em>) – </p></li>
<li><p><strong>y</strong> (<em>float</em>) – </p></li>
<li><p><strong>z</strong> (<em>float</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em>) – True if returned azimuth and zenith are in radians instead of degrees.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>radius</strong> (<em>float</em>)</p></li>
<li><p><strong>azimuth</strong> (<em>float</em>)</p></li>
<li><p><strong>zenith</strong> (<em>float</em>) – Spherical coordinate parameters. See spherical_to_cartesian for more information.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.centre_of_mass">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">centre_of_mass</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.centre_of_mass" title="Permalink to this definition">¶</a></dt>
<dd><p>Find centre of mass of list of points.</p>
<p class="rubric">Notes</p>
<p>Uniform weighting applied if masses not supplied.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>points</strong> (<em>list</em><em>(</em><em>tuple</em><em>, </em><em>list</em><em> or </em><em>numpy.array</em><em>)</em>) – </p></li>
<li><p><strong>masses</strong> (<em>list</em><em>(</em><em>float</em><em>)</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>com</strong> – Centre of mass of points.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – if len(masses) != len(points)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.closest_distance">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">closest_distance</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.closest_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Minimun distance between two lists of points in R^3.</p>
<p>Distances are calculated pairwise between points in points1 and points2.
The minimum of all these distance values is returned.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>points1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em> of </em><em>triples.</em>) – A number of points in R^3.</p></li>
<li><p><strong>points2</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em> of </em><em>triples.</em>) – A number of points in R^3.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>min_dist</strong> – The minimum distance between any point from points1 and any point from point2.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.closest_points_on_lines">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">closest_points_on_lines</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.closest_points_on_lines" title="Permalink to this definition">¶</a></dt>
<dd><p class="rubric">Notes</p>
<p>p0, p1, q0, q1 describe two lines, L1 (p0 -&gt; p1) and L2 (q0 -&gt; q1).
L1 (2) is the infinite extension of the line that passes through a (q0) and has direction p0-&gt;p1 (q0-&gt;q1).
Two points point_1, and point_2 on L1 and L2 respectively, are returned.
The distance between point_1 and point_2 is smaller than any other points on L1 and L2.
Code adapted from <a class="reference external" href="http://geomalgorithms.com/a07-_distance.html">http://geomalgorithms.com/a07-_distance.html</a>, where a more detailed explanation is available.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>p1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>q0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>q1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>segments</strong> (<em>bool</em>) – If True, the points returned must lie on line segments p0-&gt;p1 and q0-&gt;q1, rather than their infinite extensions.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>point_1</strong> (<em>numpy.array</em>)</p></li>
<li><p><strong>point_2</strong> (<em>numpy.array</em>)</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.cylindrical_to_cartesian">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">cylindrical_to_cartesian</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.cylindrical_to_cartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from cylindrical to cartesian coordinates system</p>
<p class="rubric">Notes</p>
<p>Cylindrical coordinates are here defined by a radius, an azimuth angle and a z value.
The azimuth is the angle in the x-y plane, with counter-clockwise direction being positive.
So: azimuth = 0 =&gt; lies on x-axis, azimuth = 90 =&gt; lies in y axis, zenith = 180 =&gt; lies antiparallel to x-axis.
The radius and the z value are the radius and height of the cylinder, respectively.
See <a class="reference external" href="https://en.wikipedia.org/wiki/Cylindrical_coordinate_system">https://en.wikipedia.org/wiki/Cylindrical_coordinate_system</a>
and/or <a class="reference external" href="http://mathworld.wolfram.com/CylindricalCoordinates.html">http://mathworld.wolfram.com/CylindricalCoordinates.html</a>
for more information.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>radius</strong> (<em>float</em>) – </p></li>
<li><p><strong>azimuth</strong> (<em>float</em>) – </p></li>
<li><p><strong>z</strong> (<em>float</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em>) – True if azimuth is given in radians instead of degrees.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>cartesian coordinates corresponding to the input cylindrical coordinates.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array([x, y, z])</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.dihedral">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">dihedral</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.dihedral" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the dihedral angle defined by four points in R^3.</p>
<p>The points a, b and c form a plane in R^3.
The points b, c and d form a plane in R^3.
The dihedral angle is the angle between these two planes.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://en.wikipedia.org/wiki/Dihedral_angle">http://en.wikipedia.org/wiki/Dihedral_angle</a> and <a class="reference external" href="https://en.wikipedia.org/wiki/Atan2">https://en.wikipedia.org/wiki/Atan2</a> for more information.
Dihedral between 4 points.
a, b, c, d are all 3-vectors (x,y,z) coordinates.
(a,b,c) and (b,c,d) must both be non-colinear.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>c</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>d</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, returns angle in radians.
If False, returns angle in degrees.
Default value is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>angle</strong> – The dihedral angle formed between the points a, b, c and d.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>double</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.distance">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">distance</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the Euclidian distance between two points in R^3.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>dist</strong> – Euclidean distance between a and b.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>double</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.find_foot">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">find_foot</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_foot" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the perpendicular foot of a point (p) and a line (ab).</p>
<p>The points a and b define a line in R^3.
There is a point on this line, q, that has the shortest Euclidian distance from the point p.
This point is the perpendicular foot, and is returned by this function.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html">http://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html</a> for a useful work-through of the maths.
Parameterise the line ab by t, so that a point on the line is of the form: a + (b-a)*t.
Define f(t) = [distance(t, p)]^2. Want t such that f(t) is at a minimum.
Solve f’(t) = 0 for t.
Substituting this value of t into a + (b-a)*t yields the desired point q.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Together, the points a and b define a line in R^3.</p></li>
<li><p><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – A point in R^3.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>q</strong> – The point on the line between a and b that is closest to q.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list or tuple or numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.find_foot_on_plane">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">find_foot_on_plane</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_foot_on_plane" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the perpendicular foot of a point (p) and a plane defined by points a, b and c.</p>
<p>The points a, b and c must define a plane in R^3 (i.e. they cannot be collinear).
There is a point on this plane, q, that has the shortest Euclidian distance from the point p.
This point is the perpendicular foot, and is returned by this function.</p>
<p class="rubric">Notes</p>
<dl class="simple">
<dt>This page:</dt><dd><p><a class="reference external" href="http://math.stackexchange.com/questions/723937/find-the-point-on-a-plane-3x-4y-z-1-that-is-closest-to-1-0-1">http://math.stackexchange.com/questions/723937/find-the-point-on-a-plane-3x-4y-z-1-that-is-closest-to-1-0-1</a>
gives a work-through of this kind of problem for a specific example.</p>
</dd>
</dl>
<p>The solution uses the Lagrangian multiplier: <a class="reference external" href="https://en.wikipedia.org/wiki/Lagrange_multiplier">https://en.wikipedia.org/wiki/Lagrange_multiplier</a>
We look for the minimum values of the distance function f (squared to make it easier!)</p>
<blockquote>
<div><p>f(x, y, z) = (x - p[0])^2 + (y - p[1])^2 + (z - p[2])^2</p>
</div></blockquote>
<p>subject to the constraint g(x, y, z) = 0, where g is the equation of the plane.
The Langrangian multiplier L = f(x, y, z) - lambda*(g(x, y, z)).
We calculate the partial deriviatves of this (wrt x, y, z and lambda).
Solve first for lambda gives the equation for the lagrangian_lambda value in this function.
Then substitute lambda back in to get the solution for (x, y, z) = q.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>c</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Together, the points a, b and c define a plane in R^3.</p></li>
<li><p><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – A point in R^3.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>s</strong> – The point on the plane that is closest to q.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.find_limits">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">find_limits</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_limits" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.find_transformations">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">find_transformations</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.find_transformations" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns geometrical transformations required to align the vector from s1 -&gt; e1 with the vector from s2 -&gt; e2</p>
<p class="rubric">Notes</p>
<p>Applying the transformations returned to the line from s1 -&gt; e2, will align it with the line from s2 -&gt; e2.
After transformation, s1 will have been moved onto s2.
Apply the rotation (given by angle, axis and point) first and then the translation (given by translation).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>s1</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. Start of line 1</p></li>
<li><p><strong>e1</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. End of line 1</p></li>
<li><p><strong>s2</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. Start of line 2</p></li>
<li><p><strong>e2</strong> (<em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – 3D coordinate. End of line 2</p></li>
<li><p><strong>radians</strong> (<em>bool</em>) – If True, angle will be returned in radians.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>translation</strong> (<em>numpy.array(3)</em>) – 3D vector. Translation to be applied to s1 to move it onto s2.</p></li>
<li><p><strong>angle</strong> (<em>float</em>) – angle for rotation (default: degrees).</p></li>
<li><p><strong>axis</strong> (<em>numpy.array(3)</em>) – axis about which to rotate line 1 to align it with line 2.</p></li>
<li><p><strong>point</strong> (<em>numpy.array(3)</em>) – a point lying on the rotation axis.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.gen_sectors">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">gen_sectors</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.gen_sectors" title="Permalink to this definition">¶</a></dt>
<dd><p>Separates the coordinates into overlapping sectors.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.intersection_start_and_end">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">intersection_start_and_end</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.intersection_start_and_end" title="Permalink to this definition">¶</a></dt>
<dd><p>Find start and end indices of the part of the list of points ‘subject’ that will intersect with ‘reference’</p>
<p class="rubric">Notes</p>
<p>This is a function designed to find assemblies from within bundles of helices.
The idea here is to take two lists of points, a subject and a reference. We want to find the part of the subject
(this could be all of it, or none of it) that would intersect the line joining the points in the reference, if
the line joining the points of subject was the prinicipal axis of a cylinder of infinite radius.
If subject was a list of ten points joining (0, 0 , 0) and (0, 0, 10), and reference was a list of 1000 points
joining (10, 0, 7) and (10, 0, 12) this function would return (7, 10).
If the subject and reference for the above example were swapped around, however, the function would return (0, 599).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>subject</strong> (<em>list</em><em> of </em><em>numpy.array</em>) – A list of points in R^3.</p></li>
<li><p><strong>reference</strong> (<em>list</em><em> of </em><em>numpy.array</em>) – A list of points in R^3.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>start_index</strong> (<em>int</em>) – The index of the first point in subject that is part of the intersection described above.</p></li>
<li><p><strong>end_index</strong> (<em>int</em>) – The index of the final point in subject that is part of the intersection described above.</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.is_acute">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">is_acute</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.is_acute" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if angle between two vectors is less than 90 degrees.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>b</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>True if angle between v1 and v2 is less than 90 degrees.
False otherwise.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.minimal_distance_between_lines">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">minimal_distance_between_lines</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.minimal_distance_between_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the minimal distance between two the two lines joining p0 to p1 and q0 to q1.</p>
<p>The points p0, p1, q0 and d are used to construct two lines in R^3.
The first line L1 is from p0 to p1, the second line L2 is from q0 to q1.
The minimal distance between these lines is the length of the line that
connects them and is perpendicular to both L1 and L2.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="https://en.wikipedia.org/wiki/Skew_lines#Distance">https://en.wikipedia.org/wiki/Skew_lines#Distance</a> for more information.
Also <a class="reference external" href="http://geomalgorithms.com/a07-_distance.html">http://geomalgorithms.com/a07-_distance.html</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>p1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>q0</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>q1</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>segments</strong> (<em>bool</em>) – If True, the points returned must lie on line segments p0-&gt;p1 and q0-&gt;q1, rather than their infinite extensions.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>dist</strong> – The distance between the two lines.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.points_on_a_circle">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">points_on_a_circle</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.points_on_a_circle" title="Permalink to this definition">¶</a></dt>
<dd><p>List of n uniformly distributed (x, y) coordinates on the circumference of a circle.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> (<em>int</em>) – Number of points to return.</p></li>
<li><p><strong>radius</strong> (<em>float</em>) – </p></li>
<li><p><strong>centre</strong> (<em>tuple</em><em> or </em><em>list</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>rotation</strong> (<em>float</em>) – Angle in degrees by which all points will be rotated.
rotation = 0 means that the line from the centre to the first point is parallel to the x-axis.
rotation &gt; 0 =&gt; anti-clockwise rotation</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>points</strong> – (x, y) coordinates of uniformly distributed points on the circumference of the circle</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list(2-tuples)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.radius_of_circumcircle">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">radius_of_circumcircle</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.radius_of_circumcircle" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the radius the circumcircle of a triangle, when supplied with three points (p, q, r).</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="http://www.analyzemath.com/Geometry/Circumcircle/CircumcircleRadius.html">http://www.analyzemath.com/Geometry/Circumcircle/CircumcircleRadius.html</a>
for more information.
Three (non-colinear) points define a traingle.
The circumcircle is the circle that passes though these points.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>q</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
<li><p><strong>r</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>radius</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.rmsd">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">rmsd</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.rmsd" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns average distance between points in points1 and their counterparts in points2.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>points1</strong> (<em>list</em><em> of </em><em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – </p></li>
<li><p><strong>points2</strong> (<em>list</em><em> of </em><em>[</em><em>float</em><em>, </em><em>float</em><em>, </em><em>float</em><em>]</em>) – </p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>rmsd</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
<p>:raises ValueError : if len(points1) != len(points2):</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.rotation_matrix">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">rotation_matrix</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.rotation_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Find rotation matrix for rotating about unit_vect by angle.</p>
<p class="rubric">Notes</p>
<p>See <a class="reference external" href="https://en.wikipedia.org/wiki/Rotation_matrix">https://en.wikipedia.org/wiki/Rotation_matrix</a> for more information.
In particular, the section headed ‘Rotation matrix from axis and angle’.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>angle</strong> (<em>float</em>) – Angle about which to rotate.</p></li>
<li><p><strong>unit_vect</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – Vector about which to rotate.</p></li>
<li><p><strong>point</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em><em>, </em><em>optional.</em>) – Specify a point about which to rotate.
If point is None, then the rotation will be about the origin.
Default value is None.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>m</strong> – A (4 x 4) rotation matrix.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.spherical_to_cartesian">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">spherical_to_cartesian</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.spherical_to_cartesian" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts from spherical to cartesian coordinates system</p>
<p class="rubric">Notes</p>
<p>Spherical coordinates are here defined by a radius, an azimuth angle and a zenith angle.
The zenith is the angle between the point and the z-axis.
So: zenith = 0 =&gt; lies on z-axis, zenith = 90 =&gt; lies in x-y plane, zenith = 180 =&gt; lies antiparallel to z-axis.
The azimuth is the angle in the x-y plane, with counter-clockwise direction being positive.
So: azimuth = 0 =&gt; lies on x-axis, azimuth = 90 =&gt; lies in y axis, zenith = 180 =&gt; lies antiparallel to x-axis.
See <a class="reference external" href="https://en.wikipedia.org/wiki/Spherical_coordinate_system">https://en.wikipedia.org/wiki/Spherical_coordinate_system</a>
and/or <a class="reference external" href="http://mathworld.wolfram.com/SphericalCoordinates.html">http://mathworld.wolfram.com/SphericalCoordinates.html</a>
for more information.
If zenith = 0, then x and y are both 0 regardless of azimuth value.
If azimuth = 0, then y is 0 regardless of zenith value.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>radius</strong> (<em>float</em>) – </p></li>
<li><p><strong>azimuth</strong> (<em>float</em>) – </p></li>
<li><p><strong>zenith</strong> (<em>float</em>) – </p></li>
<li><p><strong>radians</strong> (<em>bool</em>) – True if azimuth and zenith are given in radians instead of degrees.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>cartesian coordinates corresponding to the input spherical coordinates.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array([x, y, z])</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.geometry.unit_vector">
<span class="sig-prename descclassname"><span class="pre">ampal.geometry.</span></span><span class="sig-name descname"><span class="pre">unit_vector</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ampal.geometry.unit_vector" title="Permalink to this definition">¶</a></dt>
<dd><p>Return vector normalised by length.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>a</strong> (<em>list</em><em> or </em><em>tuple</em><em> or </em><em>numpy.array</em>) – The vector to be normalised.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>v_out</strong> – A numpy.array that is a unit vector.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.array</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ZeroDivisionError</strong> – If the vector has length 0.</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.interactions">
<span id="ampal-interactions-module"></span><h2>ampal.interactions module<a class="headerlink" href="#module-ampal.interactions" title="Permalink to this heading">¶</a></h2>
<p>Contains code for analysing chemical interactions in AMPAL objects.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.interactions.CovalentBond">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">CovalentBond</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#CovalentBond"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.CovalentBond" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.Interaction" title="ampal.interactions.Interaction"><code class="xref py py-class docutils literal notranslate"><span class="pre">Interaction</span></code></a></p>
<p>Defines a covalent bond.</p>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.interactions.CovalentBond.a">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">a</span></span><a class="headerlink" href="#ampal.interactions.CovalentBond.a" title="Permalink to this definition">¶</a></dt>
<dd><p>One <cite>Atom</cite> involved in the covalent bond.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.interactions.CovalentBond.b">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">b</span></span><a class="headerlink" href="#ampal.interactions.CovalentBond.b" title="Permalink to this definition">¶</a></dt>
<dd><p>One <cite>Atom</cite> involved in the covalent bond.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">HydrogenBond</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">donor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">acceptor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ang_d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ang_a</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#HydrogenBond"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.HydrogenBond" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.NonCovalentInteraction" title="ampal.interactions.NonCovalentInteraction"><code class="xref py py-class docutils literal notranslate"><span class="pre">NonCovalentInteraction</span></code></a></p>
<p>Defines a hydrogen bond in terms of a donor and an acceptor.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>donor</strong> (<em>ampal.Atom</em>) – The donor <cite>Atom</cite> in the interaction.</p></li>
<li><p><strong>acceptor</strong> (<em>ampal.Atom</em>) – The acceptor atom in the interaction.</p></li>
<li><p><strong>dist</strong> (<em>float</em>) – The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p></li>
<li><p><strong>ang_a</strong> (<em>float</em>) – Angle between the acceptor and the interaction vector.</p></li>
<li><p><strong>ang_d</strong> (<em>float</em>) – Angle between the donor and the interaction vector.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.donor">
<span class="sig-name descname"><span class="pre">donor</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.donor" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.acceptor">
<span class="sig-name descname"><span class="pre">acceptor</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.acceptor" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor atom in the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.dist">
<span class="sig-name descname"><span class="pre">dist</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.ang_a">
<span class="sig-name descname"><span class="pre">ang_a</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.ang_a" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between the acceptor and the interaction vector.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.ang_d">
<span class="sig-name descname"><span class="pre">ang_d</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.ang_d" title="Permalink to this definition">¶</a></dt>
<dd><p>Angle between the donor and the interaction vector.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.acceptor_monomer">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">acceptor_monomer</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.acceptor_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor <cite>Monomer</cite> in the interaction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.interactions.HydrogenBond.donor_monomer">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">donor_monomer</span></span><a class="headerlink" href="#ampal.interactions.HydrogenBond.donor_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Monomer</cite> in the interaction.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.interactions.Interaction">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">Interaction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#Interaction"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.Interaction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A container for all types of interaction with donor and acceptor.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> (<em>ampal.Atom</em>) – A member of a pairwise interaction.</p></li>
<li><p><strong>b</strong> (<em>ampal.Atom</em>) – A member of a pairwise interaction.</p></li>
<li><p><strong>dist</strong> (<em>float</em>) – The distance between <cite>a</cite> and <cite>b</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.Interaction.a">
<span class="sig-name descname"><span class="pre">a</span></span><a class="headerlink" href="#ampal.interactions.Interaction.a" title="Permalink to this definition">¶</a></dt>
<dd><p>A member of a pairwise interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.Interaction.b">
<span class="sig-name descname"><span class="pre">b</span></span><a class="headerlink" href="#ampal.interactions.Interaction.b" title="Permalink to this definition">¶</a></dt>
<dd><p>A member of a pairwise interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.Interaction.dist">
<span class="sig-name descname"><span class="pre">dist</span></span><a class="headerlink" href="#ampal.interactions.Interaction.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.interactions.NonCovalentInteraction">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">NonCovalentInteraction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">donor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">acceptor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#NonCovalentInteraction"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.interactions.Interaction" title="ampal.interactions.Interaction"><code class="xref py py-class docutils literal notranslate"><span class="pre">Interaction</span></code></a></p>
<p>A container for all non-covalent interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>donor</strong> (<em>ampal.Atom</em>) – The donor <cite>Atom</cite> in the interaction.</p></li>
<li><p><strong>acceptor</strong> (<em>ampal.Atom</em>) – The acceptor atom in the interaction.</p></li>
<li><p><strong>dist</strong> (<em>float</em>) – The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.NonCovalentInteraction.donor">
<span class="sig-name descname"><span class="pre">donor</span></span><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.donor" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.NonCovalentInteraction.acceptor">
<span class="sig-name descname"><span class="pre">acceptor</span></span><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.acceptor" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor atom in the interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Atom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.interactions.NonCovalentInteraction.dist">
<span class="sig-name descname"><span class="pre">dist</span></span><a class="headerlink" href="#ampal.interactions.NonCovalentInteraction.dist" title="Permalink to this definition">¶</a></dt>
<dd><p>The distance between <cite>Atom</cite> <cite>a</cite> and <cite>b</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id0">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">acceptor</span></span><a class="headerlink" href="#id0" title="Permalink to this definition">¶</a></dt>
<dd><p>The acceptor in the interaction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id7">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">donor</span></span><a class="headerlink" href="#id7" title="Permalink to this definition">¶</a></dt>
<dd><p>The donor <cite>Atom</cite> in the interaction.</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.interactions.covalent_bonds">
<span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">covalent_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#covalent_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.covalent_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all the covalent bonds in a list of <cite>Atom</cite> pairs.</p>
<p class="rubric">Notes</p>
<p>Uses information <cite>ELEMENT_DATA</cite>, which can be accessed directly
through this module i.e. <cite>isambard.ampal.interactions.ELEMENT_DATA</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> ([(<cite>Atom</cite>, <cite>Atom</cite>)]) – List of pairs of <cite>Atoms</cite>.</p></li>
<li><p><strong>threshold</strong> (<em>float</em><em>, </em><em>optional</em>) – Allows deviation from ideal covalent bond distance to be included.
For example, a value of 1.1 would allow interactions up to 10% further
from the ideal distance to be included.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.interactions.find_covalent_bonds">
<span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">find_covalent_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ampal</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_range</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tag</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#find_covalent_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.find_covalent_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds all covalent bonds in the AMPAL object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ampal</strong> (<em>AMPAL Object</em>) – Any AMPAL object with a <cite>get_atoms</cite> method.</p></li>
<li><p><strong>max_range</strong> (<em>float</em><em>, </em><em>optional</em>) – Used to define the sector size, so interactions at longer ranges
will not be found.</p></li>
<li><p><strong>threshold</strong> (<em>float</em><em>, </em><em>optional</em>) – Allows deviation from ideal covalent bond distance to be included.
For example, a value of 1.1 would allow interactions up to 10% further
from the ideal distance to be included.</p></li>
<li><p><strong>tag</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, will add the covalent bond to the tags dictionary of
each <cite>Atom</cite> involved in the interaction under the <cite>covalent_bonds</cite>
key.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.interactions.generate_bond_subgraphs_from_break">
<span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">generate_bond_subgraphs_from_break</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bond_graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#generate_bond_subgraphs_from_break"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.generate_bond_subgraphs_from_break" title="Permalink to this definition">¶</a></dt>
<dd><p>Splits the bond graph between two atoms to producing subgraphs.</p>
<p class="rubric">Notes</p>
<p>This will not work if there are cycles in the bond graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bond_graph</strong> (<em>networkx.Graph</em>) – Graph of covalent bond network</p></li>
<li><p><strong>atom1</strong> (<em>isambard.ampal.Atom</em>) – First atom in the bond.</p></li>
<li><p><strong>atom2</strong> (<em>isambard.ampal.Atom</em>) – Second atom in the bond.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>subgraphs</strong> – A list of subgraphs generated when a bond is broken in the covalent
bond network.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[networkx.Graph]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.interactions.generate_covalent_bond_graph">
<span class="sig-prename descclassname"><span class="pre">ampal.interactions.</span></span><span class="sig-name descname"><span class="pre">generate_covalent_bond_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">covalent_bonds</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/interactions.html#generate_covalent_bond_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.interactions.generate_covalent_bond_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a graph of the covalent bond network described by the interactions.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>covalent_bonds</strong> (<em>[</em><a class="reference internal" href="#ampal.interactions.CovalentBond" title="ampal.interactions.CovalentBond"><em>CovalentBond</em></a><em>]</em>) – List of <cite>CovalentBond</cite>.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>bond_graph</strong> – A graph of the covalent bond network.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>networkx.Graph</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.ligands">
<span id="ampal-ligands-module"></span><h2>ampal.ligands module<a class="headerlink" href="#module-ampal.ligands" title="Permalink to this heading">¶</a></h2>
<p>AMPAL objects that represent ligands.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.ligands.Ligand">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ligands.</span></span><span class="sig-name descname"><span class="pre">Ligand</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mol_code</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertion_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_hetero</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ligands.html#Ligand"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ligands.Ligand" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Monomer</span></code></a></p>
<p><cite>Monomer</cite> that represents a <cite>Ligand</cite>.</p>
<p class="rubric">Notes</p>
<p>All <cite>Monomers</cite> that do not have dedicated classes are
represented using the <cite>Ligand</cite> class.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mol_code</strong> (<em>str</em>) – PDB molecule code that represents the monomer.</p></li>
<li><p><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</p></li>
<li><p><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of monomer, used if reading from pdb.</p></li>
<li><p><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.atoms">
<span class="sig-name descname"><span class="pre">atoms</span></span><a class="headerlink" href="#ampal.ligands.Ligand.atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>OrderedDict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.mol_code">
<span class="sig-name descname"><span class="pre">mol_code</span></span><a class="headerlink" href="#ampal.ligands.Ligand.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Ligand</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.insertion_code">
<span class="sig-name descname"><span class="pre">insertion_code</span></span><a class="headerlink" href="#ampal.ligands.Ligand.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Ligand</cite>, used if reading from pdb.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.is_hetero">
<span class="sig-name descname"><span class="pre">is_hetero</span></span><a class="headerlink" href="#ampal.ligands.Ligand.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.self.states">
<span class="sig-prename descclassname"><span class="pre">self.</span></span><span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.ligands.Ligand.self.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Ligand</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.ligands.Ligand.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.ligands.Ligand.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>LigandGroup</cite> containing this <cite>Ligand</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.ligands.Ligand.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.ligands.Ligand.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.ligands.LigandGroup">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.ligands.</span></span><span class="sig-name descname"><span class="pre">LigandGroup</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/ligands.html#LigandGroup"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.ligands.LigandGroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Polymer</span></code></a></p>
<p>A container for <cite>Ligand</cite> <cite>Monomers</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomers</strong> (<a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><em>Monomer</em></a><em>]</em><em>, </em><em>optional</em>) – Monomer or list containing Monomer objects to form the Polymer().</p></li>
<li><p><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polymer</cite>. This is
used when generating a pdb file using <cite>Polymer().pdb</cite>.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</p></li>
<li><p><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.nucleic_acid">
<span id="ampal-nucleic-acid-module"></span><h2>ampal.nucleic_acid module<a class="headerlink" href="#module-ampal.nucleic_acid" title="Permalink to this heading">¶</a></h2>
<p>Contains AMPAL objects representing nucleic acids.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.nucleic_acid.</span></span><span class="sig-name descname"><span class="pre">Nucleotide</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mol_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'UNK'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertion_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_hetero</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/nucleic_acid.html#Nucleotide"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Monomer</span></code></a></p>
<p>Represents a nucleotide base.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>Nucleotide</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</p></li>
<li><p><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>Nucleotide</cite>.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>Nucleotide</cite>.</p></li>
<li><p><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p></li>
<li><p><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Polynucleotide</em><em>, </em><em>optional</em>) – Reference to <cite>Polynucleotide</cite> containing the <cite>Nucleotide</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.mol_code">
<span class="sig-name descname"><span class="pre">mol_code</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.insertion_code">
<span class="sig-name descname"><span class="pre">insertion_code</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.is_hetero">
<span class="sig-name descname"><span class="pre">is_hetero</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.states">
<span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.reference_atom">
<span class="sig-name descname"><span class="pre">reference_atom</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Nucleotide.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.nucleic_acid.Nucleotide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – Raised if <cite>mol_code</cite> is not length 1 or 3.</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.nucleic_acid.</span></span><span class="sig-name descname"><span class="pre">Polynucleotide</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/nucleic_acid.html#Polynucleotide"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Polymer</span></code></a></p>
<p><cite>Polymer</cite> type object that represents a <cite>Polynucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomers</strong> (<a class="reference internal" href="#ampal.nucleic_acid.Nucleotide" title="ampal.nucleic_acid.Nucleotide"><em>Nucleotide</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.nucleic_acid.Nucleotide" title="ampal.nucleic_acid.Nucleotide"><em>Nucleotide</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Nucleotide</cite> or list containing <cite>Nucleotide</cite> objects to form the
<cite>Polynucleotide</cite>.</p></li>
<li><p><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polynucleotide</cite>. This is
used when generating a pdb file using <cite>Polynucleotide().pdb</cite>.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polynucleotide</cite>.</p></li>
<li><p><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>Polynucleotide</cite> ID</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polynucleotide</cite></p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Assembly or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.molecule_type">
<span class="sig-name descname"><span class="pre">molecule_type</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.ligands">
<span class="sig-name descname"><span class="pre">ligands</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polynucleotide</cite> chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.LigandGroup</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.sl">
<span class="sig-name descname"><span class="pre">sl</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – <cite>Polymer</cite> type objects can only be initialised empty or using
    a <cite>Monomer</cite>.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.nucleic_acid.Polynucleotide.sequence">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">sequence</span></span><a class="headerlink" href="#ampal.nucleic_acid.Polynucleotide.sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of the <cite>Polynucleotide</cite> as a string.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>sequence</strong> – String of the monomer sequence of the <cite>Polynucleotide</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>
<section id="module-ampal.pdb_parser">
<span id="ampal-pdb-parser-module"></span><h2>ampal.pdb_parser module<a class="headerlink" href="#module-ampal.pdb_parser" title="Permalink to this heading">¶</a></h2>
<p>Contains code for parsing PDB files.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.pdb_parser.</span></span><span class="sig-name descname"><span class="pre">PdbParser</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ignore_end</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Parses a PDB file and produces and AMPAL Assembly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb</strong> (<em>str</em>) – Either a path to a PDB file or a string containing PDB
format structural data.</p></li>
<li><p><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, flags <cite>pdb</cite> as a path and not a PDB string.</p></li>
<li><p><strong>pdb_id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the <cite>Assembly</cite>.</p></li>
<li><p><strong>ignore_end</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_functions">
<span class="sig-name descname"><span class="pre">proc_functions</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_functions" title="Permalink to this definition">¶</a></dt>
<dd><p>Keys are PDB labels i.e. “ATOM” or “END”, values are the
function that processes that specific line.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for the <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.pdb_lines">
<span class="sig-name descname"><span class="pre">pdb_lines</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.pdb_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Input PDB split into line.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.new_labels">
<span class="sig-name descname"><span class="pre">new_labels</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.new_labels" title="Permalink to this definition">¶</a></dt>
<dd><p>Indicates if new atom or residue labels have been found while
parsing the PDB file.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.state">
<span class="sig-name descname"><span class="pre">state</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.state" title="Permalink to this definition">¶</a></dt>
<dd><p>Current state being appended to. This is used on multi-state
files like NMR structures.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.pdb_parse_tree">
<span class="sig-name descname"><span class="pre">pdb_parse_tree</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.pdb_parse_tree" title="Permalink to this definition">¶</a></dt>
<dd><p>This dictionary represents the parse tree of the PDB file.
Each line of the structure is broken down into a key, the
entry label, and a value, the data.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.current_line">
<span class="sig-name descname"><span class="pre">current_line</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.current_line" title="Permalink to this definition">¶</a></dt>
<dd><p>The line that is currently being parsed.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.ignore_end">
<span class="sig-name descname"><span class="pre">ignore_end</span></span><a class="headerlink" href="#ampal.pdb_parser.PdbParser.ignore_end" title="Permalink to this definition">¶</a></dt>
<dd><p>If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool, optional</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.change_state">
<span class="sig-name descname"><span class="pre">change_state</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.change_state"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.change_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Increments current state and adds a new dict to the parse tree.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.check_for_non_canonical">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">check_for_non_canonical</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">residue</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.check_for_non_canonical"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.check_for_non_canonical" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks to see if the residue is non-canonical.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.end">
<span class="sig-name descname"><span class="pre">end</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.end"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.end" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes an “END” record.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.gen_states">
<span class="sig-name descname"><span class="pre">gen_states</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.gen_states"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.gen_states" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates the <cite>states</cite> dictionary for a <cite>Monomer</cite>.</p>
<dl class="simple">
<dt>monomer_data<span class="classifier">list</span></dt><dd><p>A list of atom data parsed from the input PDB.</p>
</dd>
<dt>parent<span class="classifier">ampal.Monomer</span></dt><dd><p><cite>Monomer</cite> used to assign <cite>parent</cite> on created
<cite>Atoms</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.make_ampal">
<span class="sig-name descname"><span class="pre">make_ampal</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.make_ampal"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.make_ampal" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates an AMPAL object from the parse tree.</p>
<p class="rubric">Notes</p>
<p>Will create an <cite>Assembly</cite> if there is a single state in the
parese tree or an <cite>AmpalContainer</cite> if there is more than one.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.parse_pdb_file">
<span class="sig-name descname"><span class="pre">parse_pdb_file</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.parse_pdb_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.parse_pdb_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs the PDB parser.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_atom">
<span class="sig-name descname"><span class="pre">proc_atom</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_atom"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes an “ATOM” or “HETATM” record.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_chain">
<span class="sig-name descname"><span class="pre">proc_chain</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chain_info</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_chain"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a chain into a <cite>Polymer</cite> type object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>chain_info</strong> (<em>(</em><em>set</em><em>, </em><em>OrderedDict</em><em>)</em>) – Contains a set of chain labels and atom records.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em>) – <cite>Assembly</cite> used to assign <cite>parent</cite> on created
<cite>Polymer</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>ValueError</strong> – Raised if multiple or unknown atom types found
    within the same chain.</p></li>
<li><p><strong>AttributeError</strong> – Raised if unknown <cite>Monomer</cite> type encountered.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_line_coordinate">
<span class="sig-name descname"><span class="pre">proc_line_coordinate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">line</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_line_coordinate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_line_coordinate" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts data from columns in ATOM/HETATM record.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_monomer">
<span class="sig-name descname"><span class="pre">proc_monomer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_info</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mon_cls</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_monomer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_monomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes a records into a <cite>Monomer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomer_info</strong> (<em>(</em><em>set</em><em>, </em><em>OrderedDict</em><em>)</em>) – Labels and data for a monomer.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Polymer</em>) – <cite>Polymer</cite> used to assign <cite>parent</cite> on created
<cite>Monomer</cite>.</p></li>
<li><p><strong>mon_cls</strong> (<cite>Monomer class or subclass</cite>, optional) – A <cite>Monomer</cite> class can be defined explicitly.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pdb_parser.PdbParser.proc_state">
<span class="sig-name descname"><span class="pre">proc_state</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">state_data</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">state_id</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#PdbParser.proc_state"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.PdbParser.proc_state" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes a state into an <cite>Assembly</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>state_data</strong> (<em>dict</em>) – Contains information about the state, including all
the per line structural data.</p></li>
<li><p><strong>state_id</strong> (<em>str</em>) – ID given to <cite>Assembly</cite> that represents the state.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.pdb_parser.load_pdb">
<span class="sig-prename descclassname"><span class="pre">ampal.pdb_parser.</span></span><span class="sig-name descname"><span class="pre">load_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ignore_end</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pdb_parser.html#load_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pdb_parser.load_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a PDB file into an AMPAL object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb</strong> (<em>str</em>) – Either a path to a PDB file or a string containing PDB
format structural data.</p></li>
<li><p><strong>path</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>true</cite>, flags <cite>pdb</cite> as a path and not a PDB string.</p></li>
<li><p><strong>pdb_id</strong> (<em>str</em><em>, </em><em>optional</em>) – Identifier for the <cite>Assembly</cite>.</p></li>
<li><p><strong>ignore_end</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>false</cite>, parsing of the file will stop when an “END”
record is encountered.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ampal</strong> – AMPAL object that contains the structural information from
the PDB file provided. If the PDB file has a single state
then an <cite>Assembly</cite> will be returned, otherwise an
<cite>AmpalContainer</cite> will be returned.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Assembly or ampal.AmpalContainer</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.protein">
<span id="ampal-protein-module"></span><h2>ampal.protein module<a class="headerlink" href="#module-ampal.protein" title="Permalink to this heading">¶</a></h2>
<p>AMPAL objects that represent protein.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.protein.</span></span><span class="sig-name descname"><span class="pre">Polypeptide</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Polymer</span></code></a></p>
<p>Container for <cite>Residues</cite>, inherits from <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomers</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>Residue</cite> or list containing <cite>Residue</cite> objects to form the
<cite>Polypeptide</cite>.</p></li>
<li><p><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>Polypeptide</cite>. This is
used when generating a pdb file using <cite>Polypeptide().pdb</cite>.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>Polymer</cite>.</p></li>
<li><p><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>Polypeptide</cite> ID</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>Polypeptide</cite></p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Assembly or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.molecule_type">
<span class="sig-name descname"><span class="pre">molecule_type</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.ligands">
<span class="sig-name descname"><span class="pre">ligands</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>Polypeptide</cite> chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.LigandGroup</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.sl">
<span class="sig-name descname"><span class="pre">sl</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – <cite>Polymer</cite> type objects can only be initialised empty or using
    a <cite>Monomer</cite>.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.backbone">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">backbone</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Polymer</cite> containing only the backbone atoms.</p>
<p class="rubric">Notes</p>
<p>Metadata is not currently preserved from the parent object.
Sequence data is retained, but only the main chain atoms are retained.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bb_poly</strong> – Polymer containing only the backbone atoms of the original
Polymer.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.backbone_bond_angles">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">backbone_bond_angles</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.backbone_bond_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary containing backbone bond angles as lists of floats.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bond_angles</strong> – Keys are <cite>n_ca_c</cite>, <cite>ca_c_o</cite>, <cite>ca_c_n</cite> and <cite>c_n_ca</cite>, referring
to the N-CA-C, CA-C=O, CA-C-N and C-N-CA angles respectively.
Values are lists of floats : the bond angles in degrees.
The lists of n_ca_c, ca_c_o are of length k for a <cite>Polypeptide</cite>
containing k <cite>Residues</cite>. The list of ca_c_n and c_n_ca are of
length k-1 for a <cite>Polypeptide</cite> containing k <cite>Residues</cite> (These
angles are across the peptide bond, and are therefore formed
between successive <cite>Residue</cite> pairs).</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.backbone_bond_lengths">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">backbone_bond_lengths</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.backbone_bond_lengths" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary containing backbone bond lengths as lists of floats.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bond_lengths</strong> – Keys are <cite>n_ca</cite>, <cite>ca_c</cite>, <cite>c_o</cite> and <cite>c_n</cite>, referring to the
N-CA, CA-C, C=O and C-N bonds respectively. Values are
lists of floats : the bond lengths in Angstroms.
The lists of n_ca, ca_c and c_o are of length k for
a Polypeptide containing k Residues. The list of c_n bonds
is of length k-1 for a Polypeptide containing k Residues
(C-N formed between successive <cite>Residue</cite> pairs).</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.c_join">
<span class="sig-name descname"><span class="pre">c_join</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">other</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">psi</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-40.76</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">omega</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-178.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-65.07</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o_c_n_angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">c_n_ca_angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">c_n_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relabel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.c_join"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.c_join" title="Permalink to this definition">¶</a></dt>
<dd><p>Joins other to self at the C-terminus via a peptide bond.</p>
<p class="rubric">Notes</p>
<p>This function directly modifies self. It does not return a new object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>other</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – </p></li>
<li><p><strong>psi</strong> (<em>float</em><em>, </em><em>optional</em>) – Psi torsion angle (degrees) between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other.</p></li>
<li><p><strong>omega</strong> (<em>float</em><em>, </em><em>optional</em>) – Omega torsion angle (degrees) between final <cite>Residue</cite> of
self and first <cite>Residue</cite> of other.</p></li>
<li><p><strong>phi</strong> (<em>float</em><em>, </em><em>optional</em>) – Phi torsion angle (degrees) between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other.</p></li>
<li><p><strong>o_c_n_angle</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired angle between O, C (final <cite>Residue</cite> of self) and N
(first <cite>Residue</cite> of other) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</p></li>
<li><p><strong>c_n_ca_angle</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired angle between C (final <cite>Residue</cite> of self) and N, CA
(first <cite>Residue</cite> of other) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</p></li>
<li><p><strong>c_n_length</strong> (<em>float</em><em> or </em><em>None</em><em>, </em><em>optional</em>) – Desired peptide bond length between final <cite>Residue</cite> of self
and first <cite>Residue</cite> of other. If <cite>None</cite>, default value is taken
from <cite>ideal_backbone_bond_lengths</cite>.</p></li>
<li><p><strong>relabel</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite>, <cite>relabel_all</cite> is run on self before returning.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>TypeError:</strong> – If other is not a <cite>Residue</cite> or a Polypeptide.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.fasta">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">fasta</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates sequence data for the protein in FASTA format.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.get_slice_from_res_id">
<span class="sig-name descname"><span class="pre">get_slice_from_res_id</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.get_slice_from_res_id"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.get_slice_from_res_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Polypeptide</cite> containing the <cite>Residues</cite> in start/end range.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>start</strong> (<em>str</em>) – string representing start residue id (PDB numbering)</p></li>
<li><p><strong>end</strong> (<em>str</em>) – string representing end residue id (PDB numbering)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>slice_polymer</strong> – Polymer containing the residue range specified by start-end</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer">Polymer</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.isoelectric_point">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">isoelectric_point</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the isoelectric point of the <cite>Assembly</cite>.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.molar_extinction_280">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">molar_extinction_280</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.molar_extinction_280" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the extinction co-efficient of the <cite>Assembly</cite> at 280 nm.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.molecular_weight">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">molecular_weight</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.molecular_weight" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the molecular weight of the <cite>Assembly</cite> in Daltons.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.n_join">
<span class="sig-name descname"><span class="pre">n_join</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">other</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">psi</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-40.76</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">omega</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-178.25</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">phi</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">-65.07</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">o_c_n_angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">c_n_ca_angle</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">c_n_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relabel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.n_join"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.n_join" title="Permalink to this definition">¶</a></dt>
<dd><p>Joins other to self at the N-terminus via a peptide bond.</p>
<p class="rubric">Notes</p>
<p>This function directly modifies self. It does not return a new object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>other</strong> (<a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue"><em>Residue</em></a><em> or </em><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – </p></li>
<li><p><strong>psi</strong> (<em>float</em>) – Psi torsion angle (degrees) between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self.</p></li>
<li><p><strong>omega</strong> (<em>float</em>) – Omega torsion angle (degrees) between final <cite>Residue</cite> of
other and first <cite>Residue</cite> of self.</p></li>
<li><p><strong>phi</strong> (<em>float</em>) – Phi torsion angle (degrees) between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self.</p></li>
<li><p><strong>o_c_n_angle</strong> (<em>float</em><em> or </em><em>None</em>) – Desired angle between O, C (final <cite>Residue</cite> of other) and N
(first <cite>Residue</cite> of self) atoms. If <cite>None</cite>, default value is
taken from <cite>ideal_backbone_bond_angles</cite>.</p></li>
<li><p><strong>c_n_ca_angle</strong> (<em>float</em><em> or </em><em>None</em>) – Desired angle between C (final <cite>Residue</cite> of other) and N, CA
(first <cite>Residue</cite> of self) atoms. If <cite>None</cite>, default value is taken
from <cite>ideal_backbone_bond_angles</cite>.</p></li>
<li><p><strong>c_n_length</strong> (<em>float</em><em> or </em><em>None</em>) – Desired peptide bond length between final <cite>Residue</cite> of other
and first <cite>Residue</cite> of self. If None, default value is taken
from ideal_backbone_bond_lengths.</p></li>
<li><p><strong>relabel</strong> (<em>bool</em>) – If True, relabel_all is run on self before returning.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>TypeError:</strong> – If other is not a <cite>Residue</cite> or a <cite>Polypeptide</cite></p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.primitive">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">primitive</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Primitive of the backbone.</p>
<p class="rubric">Notes</p>
<p>This is the average of the positions of all the CAs in frames
of <cite>sl</cite> <cite>Residues</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.radii_of_curvature">
<span class="sig-name descname"><span class="pre">radii_of_curvature</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.radii_of_curvature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.radii_of_curvature" title="Permalink to this definition">¶</a></dt>
<dd><p>List of radius of curvature values along the <cite>Polypeptide</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.rise_per_residue">
<span class="sig-name descname"><span class="pre">rise_per_residue</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.rise_per_residue"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.rise_per_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>List of rise per residue values along the <cite>Polypeptide</cite>.
.. rubric:: Notes</p>
<p>Calculated from <cite>Polypeptide.primitive</cite>.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.sequence">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">sequence</span></span><a class="headerlink" href="#ampal.protein.Polypeptide.sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the sequence of the <cite>Polymer</cite> as a string.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>sequence</strong> – String of the <cite>Residue</cite> sequence of the <cite>Polypeptide</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.tag_ca_geometry">
<span class="sig-name descname"><span class="pre">tag_ca_geometry</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_axis_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'ref_axis'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_ca_geometry"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_ca_geometry" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each <cite>Residue</cite> with rise_per_residue, radius_of_curvature and residues_per_turn.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are already
tagged.</p></li>
<li><p><strong>reference_axis</strong> (<em>list</em><em>(</em><em>numpy.array</em><em> or </em><em>tuple</em><em> or </em><em>list</em><em>)</em><em>, </em><em>optional</em>) – Coordinates to feed to geometry functions that depend on
having a reference axis.</p></li>
<li><p><strong>reference_axis_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Used to name the keys in tags at <cite>Polypeptide</cite> and <cite>Residue</cite> level.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.tag_sidechain_dihedrals">
<span class="sig-name descname"><span class="pre">tag_sidechain_dihedrals</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_sidechain_dihedrals"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_sidechain_dihedrals" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each monomer with side-chain dihedral angles. For residues that
do not have any rotamers (Alanine and Glycine) the rotamer is tagged as
0 and chi_angles as None.</p>
<dl class="simple">
<dt>force: bool, optional</dt><dd><p>If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are
already tagged.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.tag_torsion_angles">
<span class="sig-name descname"><span class="pre">tag_torsion_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.tag_torsion_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.tag_torsion_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>Tags each Monomer of the Polymer with its omega, phi and psi torsion angle.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>force</strong> (<em>bool</em><em>, </em><em>optional</em>) – If <cite>True</cite> the tag will be run even if <cite>Residues</cite> are
already tagged.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.valid_backbone_bond_angles">
<span class="sig-name descname"><span class="pre">valid_backbone_bond_angles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.valid_backbone_bond_angles"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.valid_backbone_bond_angles" title="Permalink to this definition">¶</a></dt>
<dd><p>True if all backbone bond angles are within atol degrees of their expected values.</p>
<p class="rubric">Notes</p>
<p>Ideal bond angles taken from [1].</p>
<p class="rubric">References</p>
<aside class="footnote-list brackets">
<aside class="footnote brackets" id="id8" role="doc-footnote">
<span class="label"><span class="fn-bracket">[</span>1<span class="fn-bracket">]</span></span>
<p>Schulz, G. E, and R. Heiner Schirmer. Principles Of
Protein Structure. New York: Springer-Verlag, 1979.</p>
</aside>
</aside>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>atol</strong> (<em>float</em><em>, </em><em>optional</em>) – Tolerance value in degrees for the absolute deviation
away from ideal backbone bond angles.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.protein.Polypeptide.valid_backbone_bond_lengths">
<span class="sig-name descname"><span class="pre">valid_backbone_bond_lengths</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Polypeptide.valid_backbone_bond_lengths"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Polypeptide.valid_backbone_bond_lengths" title="Permalink to this definition">¶</a></dt>
<dd><p>True if all backbone bonds are within atol Angstroms of the expected distance.</p>
<p class="rubric">Notes</p>
<p>Ideal bond lengths taken from [1].</p>
<p class="rubric">References</p>
<aside class="footnote-list brackets">
<aside class="footnote brackets" id="id9" role="doc-footnote">
<span class="label"><span class="fn-bracket">[</span>1<span class="fn-bracket">]</span></span>
<p>Schulz, G. E, and R. Heiner Schirmer. Principles Of
Protein Structure. New York: Springer-Verlag, 1979.</p>
</aside>
</aside>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>atol</strong> (<em>float</em><em>, </em><em>optional</em>) – Tolerance value in Angstoms for the absolute deviation
away from ideal backbone bond lengths.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.protein.Residue">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.protein.</span></span><span class="sig-name descname"><span class="pre">Residue</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mol_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'UNK'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertion_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_hetero</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#Residue"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.Residue" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Monomer</span></code></a></p>
<p>Represents a amino acid <cite>Residue</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the Monomer. OrderedDict
is used to maintain the order items were added to the
dictionary.</p></li>
<li><p><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the monomer.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the residue.</p></li>
<li><p><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of monomer, used if reading from pdb.</p></li>
<li><p><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Polypeptide</em><em>, </em><em>optional</em>) – Reference to <cite>Polypeptide</cite> containing the <cite>Residue</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.mol_code">
<span class="sig-name descname"><span class="pre">mol_code</span></span><a class="headerlink" href="#ampal.protein.Residue.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Residue</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.insertion_code">
<span class="sig-name descname"><span class="pre">insertion_code</span></span><a class="headerlink" href="#ampal.protein.Residue.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Residue</cite>, used if reading from pdb.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.is_hetero">
<span class="sig-name descname"><span class="pre">is_hetero</span></span><a class="headerlink" href="#ampal.protein.Residue.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.states">
<span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.protein.Residue.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Residue</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.protein.Residue.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the residue.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.reference_atom">
<span class="sig-name descname"><span class="pre">reference_atom</span></span><a class="headerlink" href="#ampal.protein.Residue.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference atom. This is used
by various functions, for example backbone primitives are
calculated using the atom defined using this key.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.protein.Residue.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polypeptide</cite> containing this <cite>Residue</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.protein.Residue.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.protein.Residue.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – Raised if <cite>mol_code</cite> is not length 1 or 3.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Residue.backbone">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">backbone</span></span><a class="headerlink" href="#ampal.protein.Residue.backbone" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a new <cite>Residue</cite> containing only the backbone atoms.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>bb_monomer</strong> – <cite>Residue</cite> containing only the backbone atoms of the original
<cite>Monomer</cite>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p><a class="reference internal" href="#ampal.protein.Residue" title="ampal.protein.Residue">Residue</a></p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>IndexError</strong> – Raise if the <cite>atoms</cite> dict does not contain the backbone
    atoms (N, CA, C, O).</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Residue.centroid">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">centroid</span></span><a class="headerlink" href="#ampal.protein.Residue.centroid" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the centroid of the residue.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>centroid</strong> – Returns a 3D coordinate for the residue unless a CB
atom is not available, in which case <cite>None</cite> is
returned.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>numpy.array or None</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Uses the definition of the centroid from Huang <em>et al</em> <a class="footnote-reference brackets" href="#id11" id="id10" role="doc-noteref"><span class="fn-bracket">[</span>2<span class="fn-bracket">]</span></a>.</p>
<p class="rubric">References</p>
<aside class="footnote-list brackets">
<aside class="footnote brackets" id="id11" role="doc-footnote">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id10">2</a><span class="fn-bracket">]</span></span>
<p>Huang ES, Subbiah S and Levitt M (1995) Recognizing Native
Folds by the Arrangement of Hydrophobic and Polar Residues, J. Mol.
Biol return., <strong>252</strong>, 709-720.</p>
</aside>
</aside>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Residue.side_chain">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">side_chain</span></span><a class="headerlink" href="#ampal.protein.Residue.side_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>List of the side-chain atoms (R-group).</p>
<p class="rubric">Notes</p>
<p>Returns empty list for glycine.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>side_chain_atoms</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>list(<cite>Atoms</cite>)</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="ampal.protein.Residue.unique_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">unique_id</span></span><a class="headerlink" href="#ampal.protein.Residue.unique_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a tuple that uniquely identifies a <cite>Monomer</cite> in an <cite>Assembly</cite>.</p>
<p class="rubric">Notes</p>
<p>The unique_id will uniquely identify each monomer within a polymer.
If each polymer in an assembly has a distinct id, it will uniquely
identify each monomer within the assembly.</p>
<p>The hetero-flag is defined as in Biopython as a string that is
either a single whitespace in the case of a non-hetero atom,
or ‘<a href="#id12"><span class="problematic" id="id13">H_</span></a>’ plus the name of the hetero-residue (e.g. ‘H_GLC’ in
the case of a glucose molecule), or ‘W’ in the case of a water
molecule.</p>
<p>For more information, see the Biopython documentation or this
Biopython wiki page:
<a class="reference external" href="http://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ">http://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ</a></p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>unique_id</strong> – unique_id[0] is the polymer_id unique_id[1] is a triple
of the hetero-flag, the monomer id (residue number) and the
insertion code.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>tuple</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.protein.align">
<span class="sig-prename descclassname"><span class="pre">ampal.protein.</span></span><span class="sig-name descname"><span class="pre">align</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">target</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mobile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_i</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mobile_i</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#align"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.align" title="Permalink to this definition">¶</a></dt>
<dd><p>Aligns one Polypeptide (mobile) to another (target).</p>
<p class="rubric">Notes</p>
<p>This function directly modifies atoms of the mobile Polypeptide!
It does not return a new object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>target</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – Polypeptide to be aligned to.</p></li>
<li><p><strong>mobile</strong> (<a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide"><em>Polypeptide</em></a>) – Polypeptide to be moved during alignment.</p></li>
<li><p><strong>target_i</strong> (<em>int</em><em>, </em><em>optional</em>) – Index of <cite>Residue</cite> in target to align to.</p></li>
<li><p><strong>mobile_i</strong> (<em>int</em><em>, </em><em>optional</em>) – Index of <cite>Residue</cite> in mobile to be aligned.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.protein.flat_list_to_dummy_chain">
<span class="sig-prename descclassname"><span class="pre">ampal.protein.</span></span><span class="sig-name descname"><span class="pre">flat_list_to_dummy_chain</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_group_s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#flat_list_to_dummy_chain"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.flat_list_to_dummy_chain" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts flat list of coordinates into dummy C-alpha carbons</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atom_list</strong> (<em>[</em><a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><em>Atom</em></a><em>]</em>) – Flat list of co-ordinates.</p></li>
<li><p><strong>atom_group_s</strong> (<em>int</em><em>, </em><em>optional</em>) – Size of atom groups.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>polymer</strong> – <cite>Polymer</cite> object containing atom coord converted <cite>Monomers</cite>
with ‘DUM’ atom name.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ampal.protein.flat_list_to_polymer">
<span class="sig-prename descclassname"><span class="pre">ampal.protein.</span></span><span class="sig-name descname"><span class="pre">flat_list_to_polymer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atom_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_group_s</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/protein.html#flat_list_to_polymer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.protein.flat_list_to_polymer" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes a flat list of atomic coordinates and converts it to a <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atom_list</strong> (<em>[</em><a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><em>Atom</em></a><em>]</em>) – Flat list of coordinates.</p></li>
<li><p><strong>atom_group_s</strong> (<em>int</em><em>, </em><em>optional</em>) – Size of atom groups.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>polymer</strong> – <cite>Polymer</cite> object containing atom coords converted <cite>Monomers</cite>.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.protein.Polypeptide" title="ampal.protein.Polypeptide">Polypeptide</a></p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ValueError</strong> – Raised if <cite>atom_group_s</cite> != 4 or 5</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ampal.pseudo_atoms">
<span id="ampal-pseudo-atoms-module"></span><h2>ampal.pseudo_atoms module<a class="headerlink" href="#module-ampal.pseudo_atoms" title="Permalink to this heading">¶</a></h2>
<p>Contains AMPAL objects representing pseudo atoms.</p>
<dl class="py class">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.pseudo_atoms.</span></span><span class="sig-name descname"><span class="pre">Primitive</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.pseudo_atoms.PseudoGroup" title="ampal.pseudo_atoms.PseudoGroup"><code class="xref py py-class docutils literal notranslate"><span class="pre">PseudoGroup</span></code></a></p>
<p>A backbone path composed of <cite>PseudoAtoms</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pseudo_atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>PsuedoMonomer</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</p></li>
<li><p><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>PsuedoMonomer</cite>.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>PsuedoMonomer</cite>.</p></li>
<li><p><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>PsuedoMonomer</cite>, used if reading from pdb.</p></li>
<li><p><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</p></li>
<li><p><strong>parent</strong> (<em>ampal.PseudoGroup</em><em>, </em><em>optional</em>) – Reference to <cite>PseudoGroup</cite> containing the <cite>PsuedoMonomer</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.mol_code">
<span class="sig-name descname"><span class="pre">mol_code</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.insertion_code">
<span class="sig-name descname"><span class="pre">insertion_code</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.is_hetero">
<span class="sig-name descname"><span class="pre">is_hetero</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.states">
<span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.reference_atom">
<span class="sig-name descname"><span class="pre">reference_atom</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – Raised if <cite>mol_code</cite> is not length 1 or 3.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.coordinates">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">coordinates</span></span><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the backbone coordinates for the <cite>Primitive</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.from_coordinates">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_coordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coordinates</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.from_coordinates"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.from_coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a <cite>Primitive</cite> from a list of coordinates.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.radii_of_curvature">
<span class="sig-name descname"><span class="pre">radii_of_curvature</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.radii_of_curvature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.radii_of_curvature" title="Permalink to this definition">¶</a></dt>
<dd><p>The radius of curvature at each point on the Polymer primitive.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the radius of curvature,
at a point on the Polymer primitive. Element i is the radius
of the circumcircle formed from indices [i-1, i, i+1] of the
primitve. The first and final values are None.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.Primitive.rise_per_residue">
<span class="sig-name descname"><span class="pre">rise_per_residue</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#Primitive.rise_per_residue"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.Primitive.rise_per_residue" title="Permalink to this definition">¶</a></dt>
<dd><p>The rise per residue at each point on the Primitive.</p>
<p class="rubric">Notes</p>
<p>Each element of the returned list is the rise per residue,
at a point on the Primitive. Element i is the distance
between primitive[i] and primitive[i + 1]. The final value
is None.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.pseudo_atoms.</span></span><span class="sig-name descname"><span class="pre">PseudoAtom</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coordinates</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">occupancy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bfactor</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">charge</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoAtom"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Atom" title="ampal.base_ampal.Atom"><code class="xref py py-class docutils literal notranslate"><span class="pre">Atom</span></code></a></p>
<p>Object containing 3D coordinates and name.</p>
<p class="rubric">Notes</p>
<p>Used to represent pseudo atoms (e.g. centre_of_mass) in ISAMBARD.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>coordinates</strong> (<em>3D Vector</em><em> (</em><em>tuple</em><em>, </em><em>list</em><em>, </em><em>numpy.array</em><em>)</em>) – Position of <cite>PseudoAtom</cite> in 3D space.</p></li>
<li><p><strong>element</strong> (<em>str</em>) – Element of <cite>PseudoAtom</cite>.</p></li>
<li><p><strong>atom_id</strong> (<em>str</em>) – Identifier for <cite>PseudoAtom</cite>, usually a number.</p></li>
<li><p><strong>res_label</strong> (<em>str</em><em>, </em><em>optional</em>) – Label used in <cite>Monomer</cite> to refer to the <cite>PseudoAtom</cite> type i.e.
“CA” or “OD1”.</p></li>
<li><p><strong>occupancy</strong> (<em>float</em><em>, </em><em>optional</em>) – The occupancy of the <cite>PseudoAtom</cite>.</p></li>
<li><p><strong>bfactor</strong> (<em>float</em><em>, </em><em>optional</em>) – The bfactor of the <cite>PseudoAtom</cite>.</p></li>
<li><p><strong>charge</strong> (<em>str</em><em>, </em><em>optional</em>) – The point charge of the <cite>PseudoAtom</cite>.</p></li>
<li><p><strong>state</strong> (<em>str</em>) – The state of this <cite>PseudoAtom</cite>. Used to identify <cite>PseudoAtoms</cite>
with a number of conformations.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Monomer</em><em>, </em><em>optional</em>) – A reference to the <cite>Monomer</cite> containing this <cite>PseudoAtom</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.id" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifier for <cite>PseudoAtom</cite>, usually a number.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom.res_label">
<span class="sig-name descname"><span class="pre">res_label</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.res_label" title="Permalink to this definition">¶</a></dt>
<dd><p>Label used in <cite>PseudoGroup</cite> to refer to the <cite>Atom</cite> type i.e. “CA” or “OD1”.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom.element">
<span class="sig-name descname"><span class="pre">element</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.element" title="Permalink to this definition">¶</a></dt>
<dd><p>Element of <cite>Atom</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.parent" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>A reference to the <cite>PseudoGroup</cite> containing this <cite>PseudoAtom</cite>.</dt><dd><p>number of conformations.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.PseudoAtom</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoAtom.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoAtom.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.pseudo_atoms.</span></span><span class="sig-name descname"><span class="pre">PseudoGroup</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polymer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sl</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoGroup"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Polymer" title="ampal.base_ampal.Polymer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Polymer</span></code></a></p>
<p>Container for <cite>PseudoMonomer</cite>, inherits from <cite>Polymer</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomers</strong> (<a class="reference internal" href="#ampal.pseudo_atoms.PseudoAtom" title="ampal.pseudo_atoms.PseudoAtom"><em>PseudoAtom</em></a><em> or </em><em>[</em><a class="reference internal" href="#ampal.pseudo_atoms.PseudoGroup" title="ampal.pseudo_atoms.PseudoGroup"><em>PseudoGroup</em></a><em>]</em><em>, </em><em>optional</em>) – <cite>PseudoMonomer</cite> or list containing <cite>PseudoMonomer</cite> objects to form the
<cite>PseudoGroup</cite>.</p></li>
<li><p><strong>polymer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – An ID that the user can use to identify the <cite>PseudoGroup</cite>. This is
used when generating a pdb file using <cite>PseudoGroup().pdb</cite>.</p></li>
<li><p><strong>parent</strong> (<em>ampal.Assembly</em><em>, </em><em>optional</em>) – Reference to <cite>Assembly</cite> containing the <cite>PseudoGroup</cite>.</p></li>
<li><p><strong>sl</strong> (<em>int</em><em>, </em><em>optional</em>) – The default smoothing level used when calculating the
backbone primitive.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.id" title="Permalink to this definition">¶</a></dt>
<dd><p><cite>PseudoGroup</cite> ID</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Reference to <cite>Assembly</cite> containing the <cite>PseudoGroup</cite></p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.Assembly or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.molecule_type">
<span class="sig-name descname"><span class="pre">molecule_type</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.molecule_type" title="Permalink to this definition">¶</a></dt>
<dd><p>A description of the type of <cite>Polymer</cite> i.e. Protein, DNA etc.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.ligands">
<span class="sig-name descname"><span class="pre">ligands</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.ligands" title="Permalink to this definition">¶</a></dt>
<dd><p>A <cite>LigandGroup</cite> containing all the <cite>Ligands</cite> associated with this
<cite>PseudoGroup</cite> chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>ampal.LigandGroup</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoGroup.sl">
<span class="sig-name descname"><span class="pre">sl</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoGroup.sl" title="Permalink to this definition">¶</a></dt>
<dd><p>The default smoothing level used when calculating the
backbone primitive.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – <cite>Polymer</cite> type objects can only be initialised empty or using
    a <cite>Monomer</cite>.</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ampal.pseudo_atoms.</span></span><span class="sig-name descname"><span class="pre">PseudoMonomer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pseudo_atoms</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mol_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'UNK'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer_id</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertion_code</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'</span> <span class="pre">'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ampal/pseudo_atoms.html#PseudoMonomer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#ampal.base_ampal.Monomer" title="ampal.base_ampal.Monomer"><code class="xref py py-class docutils literal notranslate"><span class="pre">Monomer</span></code></a></p>
<p>Represents a collection of <cite>PsuedoAtoms</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pseudo_atoms</strong> (<em>OrderedDict</em><em>, </em><em>optional</em>) – OrderedDict containing Atoms for the <cite>PsuedoMonomer</cite>. OrderedDict
is used to maintain the order items were added to the
dictionary.</p></li>
<li><p><strong>mol_code</strong> (<em>str</em><em>, </em><em>optional</em>) – One or three letter code that represents the <cite>PsuedoMonomer</cite>.</p></li>
<li><p><strong>monomer_id</strong> (<em>str</em><em>, </em><em>optional</em>) – String used to identify the <cite>PsuedoMonomer</cite>.</p></li>
<li><p><strong>insertion_code</strong> (<em>str</em><em>, </em><em>optional</em>) – Insertion code of <cite>PsuedoMonomer</cite>, used if reading from pdb.</p></li>
<li><p><strong>is_hetero</strong> (<em>bool</em><em>, </em><em>optional</em>) – True if is a hetero atom in pdb. Helps with PDB formatting.</p></li>
<li><p><strong>parent</strong> (<em>ampal.PseudoGroup</em><em>, </em><em>optional</em>) – Reference to <cite>PseudoGroup</cite> containing the <cite>PsuedoMonomer</cite>.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.mol_code">
<span class="sig-name descname"><span class="pre">mol_code</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.mol_code" title="Permalink to this definition">¶</a></dt>
<dd><p>PDB molecule code that represents the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.insertion_code">
<span class="sig-name descname"><span class="pre">insertion_code</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.insertion_code" title="Permalink to this definition">¶</a></dt>
<dd><p>Insertion code of <cite>Nucleotide</cite>, used if reading from pdb.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.is_hetero">
<span class="sig-name descname"><span class="pre">is_hetero</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.is_hetero" title="Permalink to this definition">¶</a></dt>
<dd><p>True if is a hetero atom in pdb. Helps with PDB formatting.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.states">
<span class="sig-name descname"><span class="pre">states</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.states" title="Permalink to this definition">¶</a></dt>
<dd><p>Contains an <cite>OrderedDicts</cite> containing atom information for each
state available for the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.id">
<span class="sig-name descname"><span class="pre">id</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.id" title="Permalink to this definition">¶</a></dt>
<dd><p>String used to identify the <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.reference_atom">
<span class="sig-name descname"><span class="pre">reference_atom</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.reference_atom" title="Permalink to this definition">¶</a></dt>
<dd><p>The key that corresponds to the reference <cite>Atom</cite>. This is used
by various functions, for example backbone primitives are
calculated using the <cite>Atom</cite> defined using this key.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.parent">
<span class="sig-name descname"><span class="pre">parent</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.parent" title="Permalink to this definition">¶</a></dt>
<dd><p>A reference to the <cite>Polynucleotide</cite> containing this <cite>Nucleotide</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#ampal.nucleic_acid.Polynucleotide" title="ampal.nucleic_acid.Polynucleotide">Polynucleotide</a> or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.tags">
<span class="sig-name descname"><span class="pre">tags</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.tags" title="Permalink to this definition">¶</a></dt>
<dd><p>A dictionary containing information about this AMPAL object.
The tags dictionary is used by AMPAL to cache information
about this object, but is also intended to be used by users
to store any relevant information they have.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>ValueError</strong> – Raised if <cite>mol_code</cite> is not length 1 or 3.</p>
</dd>
</dl>
<dl class="py property">
<dt class="sig sig-object py" id="ampal.pseudo_atoms.PseudoMonomer.pdb">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pdb</span></span><a class="headerlink" href="#ampal.pseudo_atoms.PseudoMonomer.pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates a PDB string for the <cite>PseudoMonomer</cite>.</p>
</dd></dl>

</dd></dl>

</section>
<section id="module-ampal">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-ampal" title="Permalink to this heading">¶</a></h2>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">AMPAL</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="AMPAL_Introduction_and_Selections.html">Introduction to AMPAL</a></li>
<li class="toctree-l1"><a class="reference internal" href="Advanced_Selections_and_Analysis.html">Advanced Selections</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API Reference</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">ampal package</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="modules.html">ampal</a><ul>
      <li>Previous: <a href="modules.html" title="previous chapter">ampal</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Woolfson Group, UoB.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 7.0.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>
      
      |
      <a href="_sources/ampal.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>