Search.setIndex({"docnames": ["AMPAL_Introduction_and_Selections", "Advanced_Selections_and_Analysis", "ampal", "index", "installation", "modules"], "filenames": ["AMPAL_Introduction_and_Selections.rst", "Advanced_Selections_and_Analysis.rst", "ampal.rst", "index.rst", "installation.rst", "modules.rst"], "titles": ["An Introduction to the AMPAL Framework", "Advanced Selections and Analysis", "ampal package", "AMPAL Documentation", "Installation", "ampal"], "terms": {"isambard": [0, 2], "repres": [0, 2, 3], "protein": [0, 1, 5], "python": [0, 1, 3], "ar": [0, 1, 2], "part": [0, 2], "modul": [0, 3, 5], "stand": [0, 3], "monom": [0, 1, 2, 3, 5], "polym": [0, 1, 2, 3, 5], "assembli": [0, 1, 3, 5], "which": [0, 1, 2, 4], "main": [0, 2], "function": [0, 1, 2], "element": [0, 1, 2, 5], "These": [0, 2], "design": [0, 2], "hierarch": 0, "natur": 0, "intuit": [0, 3], "wai": [0, 1], "all": [0, 2, 3], "model": [0, 2], "produc": [0, 1, 2], "you": [0, 1, 2, 4], "can": [0, 1, 2, 4], "also": [0, 2, 4], "load": [0, 1, 2], "crystal": [0, 1], "structur": [0, 1, 2, 3], "q": [0, 1, 2], "so": [0, 1, 2, 4], "why": 0, "aren": 0, "t": [0, 1, 2], "just": [0, 1, 2], "call": [0, 1, 2], "well": [0, 1, 2], "polypeptid": [0, 1, 2, 5], "thei": [0, 1, 2], "specif": [0, 1, 2], "version": [0, 2], "we": [0, 1, 2], "want": [0, 1, 2], "keep": 0, "base": [0, 2], "gener": [0, 1, 2], "possibl": [0, 2], "allow": [0, 2], "other": [0, 1, 2], "biomolecul": 0, "like": [0, 1, 2], "dna": [0, 2], "rna": 0, "even": [0, 1, 2], "unnatur": 0, "beta": 0, "amino": [0, 1, 2], "acid": [0, 1, 2], "us": [0, 1, 2, 4], "thi": [0, 1, 2, 4], "architectur": 0, "while": [0, 1, 2], "some": [0, 1, 2], "featur": [0, 2], "current": [0, 2], "implement": [0, 2], "lead": 0, "more": [0, 1, 2], "scalabl": 0, "code": [0, 1, 2], "reduc": 0, "duplic": 0, "futur": 0, "import": [0, 1, 2], "ani": [0, 1, 2], "pars": [0, 2], "easili": [0, 1], "analys": [0, 2, 3], "The": [0, 1, 2], "onli": [0, 1, 2], "need": [0, 1, 4], "i": [0, 2, 3], "load_pdb": [0, 1, 2, 5], "It": [0, 1, 2], "take": [0, 1, 2], "path": [0, 2], "string": [0, 1, 2], "input": [0, 1, 2], "argument": 0, "3qy1": [0, 1], "contain": [0, 1, 2], "449": 0, "return": [0, 1, 2], "assign": [0, 2], "variabl": 0, "look": [0, 1, 2], "insid": [0, 1], "my_protein": 0, "rememb": 0, "jupyt": 0, "notebook": 0, "onc": 0, "": [0, 1, 2], "have": [0, 1, 2, 4], "its": [0, 1, 2], "attribut": 0, "method": [0, 1, 2], "type": [0, 2, 4], "press": 0, "tab": 0, "two": [0, 1, 2], "easi": 0, "check": [0, 1, 2], "sequenc": [0, 1, 2, 5], "didtlisnnalwskmlveedpgffeklaqaqkprflwigcsdsrvpaerltglepgelfvhrnvanlvihtdlnclsvvqyavdvlevehiiicghsgcggikaavenpelglinnwllhirdiwlkhssllgkmpeeqrldalyelnvmeqvynlghstimqsawkrgqnvtihgwaysindgllrdldvtatnretlengyhkgisalslkyi": [0, 1], "kdidtlisnnalwskmlveedpgffeklaqaqkprflwigcsdsrvpaerltglepgelfvhrnvanlvihtdlnclsvvqyavdvlevehiiicghsgcggikaavenpelglinnwllhirdiwlkhssllgkmpeeqrldalyelnvmeqvynlghstimqsawkrgqnvtihgwaysindgllrdldvtatnretlengyhkgisalslkyi": 0, "list": [0, 1, 2], "one": [0, 1, 2], "each": [0, 1, 2], "determin": [0, 2], "properti": [0, 2], "molecular_weight": [0, 2, 5], "48508": 0, "931580000004": 0, "molar_extinction_280": [0, 2, 5], "83640": 0, "isoelectric_point": [0, 2, 5], "400000000000004": 0, "id": [0, 2, 5], "item": [0, 2], "access": [0, 2], "analog": 0, "standard": 0, "0": [0, 1, 2], "first": [0, 1, 2], "215": 0, "didtlisnnalw": 0, "from": [0, 1, 2, 3], "In": [0, 2], "case": [0, 2], "A": [0, 1, 2], "b": [0, 1, 2, 5], "216": 0, "kdidtlisnn": 0, "ha": [0, 1, 2], "lot": [0, 1], "same": [0, 1, 2], "my_chain_a": 0, "24199": 0, "38728": 0, "41820": 0, "made": 0, "squar": [0, 2], "bracket": 0, "asp": [0, 1], "leu": [0, 1], "20": [0, 1, 2], "pro": 0, "23": [0, 1], "40": [0, 2], "il": [0, 1], "If": [0, 1, 2, 4], "number": [0, 2], "isn": 0, "defin": [0, 2], "keyerror": 0, "rais": [0, 2], "traceback": 0, "most": [0, 1], "recent": 0, "last": 0, "ipython": 0, "22": [0, 1, 2], "52706ed7e5b3": 0, "anaconda3": 0, "lib": 0, "python3": 0, "site": [0, 2], "packag": [0, 3, 5], "py": 0, "__getitem__": 0, "self": [0, 2, 5], "190": 0, "isinst": 0, "str": [0, 2], "191": 0, "id_dict": 0, "m": [0, 1, 2], "_monom": 0, "192": 0, "193": 0, "elif": 0, "int": [0, 2], "194": 0, "my_residue_a23": 0, "ordereddict": [0, 1, 2], "special": [0, 2], "dictionari": [0, 2], "retain": [0, 2], "order": [0, 2], "add": [0, 2], "identifi": [0, 2], "make": [0, 1, 2, 4], "up": [0, 1, 2], "n": [0, 1, 2], "nitrogen": [0, 1], "coordin": [0, 1, 2, 5], "124": 0, "140": 0, "35": 0, "654": 0, "ca": [0, 1, 2], "carbon": [0, 1, 2], "664": 0, "954": 0, "34": 0, "292": 0, "c": [0, 1, 2, 4], "21": [0, 1], "911": [0, 1], "875": 0, "33": [0, 1], "515": 0, "o": [0, 1, 2], "oxygen": [0, 1], "863": 0, "926": 0, "32": [0, 1], "283": 0, "cb": [0, 1, 2], "24": [0, 1], "120": 0, "555": 0, "534": 0, "cg": [0, 1], "964": 0, "917": 0, "cd": [0, 1], "118": 0, "764": 0, "36": 0, "681": 0, "exampl": [0, 1, 2], "alpha": [0, 2, 5], "my_atom_a23ca": 0, "individu": 0, "Or": 0, "x": [0, 2, 5], "y": [0, 1, 2, 5], "z": [0, 2, 5], "162": 0, "hopefulli": 0, "see": [0, 1, 2], "travers": 0, "down": [0, 1, 2], "level": [0, 2], "work": [0, 2], "your": [0, 2], "back": [0, 2], "With": [0, 1], "find": [0, 1, 2], "origin": [0, 1, 2], "itself": [0, 1], "mean": [0, 2], "includ": [0, 1, 2], "own": 0, "ampal_par": 0, "true": [0, 1, 2], "discuss": 0, "intend": [0, 2], "store": [0, 2], "non": [0, 2], "extract": [0, 2], "get_ligand": [0, 1, 2, 5], "none": [0, 2], "exactli": [0, 1], "my_ligand": [0, 1], "zn": [0, 1], "221": 0, "without": 0, "present": [0, 2], "sinc": [0, 2], "both": [0, 2], "my_ligand_zinc": 0, "zinc": [0, 1], "817": [0, 1], "172": [0, 1], "18": [0, 1], "798": [0, 1], "associ": [0, 2], "reflect": 0, "simpl": [0, 1, 2, 3], "variou": [0, 2], "differ": 0, "pleas": [0, 1], "try": [0, 1], "plai": 0, "around": [0, 2], "builtin": 0, "either": [0, 1, 2], "document": [0, 1, 2], "base_amp": [0, 5], "folder": [0, 4], "through": [0, 1, 2], "tip": 0, "do": [0, 1, 2], "browser": 0, "next": [0, 2], "section": [0, 1, 2], "ll": [0, 1, 4], "how": [0, 1, 2], "perform": [0, 1, 2], "complex": [0, 1], "detail": [0, 1, 2], "There": [1, 2], "tool": [1, 2, 4], "them": [1, 2], "bake": 1, "object": [1, 2, 3], "themselv": 1, "let": 1, "pdb": [1, 2, 3, 5], "file": [1, 2, 3], "my_protein_3qy1": 1, "sometim": 1, "convini": 1, "an": [1, 2], "get_monom": [1, 2, 5], "itertool": [1, 2], "chain": [1, 2, 3], "0x7fbf649db588": 1, "As": 1, "might": 1, "slightli": 1, "confus": 1, "expect": [1, 2], "what": 1, "known": [1, 2], "iter": [1, 2], "loop": [1, 2], "over": [1, 2], "repeatedli": 1, "examin": [1, 2], "content": [1, 5], "convert": [1, 2, 3], "advantag": 1, "much": 1, "memori": 1, "effici": [1, 2], "could": [1, 2], "potenti": 1, "veri": [1, 2], "larg": 1, "d": [1, 2], "know": 1, "about": [1, 2], "relat": [1, 2], "follow": [1, 2], "link": 1, "quit": 1, "essenti": 1, "framework": [1, 3], "get_atom": [1, 2, 5], "show": 1, "clariti": 1, "14": 1, "714": 1, "30": 1, "168": 1, "26": 1, "423": 1, "15": [1, 2], "518": 1, "153": 1, "25": [1, 2], "207": 1, "16": 1, "111": 1, "28": 1, "769": 1, "931": 1, "960": 1, "27": 1, "855": 1, "734": 1, "613": 1, "31": 1, "220": 1, "270": 1, "067": 1, "624": 1, "od1": [1, 2], "899": 1, "777": 1, "743": 1, "od2": 1, "807": 1, "576": 1, "474": 1, "782": 1, "637": [1, 2], "789": 1, "17": 1, "360": 1, "364": 1, "339": 1, "466": 1, "876": 1, "299": 1, "513": 1, "674": 1, "586": 1, "842": 1, "509": 1, "862": 1, "cg1": 1, "643": 1, "442": 1, "889": 1, "cg2": 1, "956": 1, "516": 1, "456": 1, "cd1": 1, "873": 1, "033": 1, "19": 1, "310": 1, "788": 1, "836": 1, "370": 1, "411": 1, "786": 1, "801": 1, "707": 1, "030": 1, "438": 1, "781": 1, "539": 1, "8": [1, 3], "7": [1, 2, 3], "thr": 1, "ser": 1, "asn": 1, "ala": [1, 2], "trp": 1, "9": 1, "ly": [1, 2], "met": 1, "val": 1, "glu": 1, "counter": [1, 2], "fed": 1, "commonli": 1, "count": 1, "occur": 1, "start": [1, 2], "collect": [1, 2], "11": 1, "l": [1, 2], "29": 1, "13": [1, 2], "w": [1, 2], "k": [1, 2], "v": [1, 2], "e": [1, 2], "p": [1, 2], "g": [1, 2], "f": [1, 2], "r": [1, 2], "h": [1, 2], "But": 1, "state": [1, 2, 5], "befor": [1, 2], "molecul": [1, 2], "ligand": [1, 3, 5], "mol_cod": [1, 2, 5], "comprehens": 1, "whichev": 1, "comfort": 1, "rel": [1, 2], "scroll": 1, "relev": [1, 2], "mol_codes_1": 1, "lig": 1, "append": [1, 2, 5], "hoh": 1, "mol_codes_2": 1, "re": [1, 4], "now": 1, "mol": [1, 2], "447": 1, "water": [1, 2], "ion": 1, "understand": 1, "close": [1, 2], "zinc_1": 1, "singl": [1, 2], "when": [1, 2], "measur": [1, 2], "3d": [1, 2], "vector": [1, 2, 5], "form": [1, 2], "tupl": [1, 2], "179990720367528": 1, "first_ca": 1, "angstrom": [1, 2], "41060972200408": 1, "\u00e5": 1, "atoms_close_to_zinc": 1, "sulfur": 1, "sg": 1, "322": 1, "933": 1, "640": 1, "771": 1, "057": 1, "213": 1, "ne2": 1, "209": 1, "569": 1, "787": 1, "753": 1, "619": 1, "709": [1, 2], "One": [1, 2], "get": [1, 2], "ignor": 1, "abov": [1, 2], "block": 1, "fals": [1, 2], "flag": [1, 2], "sulphur": 1, "parent": [1, 2, 3, 5], "cy": 1, "residues_close_to_zinc": 1, "10": [1, 2], "hi": 1, "concis": 1, "residues_close_to_zinc_2": 1, "cystein": 1, "aspart": 1, "histidin": 1, "kind": [1, 2], "oper": [1, 2], "common": [1, 2], "built": 1, "handl": 1, "is_within": [1, 2, 5], "point": [1, 2, 5], "278": 1, "112": 1, "445": 1, "525": 1, "089": 1, "676": 1, "655": 1, "177": 1, "12": [1, 2], "255": 1, "337": 1, "223": 1, "972": 1, "752": 1, "047": 1, "188": 1, "oe1": 1, "046": 1, "618": 1, "265": 1, "nd2": 1, "667": 1, "empti": [1, 2], "noth": [1, 2], "partner": 1, "everi": [1, 2], "close_monom": [1, 2, 5], "given": [1, 2], "cutoff": [1, 2], "valu": [1, 2], "default": [1, 2, 4], "arg": [1, 2], "gly": 1, "rang": [1, 2], "geometr": [1, 2], "ve": 1, "alreadi": [1, 2], "cover": 1, "angle_between_vector": [1, 2, 5], "dihedr": [1, 2, 5], "unit_vector": [1, 2, 5], "find_foot": [1, 2, 5], "radius_of_circumcircl": [1, 2, 5], "Be": 1, "sure": [1, 4], "out": [1, 2], "sourc": [1, 2, 3], "probabl": 1, "torsion": [1, 2, 5], "angl": [1, 2], "requir": [1, 2], "calcul": [1, 2], "again": 1, "numpi": [1, 2], "arrai": [1, 2, 5], "note": [1, 2], "tag": [1, 2, 5], "tutori": 1, "proper": 1, "low": 1, "effort": 1, "r1": 1, "r2": 1, "r3": 1, "omega": [1, 2], "phi": [1, 2], "psi": [1, 2], "print": [1, 2], "179": 1, "8002001034783": 1, "64": 1, "10861163046212": 1, "45": 1, "84373968479124": 1, "chi": [1, 2], "too": [1, 2], "leucin": 1, "chi_1": 1, "chi_2": 1, "186": 1, "004": 1, "362": 1, "985": 1, "346": 1, "650": 1, "434": 1, "418": 1, "770": 1, "838": 1, "437": 1, "061": 1, "653": 1, "831": 1, "cd2": 1, "478": 1, "359": 1, "736": 1, "chi1": [1, 2], "chi2": [1, 2], "51": 1, "38040366349739": 1, "70006206564494": 1, "our": [1, 2], "gauch": 1, "tran": 1, "conform": [1, 2], "combin": 1, "mani": [1, 2], "ci": 1, "peptid": [1, 2], "bond": [1, 2], "anoth": [1, 2], "class": 2, "mmcalign": [2, 5], "eval_fn": 2, "eval_arg": 2, "protocol": 2, "metropoli": 2, "mont": 2, "carlo": 2, "super": 2, "ineffici": 2, "due": 2, "TO": 2, "THE": 2, "deepcopi": 2, "plan": 2, "improv": 2, "atom": [2, 3, 5], "record": 2, "rotat": [2, 5], "translat": [2, 5], "led": 2, "paramet": 2, "float": 2, "evalu": 2, "qualiti": 2, "fit": 2, "static": 2, "unpack": 2, "_i": 2, "_": 2, "check_mov": [2, 5], "new": 2, "old": 2, "accept": 2, "float_f": [2, 5], "format": 2, "std": 2, "start_optimis": [2, 5], "round": 2, "max_angl": 2, "max_dist": 2, "temp": 2, "298": 2, "stop_when": 2, "verbos": 2, "move": 2, "maximum": 2, "variat": 2, "per": 2, "step": 2, "distanc": [2, 3, 5], "option": 2, "temperatur": 2, "dure": 2, "process": 2, "stop": 2, "energi": 2, "less": 2, "than": 2, "equal": 2, "align_backbon": [2, 5], "refer": [2, 3], "mobil": 2, "add_amino_acid_to_json": [2, 5], "descript": 2, "letter": 2, "modifi": 2, "force_add": 2, "json": 2, "popul": 2, "tabl": 2, "ad": 2, "amid": 2, "termin": 2, "carboxi": 2, "group": 2, "bool": 2, "write": 2, "exist": 2, "ioerror": 2, "get_aa_cod": [2, 5], "aa_lett": 2, "three": 2, "aa": 2, "later": 2, "filesystem": 2, "aa_cod": 2, "get_aa_info": [2, 5], "inform": 2, "databas": 2, "sent": 2, "fill": 2, "row": 2, "unabl": 2, "locat": 2, "page": [2, 3], "name": 2, "pdbe": 2, "aa_dict": 2, "kei": 2, "aminoaciddb": 2, "field": 2, "scrape": 2, "found": 2, "dict": 2, "get_aa_lett": [2, 5], "except": 2, "dependencynotfoundwarn": [2, 5], "runtimewarn": 2, "malformedpdbwarn": [2, 5], "noncanonicalwarn": [2, 5], "notparameterisedwarn": [2, 5], "check_avail": [2, 5], "program": 2, "test_func": 2, "global_set": 2, "alpha_angl": [2, 5], "reference_axi": 2, "reference_axis_nam": 2, "ref_axi": 2, "primit": [2, 5], "helix": 2, "axi": [2, 5], "final": 2, "pair": 2, "along": 2, "generalis": 2, "tilt": 2, "coil": 2, "respect": 2, "central": 2, "length": [2, 5], "must": 2, "3": [2, 3], "residu": [2, 3, 5], "degre": 2, "valueerror": 2, "unequ": 2, "cc_to_local_param": [2, 5], "pitch": [2, 5], "radiu": [2, 5], "oligo": 2, "local": 2, "oligomer": 2, "pitchloc": 2, "between": 2, "2": [2, 3], "adjac": 2, "compon": 2, "helic": 2, "rloc": 2, "alphaloc": 2, "charge_seri": [2, 5], "seq": 2, "granular": 2, "1": [2, 3], "charg": 2, "ph": 2, "classify_angle_as_rotam": [2, 5], "classifi": 2, "rotam": 2, "convent": 2, "1006": 2, "jmbi": 2, "1993": 2, "1170": 2, "dunbrack": 2, "jr": 2, "karplu": 2, "backbon": [2, 5], "depend": 2, "librari": 2, "applic": 2, "side": 2, "predict": 2, "crick_angl": [2, 5], "crick": 2, "cr_angl": 2, "flip_reference_axis_if_antiparallel": [2, 5], "start_index": 2, "end_index": 2, "flip": 2, "direct": 2, "oppos": 2, "90": 2, "revers": 2, "run": 2, "polymer_to_reference_axis_dist": [2, 5], "For": 2, "end": [2, 5], "indic": 2, "chain_vector": 2, "terminu": 2, "make_primit": [2, 5], "cas_coord": 2, "window_length": 2, "averag": 2, "fix": 2, "set": 2, "time": 2, "s_primit": 2, "smaller": 2, "make_primitive_extrapolate_end": [2, 5], "smoothing_level": 2, "smooth": 2, "extrapol": 2, "lost": 2, "dictat": 2, "higher": 2, "smoother": 2, "curv": [2, 5], "increas": 2, "To": 2, "correct": 2, "approxim": 2, "trade": 2, "off": 2, "accuraci": 2, "final_primit": 2, "make_primitive_smooth": [2, 5], "great": 2, "compar": 2, "measure_sidechain_torsion_angl": [2, 5], "sidechain": 2, "tell": 2, "doe": 2, "chi_angl": 2, "pi": 2, "chi3": 2, "chi4": 2, "accord": 2, "measure_torsion_angl": [2, 5], "torsion_angl": 2, "tripl": 2, "partial_charg": [2, 5], "partial": 2, "interest": 2, "correspond": 2, "barrel": 2, "centr": 2, "care": 2, "orient": 2, "ref": 2, "polypeptide_vector": [2, 5], "unit": 2, "chang": 2, "magnitud": 2, "shape": 2, "reference_axis_from_chain": [2, 5], "would": 2, "give": 2, "residues_per_turn": [2, 5], "turn": 2, "posit": 2, "rpt": 2, "sequence_charg": [2, 5], "4": [2, 3], "total": 2, "sequence_isoelectric_point": [2, 5], "isoelectr": 2, "sequence_molar_extinction_280": [2, 5], "molar": 2, "extinct": 2, "coeffici": 2, "280": 2, "nm": 2, "cm": 2, "sequence_molecular_weight": [2, 5], "molecular": 2, "weight": 2, "dalton": 2, "ampalcontain": [2, 5], "ampal_object": 2, "custom": 2, "hold": 2, "multipl": 2, "score": 2, "initialis": 2, "extend": [2, 5], "ampal_contain": 2, "compil": [2, 4], "sort_by_tag": [2, 5], "sort": 2, "assembly_id": 2, "baseamp": [2, 5], "hierarchi": 2, "assembl": 2, "user": 2, "typeerror": 2, "metadata": 2, "preserv": 2, "bb_assembli": 2, "fasta": [2, 5], "explan": 2, "http": 2, "en": 2, "wikipedia": 2, "org": 2, "wiki": 2, "fasta_format": 2, "recommend": 2, "line": 2, "text": 2, "shorter": 2, "80": 2, "charact": 2, "adher": 2, "pdbid": 2, "consist": 2, "download": [2, 4], "uppercas": 2, "feed": 2, "cdhit": 2, "cluster": 2, "fasta_str": 2, "filter_mol_typ": [2, 5], "mol_typ": 2, "pseudo_group": 2, "inc_alt_st": 2, "flat": 2, "altern": 2, "solvent": 2, "retriev": 2, "pseudo": 2, "cutoff_dist": 2, "within": [2, 3], "cut": 2, "make_pdb": [2, 5], "alt_stat": 2, "header": 2, "footer": 2, "output": 2, "pdb_str": 2, "collat": 2, "co": 2, "mode": 2, "prim_assembli": 2, "relabel_al": [2, 5], "relabel": 2, "label": 2, "relabel_atom": [2, 5], "numer": 2, "offset": 2, "relabel_monom": [2, 5], "relabel_polym": [2, 5], "alphabet": 2, "match": 2, "tag_ca_geometri": [2, 5], "forc": 2, "tag_torsion_angl": [2, 5], "atom_id": 2, "res_label": [2, 5], "occup": 2, "bfactor": 2, "inherit": 2, "space": 2, "usual": 2, "cach": 2, "radian": 2, "li": 2, "upon": 2, "unique_id": [2, 5], "creat": 2, "uniqu": 2, "notimplementederror": 2, "abstract": 2, "instanci": 2, "child": 2, "centre_of_mass": [2, 5], "mass": 2, "manual": 2, "select": [2, 3], "rmsd": [2, 5], "No": 2, "monomer_id": 2, "maintain": 2, "were": 2, "avail": [2, 4], "active_st": [2, 5], "should": 2, "activ": [2, 3], "subclass": 2, "search": 2, "nearby_residu": 2, "interfac": 2, "polymer_id": 2, "molecule_typ": [2, 5], "sl": [2, 5], "ligandgroup": [2, 5], "etc": 2, "updat": 2, "rather": 2, "index": [2, 3], "get_reference_coord": [2, 5], "ref_coord": 2, "second": 2, "third": 2, "th": 2, "inc_ligand": 2, "monoer": 2, "cap": [2, 5], "shorten": 2, "certain": 2, "centre_of_atom": [2, 5], "mass_weight": 2, "otherwis": 2, "find_atoms_within_dist": [2, 5], "filtered_atom": 2, "filter": 2, "write_pdb": [2, 5], "chain_id": 2, "strip_stat": 2, "els": 2, "primari": 2, "remov": 2, "provid": 2, "kabsch": 2, "sander": 2, "1983": 2, "secondari": 2, "pattern": 2, "recognit": 2, "hydrogen": 2, "biopolym": 2, "2577": 2, "dssp_avail": [2, 5], "mkdssp": 2, "extract_all_ss_dssp": [2, 5], "in_dssp": 2, "dssp_residu": 2, "intern": 2, "5": [2, 3], "6": [2, 3], "find_ss_region": [2, 5], "loop_assign": 2, "separ": 2, "strand": 2, "gather": 2, "fragment": 2, "continu": 2, "region": 2, "innermost": 2, "get_ss_region": [2, 5], "ss_type": 2, "specifi": 2, "run_dssp": [2, 5], "param": 2, "dssp_out": 2, "tag_dssp_data": [2, 5], "dssp_data": 2, "definit": 2, "place": 2, "basic": [2, 3], "preced": 2, "underscor": 2, "howev": 2, "cdef": 2, "anywai": 2, "straight": 2, "arc_length": [2, 5], "t_ref": 2, "arc": 2, "t0": 2, "trivial": 2, "particular": 2, "compat": 2, "neg": 2, "direction_vector": [2, 5], "distance_to_point": [2, 5], "get_coord": [2, 5], "n_point": 2, "52": 2, "evenli": 2, "coord": 2, "midpoint": [2, 5], "t_from_arc_length": [2, 5], "unit_binorm": [2, 5], "binorm": 2, "tangent": 2, "normal": 2, "unit_norm": [2, 5], "perpendicular": 2, "curvatur": [2, 5], "zero": 2, "differential_geometry_of_curv": 2, "special_frenet_vectors_and_generalized_curvatur": 2, "here": 2, "chosen": 2, "parallel": 2, "unit_tang": [2, 5], "appl": 2, "help": 2, "think": 2, "right": 2, "hand": 2, "rule": 2, "upload": 2, "wikimedia": 2, "thumb": 2, "d2": 2, "right_hand_rule_cross_product": 2, "svg": 2, "2000px": 2, "png": 2, "cross_product": 2, "finger": 2, "middl": 2, "Not": 2, "instanti": 2, "helicalcurv": [2, 5], "handed": 2, "parameteris": 2, "classmethod": 2, "alpha_and_pitch": [2, 5], "alpha_and_radiu": [2, 5], "axis_start": 2, "axis_end": 2, "wind": 2, "pitch_and_radiu": [2, 5], "togeth": 2, "oscul": 2, "plane": 2, "articl": 2, "quaternion": [2, 5], "j": 2, "is_rot": 2, "angle_and_axi": [2, 5], "apply_conjug": [2, 5], "as_arrai": [2, 5], "as_rot": [2, 5], "conjug": [2, 5], "extract_angle_and_axi": [2, 5], "norm": [2, 5], "real_and_vector": [2, 5], "rotate_vector": [2, 5], "dot": 2, "product": 2, "tools_geometri": 2, "1e": 2, "v1": 2, "v2": 2, "doubl": 2, "apply_unit_quaternion": [2, 5], "appli": 2, "u_in": 2, "imaginari": 2, "real": 2, "cartesian_to_cylindr": [2, 5], "cartesian": 2, "cylindr": 2, "system": 2, "cylindrical_to_cartesian": [2, 5], "azimuth": 2, "instead": 2, "cartesian_to_spher": [2, 5], "spheric": 2, "spherical_to_cartesian": [2, 5], "zenith": 2, "uniform": 2, "suppli": 2, "com": 2, "len": 2, "closest_dist": [2, 5], "minimun": 2, "pairwis": 2, "points1": 2, "points2": 2, "minimum": 2, "min_dist": 2, "point2": 2, "closest_points_on_lin": [2, 5], "p0": 2, "p1": 2, "q0": 2, "q1": 2, "describ": 2, "l1": 2, "l2": 2, "infinit": 2, "extens": 2, "pass": 2, "point_1": 2, "point_2": 2, "adapt": 2, "geomalgorithm": 2, "a07": 2, "_distanc": 2, "html": 2, "where": 2, "segment": 2, "lie": 2, "clockwis": 2, "being": 2, "180": 2, "antiparallel": 2, "height": 2, "cylind": 2, "cylindrical_coordinate_system": 2, "mathworld": 2, "wolfram": 2, "cylindricalcoordin": 2, "four": 2, "dihedral_angl": 2, "atan2": 2, "colinear": 2, "euclidian": 2, "dist": [2, 5], "euclidean": 2, "foot": 2, "ab": 2, "shortest": 2, "linedistance3": 2, "dimension": 2, "math": 2, "solv": 2, "substitut": 2, "yield": 2, "desir": 2, "closest": 2, "find_foot_on_plan": [2, 5], "cannot": 2, "collinear": 2, "stackexchang": 2, "question": 2, "723937": 2, "3x": 2, "4y": 2, "problem": 2, "solut": 2, "lagrangian": 2, "multipli": 2, "lagrange_multipli": 2, "easier": 2, "subject": 2, "constraint": 2, "equat": 2, "langrangian": 2, "lambda": 2, "deriviatv": 2, "wrt": 2, "lagrangian_lambda": 2, "Then": 2, "find_limit": [2, 5], "find_transform": [2, 5], "transform": 2, "s1": 2, "e1": 2, "s2": 2, "e2": 2, "after": 2, "been": 2, "onto": 2, "gen_sector": [2, 5], "overlap": 2, "sector": 2, "intersection_start_and_end": [2, 5], "intersect": 2, "bundl": 2, "idea": 2, "join": 2, "wa": 2, "prinicip": 2, "ten": 2, "1000": 2, "swap": 2, "599": 2, "is_acut": [2, 5], "minimal_distance_between_lin": [2, 5], "minim": 2, "construct": 2, "connect": 2, "skew_lin": 2, "points_on_a_circl": [2, 5], "uniformli": 2, "distribut": 2, "circumfer": 2, "circl": 2, "anti": 2, "circumcircl": 2, "triangl": 2, "www": 2, "analyzemath": 2, "circumcircleradiu": 2, "traingl": 2, "though": 2, "counterpart": 2, "rotation_matrix": [2, 5], "matrix": 2, "unit_vect": 2, "head": 2, "spherical_coordinate_system": 2, "sphericalcoordin": 2, "regardless": 2, "normalis": 2, "v_out": 2, "zerodivisionerror": 2, "chemic": 2, "covalentbond": [2, 5], "coval": 2, "involv": 2, "hydrogenbond": [2, 5], "donor": [2, 5], "acceptor": [2, 5], "ang_d": [2, 5], "ang_a": [2, 5], "noncovalentinteract": [2, 5], "term": 2, "acceptor_monom": [2, 5], "donor_monom": [2, 5], "member": 2, "covalent_bond": [2, 5], "threshold": 2, "element_data": 2, "directli": 2, "deviat": 2, "ideal": 2, "further": 2, "find_covalent_bond": [2, 5], "max_rang": 2, "size": 2, "longer": 2, "under": 2, "generate_bond_subgraphs_from_break": [2, 5], "bond_graph": 2, "atom1": 2, "atom2": 2, "split": 2, "graph": 2, "subgraph": 2, "cycl": 2, "networkx": 2, "network": 2, "broken": 2, "generate_covalent_bond_graph": [2, 5], "insertion_cod": [2, 5], "is_hetero": [2, 5], "dedic": 2, "insert": 2, "read": 2, "hetero": 2, "nucleic": 2, "nucleotid": [2, 5], "unk": 2, "polynucleotid": [2, 5], "reference_atom": [2, 5], "pdbparser": [2, 5], "pdb_id": 2, "ignore_end": [2, 5], "encount": 2, "proc_funct": [2, 5], "pdb_line": [2, 5], "new_label": [2, 5], "multi": 2, "nmr": 2, "pdb_parse_tre": [2, 5], "tree": 2, "entri": 2, "current_lin": [2, 5], "change_st": [2, 5], "increment": 2, "check_for_non_canon": [2, 5], "canon": 2, "gen_stat": [2, 5], "monomer_data": 2, "make_amp": [2, 5], "Will": 2, "pares": 2, "parse_pdb_fil": [2, 5], "parser": 2, "proc_atom": [2, 5], "hetatm": 2, "proc_chain": [2, 5], "chain_info": 2, "unknown": 2, "attributeerror": 2, "proc_line_coordin": [2, 5], "column": 2, "proc_monom": [2, 5], "monomer_info": 2, "mon_cl": 2, "explicitli": 2, "proc_stat": [2, 5], "state_data": 2, "state_id": 2, "bb_poli": 2, "backbone_bond_angl": [2, 5], "bond_angl": 2, "n_ca_c": 2, "ca_c_o": 2, "ca_c_n": 2, "c_n_ca": 2, "across": 2, "therefor": 2, "success": 2, "backbone_bond_length": [2, 5], "bond_length": 2, "n_ca": 2, "ca_c": 2, "c_o": 2, "c_n": 2, "c_join": [2, 5], "76": 2, "178": 2, "65": 2, "07": 2, "o_c_n_angl": 2, "c_n_ca_angl": 2, "c_n_length": 2, "via": 2, "taken": 2, "ideal_backbone_bond_angl": 2, "ideal_backbone_bond_length": 2, "get_slice_from_res_id": [2, 5], "slice_polym": 2, "n_join": [2, 5], "frame": 2, "radii_of_curvatur": [2, 5], "rise_per_residu": [2, 5], "rise": 2, "rubric": 2, "radius_of_curvatur": 2, "tag_sidechain_dihedr": [2, 5], "alanin": 2, "glycin": 2, "valid_backbone_bond_angl": [2, 5], "atol": 2, "schulz": 2, "heiner": 2, "schirmer": 2, "principl": 2, "Of": 2, "york": 2, "springer": 2, "verlag": 2, "1979": 2, "toler": 2, "absolut": 2, "awai": 2, "valid_backbone_bond_length": [2, 5], "angstom": 2, "bb_monom": 2, "indexerror": 2, "centroid": [2, 5], "unless": 2, "huang": 2, "et": 2, "al": 2, "subbiah": 2, "levitt": 2, "1995": 2, "recogn": 2, "nativ": 2, "fold": 2, "arrang": 2, "hydrophob": 2, "polar": 2, "biol": 2, "252": 2, "720": 2, "side_chain": [2, 5], "side_chain_atom": 2, "distinct": 2, "biopython": 2, "whitespac": 2, "h_": 2, "plu": 2, "h_glc": 2, "glucos": 2, "the_biopython_structural_bioinformatics_faq": 2, "target": 2, "target_i": 2, "mobile_i": 2, "flat_list_to_dummy_chain": [2, 5], "atom_list": 2, "atom_group_": 2, "dummi": 2, "ordin": 2, "dum": 2, "flat_list_to_polym": [2, 5], "pseudogroup": [2, 5], "compos": 2, "pseudoatom": [2, 5], "psuedomonom": 2, "from_coordin": [2, 5], "primitv": 2, "pseudomonom": [2, 5], "psuedoatom": 2, "welcom": 3, "biomolecular": 3, "instal": 3, "introduct": 3, "analysi": 3, "summari": 3, "advanc": 3, "composit": 3, "geometri": [3, 5], "api": 3, "ampal": 4, "pip": 4, "clone": 4, "repositori": 4, "navig": 4, "cython": 4, "furthermor": 4, "mac": 4, "linux": 4, "visual": 4, "build": 4, "submodul": 5, "align": 5, "amino_acid": 5, "ampal_warn": 5, "analyse_protein": 5, "data": 5, "dssp": 5, "interact": 5, "nucleic_acid": 5, "pdb_parser": 5, "pseudo_atom": 5}, "objects": {"": [[2, 0, 0, "-", "ampal"]], "ampal": [[2, 0, 0, "-", "align"], [2, 0, 0, "-", "amino_acids"], [2, 0, 0, "-", "ampal_warnings"], [2, 0, 0, "-", "analyse_protein"], [2, 0, 0, "-", "assembly"], [2, 0, 0, "-", "base_ampal"], [2, 0, 0, "-", "data"], [2, 0, 0, "-", "dssp"], [2, 0, 0, "-", "geometry"], [2, 0, 0, "-", "interactions"], [2, 0, 0, "-", "ligands"], [2, 0, 0, "-", "nucleic_acid"], [2, 0, 0, "-", "pdb_parser"], [2, 0, 0, "-", "protein"], [2, 0, 0, "-", "pseudo_atoms"]], "ampal.align": [[2, 1, 1, "", "MMCAlign"], [2, 3, 1, "", "align_backbones"]], "ampal.align.MMCAlign": [[2, 2, 1, "", "check_move"], [2, 2, 1, "", "float_f"], [2, 2, 1, "", "start_optimisation"]], "ampal.amino_acids": [[2, 3, 1, "", "add_amino_acid_to_json"], [2, 3, 1, "", "get_aa_code"], [2, 3, 1, "", "get_aa_info"], [2, 3, 1, "", "get_aa_letter"]], "ampal.ampal_warnings": [[2, 4, 1, "", "DependencyNotFoundWarning"], [2, 4, 1, "", "MalformedPDBWarning"], [2, 4, 1, "", "NoncanonicalWarning"], [2, 4, 1, "", "NotParameterisedWarning"], [2, 3, 1, "", "check_availability"]], "ampal.analyse_protein": [[2, 3, 1, "", "alpha_angles"], [2, 3, 1, "", "cc_to_local_params"], [2, 3, 1, "", "charge_series"], [2, 3, 1, "", "classify_angle_as_rotamer"], [2, 3, 1, "", "crick_angles"], [2, 3, 1, "", "flip_reference_axis_if_antiparallel"], [2, 3, 1, "", "make_primitive"], [2, 3, 1, "", "make_primitive_extrapolate_ends"], [2, 3, 1, "", "make_primitive_smoothed"], [2, 3, 1, "", "measure_sidechain_torsion_angles"], [2, 3, 1, "", "measure_torsion_angles"], [2, 3, 1, "", "partial_charge"], [2, 3, 1, "", "polymer_to_reference_axis_distances"], [2, 3, 1, "", "polypeptide_vector"], [2, 3, 1, "", "reference_axis_from_chains"], [2, 3, 1, "", "residues_per_turn"], [2, 3, 1, "", "sequence_charge"], [2, 3, 1, "", "sequence_isoelectric_point"], [2, 3, 1, "", "sequence_molar_extinction_280"], [2, 3, 1, "", "sequence_molecular_weight"]], "ampal.assembly": [[2, 1, 1, "", "AmpalContainer"], [2, 1, 1, "", "Assembly"]], "ampal.assembly.AmpalContainer": [[2, 2, 1, "", "append"], [2, 2, 1, "", "extend"], [2, 5, 1, "", "id"], [2, 6, 1, "", "pdb"], [2, 2, 1, "", "sort_by_tag"]], "ampal.assembly.Assembly": [[2, 2, 1, "", "append"], [2, 6, 1, "", "backbone"], [2, 2, 1, "", "extend"], [2, 6, 1, "", "fasta"], [2, 2, 1, "", "filter_mol_types"], [2, 2, 1, "", "get_atoms"], [2, 2, 1, "", "get_ligands"], [2, 2, 1, "", "get_monomers"], [2, 2, 1, "", "is_within"], [2, 6, 1, "", "isoelectric_point"], [2, 2, 1, "", "make_pdb"], [2, 6, 1, "", "molar_extinction_280"], [2, 6, 1, "", "molecular_weight"], [2, 6, 1, "", "pdb"], [2, 6, 1, "", "primitives"], [2, 2, 1, "", "relabel_all"], [2, 2, 1, "", "relabel_atoms"], [2, 2, 1, "", "relabel_monomers"], [2, 2, 1, "", "relabel_polymers"], [2, 6, 1, "", "sequences"], [2, 2, 1, "", "tag_ca_geometry"], [2, 2, 1, "", "tag_torsion_angles"]], "ampal.base_ampal": [[2, 1, 1, "", "Atom"], [2, 1, 1, "", "BaseAmpal"], [2, 1, 1, "", "Monomer"], [2, 1, 1, "", "Polymer"], [2, 3, 1, "", "cap"], [2, 3, 1, "", "centre_of_atoms"], [2, 3, 1, "", "find_atoms_within_distance"], [2, 3, 1, "", "write_pdb"]], "ampal.base_ampal.Atom": [[2, 6, 1, "", "array"], [2, 5, 1, "", "element"], [2, 5, 1, "", "id"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "res_label"], [2, 2, 1, "", "rotate"], [2, 5, 1, "", "tags"], [2, 2, 1, "", "translate"], [2, 6, 1, "", "unique_id"], [2, 6, 1, "", "x"], [2, 6, 1, "", "y"], [2, 6, 1, "", "z"]], "ampal.base_ampal.BaseAmpal": [[2, 6, 1, "", "centre_of_mass"], [2, 2, 1, "", "get_atoms"], [2, 2, 1, "", "is_within"], [2, 2, 1, "", "make_pdb"], [2, 6, 1, "", "pdb"], [2, 2, 1, "", "rmsd"], [2, 2, 1, "", "rotate"], [2, 2, 1, "", "translate"]], "ampal.base_ampal.Monomer": [[2, 6, 1, "", "active_state"], [2, 6, 1, "", "atoms"], [2, 2, 1, "", "close_monomers"], [2, 2, 1, "", "get_atoms"], [2, 2, 1, "", "get_monomers"], [2, 5, 1, "", "id"], [2, 2, 1, "", "make_pdb"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "states"], [2, 5, 1, "", "tags"]], "ampal.base_ampal.Polymer": [[2, 2, 1, "", "append"], [2, 2, 1, "", "extend"], [2, 2, 1, "", "get_atoms"], [2, 2, 1, "", "get_monomers"], [2, 2, 1, "", "get_reference_coords"], [2, 5, 1, "", "id"], [2, 5, 1, "", "ligands"], [2, 2, 1, "", "make_pdb"], [2, 5, 1, "", "molecule_type"], [2, 5, 1, "", "parent"], [2, 2, 1, "", "relabel_all"], [2, 2, 1, "", "relabel_atoms"], [2, 2, 1, "", "relabel_monomers"], [2, 5, 1, "", "sl"], [2, 5, 1, "", "tags"]], "ampal.dssp": [[2, 3, 1, "", "dssp_available"], [2, 3, 1, "", "extract_all_ss_dssp"], [2, 3, 1, "", "find_ss_regions"], [2, 3, 1, "", "get_ss_regions"], [2, 3, 1, "", "run_dssp"], [2, 3, 1, "", "tag_dssp_data"]], "ampal.geometry": [[2, 1, 1, "", "Axis"], [2, 1, 1, "", "Curve"], [2, 1, 1, "", "HelicalCurve"], [2, 1, 1, "", "Quaternion"], [2, 3, 1, "", "angle_between_vectors"], [2, 3, 1, "", "apply_unit_quaternion"], [2, 3, 1, "", "cartesian_to_cylindrical"], [2, 3, 1, "", "cartesian_to_spherical"], [2, 3, 1, "", "centre_of_mass"], [2, 3, 1, "", "closest_distance"], [2, 3, 1, "", "closest_points_on_lines"], [2, 3, 1, "", "cylindrical_to_cartesian"], [2, 3, 1, "", "dihedral"], [2, 3, 1, "", "distance"], [2, 3, 1, "", "find_foot"], [2, 3, 1, "", "find_foot_on_plane"], [2, 3, 1, "", "find_limits"], [2, 3, 1, "", "find_transformations"], [2, 3, 1, "", "gen_sectors"], [2, 3, 1, "", "intersection_start_and_end"], [2, 3, 1, "", "is_acute"], [2, 3, 1, "", "minimal_distance_between_lines"], [2, 3, 1, "", "points_on_a_circle"], [2, 3, 1, "", "radius_of_circumcircle"], [2, 3, 1, "", "rmsd"], [2, 3, 1, "", "rotation_matrix"], [2, 3, 1, "", "spherical_to_cartesian"], [2, 3, 1, "", "unit_vector"]], "ampal.geometry.Axis": [[2, 2, 1, "", "arc_length"], [2, 6, 1, "", "direction_vector"], [2, 2, 1, "", "distance_to_point"], [2, 2, 1, "", "get_coords"], [2, 6, 1, "", "length"], [2, 6, 1, "", "midpoint"], [2, 2, 1, "", "point"], [2, 2, 1, "", "t_from_arc_length"], [2, 6, 1, "", "unit_binormal"], [2, 6, 1, "", "unit_normal"], [2, 6, 1, "", "unit_tangent"]], "ampal.geometry.Curve": [[2, 2, 1, "", "point"]], "ampal.geometry.HelicalCurve": [[2, 6, 1, "", "alpha"], [2, 2, 1, "", "alpha_and_pitch"], [2, 2, 1, "", "alpha_and_radius"], [2, 2, 1, "", "arc_length"], [2, 6, 1, "", "axis"], [2, 2, 1, "", "curvature"], [2, 2, 1, "", "get_coords"], [2, 6, 1, "", "pitch"], [2, 2, 1, "", "pitch_and_radius"], [2, 2, 1, "", "point"], [2, 6, 1, "", "radius"], [2, 2, 1, "", "t_from_arc_length"], [2, 2, 1, "", "torsion"], [2, 2, 1, "", "unit_binormal"], [2, 2, 1, "", "unit_normal"], [2, 2, 1, "", "unit_tangent"], [2, 2, 1, "", "x"], [2, 2, 1, "", "y"], [2, 2, 1, "", "z"]], "ampal.geometry.Quaternion": [[2, 2, 1, "", "angle_and_axis"], [2, 2, 1, "", "apply_conjugation"], [2, 6, 1, "", "as_array"], [2, 2, 1, "", "as_rotation"], [2, 6, 1, "", "conjugate"], [2, 2, 1, "", "distance"], [2, 2, 1, "", "extract_angle_and_axis"], [2, 6, 1, "", "norm"], [2, 2, 1, "", "real_and_vector"], [2, 2, 1, "", "rotate_vector"], [2, 6, 1, "", "vector"]], "ampal.interactions": [[2, 1, 1, "", "CovalentBond"], [2, 1, 1, "", "HydrogenBond"], [2, 1, 1, "", "Interaction"], [2, 1, 1, "", "NonCovalentInteraction"], [2, 3, 1, "", "covalent_bonds"], [2, 3, 1, "", "find_covalent_bonds"], [2, 3, 1, "", "generate_bond_subgraphs_from_break"], [2, 3, 1, "", "generate_covalent_bond_graph"]], "ampal.interactions.CovalentBond": [[2, 6, 1, "", "a"], [2, 6, 1, "", "b"]], "ampal.interactions.HydrogenBond": [[2, 5, 1, "", "acceptor"], [2, 6, 1, "", "acceptor_monomer"], [2, 5, 1, "", "ang_a"], [2, 5, 1, "", "ang_d"], [2, 5, 1, "", "dist"], [2, 5, 1, "", "donor"], [2, 6, 1, "", "donor_monomer"]], "ampal.interactions.Interaction": [[2, 5, 1, "", "a"], [2, 5, 1, "", "b"], [2, 5, 1, "", "dist"]], "ampal.interactions.NonCovalentInteraction": [[2, 6, 1, "id0", "acceptor"], [2, 5, 1, "", "dist"], [2, 6, 1, "id7", "donor"]], "ampal.ligands": [[2, 1, 1, "", "Ligand"], [2, 1, 1, "", "LigandGroup"]], "ampal.ligands.Ligand": [[2, 5, 1, "", "atoms"], [2, 5, 1, "", "id"], [2, 5, 1, "", "insertion_code"], [2, 5, 1, "", "is_hetero"], [2, 5, 1, "", "mol_code"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "tags"]], "ampal.ligands.Ligand.self": [[2, 5, 1, "", "states"]], "ampal.nucleic_acid": [[2, 1, 1, "", "Nucleotide"], [2, 1, 1, "", "Polynucleotide"]], "ampal.nucleic_acid.Nucleotide": [[2, 5, 1, "", "id"], [2, 5, 1, "", "insertion_code"], [2, 5, 1, "", "is_hetero"], [2, 5, 1, "", "mol_code"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "reference_atom"], [2, 5, 1, "", "states"], [2, 5, 1, "", "tags"]], "ampal.nucleic_acid.Polynucleotide": [[2, 5, 1, "", "id"], [2, 5, 1, "", "ligands"], [2, 5, 1, "", "molecule_type"], [2, 5, 1, "", "parent"], [2, 6, 1, "", "sequence"], [2, 5, 1, "", "sl"], [2, 5, 1, "", "tags"]], "ampal.pdb_parser": [[2, 1, 1, "", "PdbParser"], [2, 3, 1, "", "load_pdb"]], "ampal.pdb_parser.PdbParser": [[2, 2, 1, "", "change_state"], [2, 2, 1, "", "check_for_non_canonical"], [2, 5, 1, "", "current_line"], [2, 2, 1, "", "end"], [2, 2, 1, "", "gen_states"], [2, 5, 1, "", "id"], [2, 5, 1, "", "ignore_end"], [2, 2, 1, "", "make_ampal"], [2, 5, 1, "", "new_labels"], [2, 2, 1, "", "parse_pdb_file"], [2, 5, 1, "", "pdb_lines"], [2, 5, 1, "", "pdb_parse_tree"], [2, 2, 1, "", "proc_atom"], [2, 2, 1, "", "proc_chain"], [2, 5, 1, "", "proc_functions"], [2, 2, 1, "", "proc_line_coordinate"], [2, 2, 1, "", "proc_monomer"], [2, 2, 1, "", "proc_state"], [2, 5, 1, "", "state"]], "ampal.protein": [[2, 1, 1, "", "Polypeptide"], [2, 1, 1, "", "Residue"], [2, 3, 1, "", "align"], [2, 3, 1, "", "flat_list_to_dummy_chain"], [2, 3, 1, "", "flat_list_to_polymer"]], "ampal.protein.Polypeptide": [[2, 6, 1, "", "backbone"], [2, 6, 1, "", "backbone_bond_angles"], [2, 6, 1, "", "backbone_bond_lengths"], [2, 2, 1, "", "c_join"], [2, 6, 1, "", "fasta"], [2, 2, 1, "", "get_slice_from_res_id"], [2, 5, 1, "", "id"], [2, 6, 1, "", "isoelectric_point"], [2, 5, 1, "", "ligands"], [2, 6, 1, "", "molar_extinction_280"], [2, 6, 1, "", "molecular_weight"], [2, 5, 1, "", "molecule_type"], [2, 2, 1, "", "n_join"], [2, 5, 1, "", "parent"], [2, 6, 1, "", "primitive"], [2, 2, 1, "", "radii_of_curvature"], [2, 2, 1, "", "rise_per_residue"], [2, 6, 1, "", "sequence"], [2, 5, 1, "", "sl"], [2, 2, 1, "", "tag_ca_geometry"], [2, 2, 1, "", "tag_sidechain_dihedrals"], [2, 2, 1, "", "tag_torsion_angles"], [2, 5, 1, "", "tags"], [2, 2, 1, "", "valid_backbone_bond_angles"], [2, 2, 1, "", "valid_backbone_bond_lengths"]], "ampal.protein.Residue": [[2, 6, 1, "", "backbone"], [2, 6, 1, "", "centroid"], [2, 5, 1, "", "id"], [2, 5, 1, "", "insertion_code"], [2, 5, 1, "", "is_hetero"], [2, 5, 1, "", "mol_code"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "reference_atom"], [2, 6, 1, "", "side_chain"], [2, 5, 1, "", "states"], [2, 5, 1, "", "tags"], [2, 6, 1, "", "unique_id"]], "ampal.pseudo_atoms": [[2, 1, 1, "", "Primitive"], [2, 1, 1, "", "PseudoAtom"], [2, 1, 1, "", "PseudoGroup"], [2, 1, 1, "", "PseudoMonomer"]], "ampal.pseudo_atoms.Primitive": [[2, 6, 1, "", "coordinates"], [2, 2, 1, "", "from_coordinates"], [2, 5, 1, "", "id"], [2, 5, 1, "", "insertion_code"], [2, 5, 1, "", "is_hetero"], [2, 5, 1, "", "mol_code"], [2, 5, 1, "", "parent"], [2, 2, 1, "", "radii_of_curvature"], [2, 5, 1, "", "reference_atom"], [2, 2, 1, "", "rise_per_residue"], [2, 5, 1, "", "states"], [2, 5, 1, "", "tags"]], "ampal.pseudo_atoms.PseudoAtom": [[2, 5, 1, "", "element"], [2, 5, 1, "", "id"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "res_label"], [2, 5, 1, "", "tags"]], "ampal.pseudo_atoms.PseudoGroup": [[2, 5, 1, "", "id"], [2, 5, 1, "", "ligands"], [2, 5, 1, "", "molecule_type"], [2, 5, 1, "", "parent"], [2, 5, 1, "", "sl"], [2, 5, 1, "", "tags"]], "ampal.pseudo_atoms.PseudoMonomer": [[2, 5, 1, "", "id"], [2, 5, 1, "", "insertion_code"], [2, 5, 1, "", "is_hetero"], [2, 5, 1, "", "mol_code"], [2, 5, 1, "", "parent"], [2, 6, 1, "", "pdb"], [2, 5, 1, "", "reference_atom"], [2, 5, 1, "", "states"], [2, 5, 1, "", "tags"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:function", "4": "py:exception", "5": "py:attribute", "6": "py:property"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "function", "Python function"], "4": ["py", "exception", "Python exception"], "5": ["py", "attribute", "Python attribute"], "6": ["py", "property", "Python property"]}, "titleterms": {"an": 0, "introduct": 0, "ampal": [0, 1, 2, 3, 5], "framework": 0, "1": [0, 1], "convert": 0, "pdb": 0, "file": 0, "object": 0, "2": [0, 1], "basic": [0, 4], "analysi": [0, 1], "3": [0, 1], "select": [0, 1], "chain": 0, "4": [0, 1], "residu": [0, 1], "5": [0, 1], "atom": [0, 1], "6": [0, 1], "parent": 0, "7": 0, "ligand": [0, 2], "8": 0, "summari": [0, 1], "activ": [0, 1], "advanc": 1, "all": 1, "analys": 1, "composit": 1, "distanc": 1, "i": 1, "within": 1, "geometri": [1, 2], "packag": 2, "submodul": 2, "align": 2, "modul": 2, "amino_acid": 2, "ampal_warn": 2, "analyse_protein": 2, "assembli": 2, "base_amp": 2, "data": 2, "dssp": 2, "interact": 2, "nucleic_acid": 2, "pdb_parser": 2, "protein": 2, "pseudo_atom": 2, "content": [2, 3], "document": 3, "indic": 3, "search": 3, "instal": 4, "from": 4, "sourc": 4}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.viewcode": 1, "sphinx": 57}, "alltitles": {"An Introduction to the AMPAL Framework": [[0, "an-introduction-to-the-ampal-framework"]], "1. Converting PDB files to AMPAL Objects": [[0, "converting-pdb-files-to-ampal-objects"]], "2. Basic Analysis": [[0, "basic-analysis"]], "3. Selecting Chains": [[0, "selecting-chains"]], "4. Selecting Residues": [[0, "selecting-residues"]], "5. Selecting Atoms": [[0, "selecting-atoms"]], "6. AMPAL Parents": [[0, "ampal-parents"]], "7. Ligands": [[0, "ligands"]], "8. Summary and activities": [[0, "summary-and-activities"]], "Advanced Selections and Analysis": [[1, "advanced-selections-and-analysis"]], "1. Selecting All Residues or Atoms": [[1, "selecting-all-residues-or-atoms"]], "2. Analysing Composition": [[1, "analysing-composition"]], "3. Distance Analysis": [[1, "distance-analysis"]], "4. Is Within": [[1, "is-within"]], "5. Geometry in AMPAL": [[1, "geometry-in-ampal"]], "6. Summary and activities": [[1, "summary-and-activities"]], "ampal package": [[2, "ampal-package"]], "Submodules": [[2, "submodules"]], "ampal.align module": [[2, "module-ampal.align"]], "ampal.amino_acids module": [[2, "module-ampal.amino_acids"]], "ampal.ampal_warnings module": [[2, "module-ampal.ampal_warnings"]], "ampal.analyse_protein module": [[2, "module-ampal.analyse_protein"]], "ampal.assembly module": [[2, "module-ampal.assembly"]], "ampal.base_ampal module": [[2, "module-ampal.base_ampal"]], "ampal.data module": [[2, "module-ampal.data"]], "ampal.dssp module": [[2, "module-ampal.dssp"]], "ampal.geometry module": [[2, "module-ampal.geometry"]], "ampal.interactions module": [[2, "module-ampal.interactions"]], "ampal.ligands module": [[2, "module-ampal.ligands"]], "ampal.nucleic_acid module": [[2, "module-ampal.nucleic_acid"]], "ampal.pdb_parser module": [[2, "module-ampal.pdb_parser"]], "ampal.protein module": [[2, "module-ampal.protein"]], "ampal.pseudo_atoms module": [[2, "module-ampal.pseudo_atoms"]], "Module contents": [[2, "module-ampal"]], "AMPAL Documentation": [[3, "ampal-documentation"]], "Contents:": [[3, null]], "Indices and Search": [[3, "indices-and-search"]], "Installation": [[4, "installation"]], "Basic Install": [[4, "basic-install"]], "Installing from Source": [[4, "installing-from-source"]], "ampal": [[5, "ampal"]]}, "indexentries": {"ampalcontainer (class in ampal.assembly)": [[2, "ampal.assembly.AmpalContainer"]], "assembly (class in ampal.assembly)": [[2, "ampal.assembly.Assembly"]], "atom (class in ampal.base_ampal)": [[2, "ampal.base_ampal.Atom"]], "axis (class in ampal.geometry)": [[2, "ampal.geometry.Axis"]], "baseampal (class in ampal.base_ampal)": [[2, "ampal.base_ampal.BaseAmpal"]], "covalentbond (class in ampal.interactions)": [[2, "ampal.interactions.CovalentBond"]], "curve (class in ampal.geometry)": [[2, "ampal.geometry.Curve"]], "dependencynotfoundwarning": [[2, "ampal.ampal_warnings.DependencyNotFoundWarning"]], "helicalcurve (class in ampal.geometry)": [[2, "ampal.geometry.HelicalCurve"]], "hydrogenbond (class in ampal.interactions)": [[2, "ampal.interactions.HydrogenBond"]], "interaction (class in ampal.interactions)": [[2, "ampal.interactions.Interaction"]], "ligand (class in ampal.ligands)": [[2, "ampal.ligands.Ligand"]], "ligandgroup (class in ampal.ligands)": [[2, "ampal.ligands.LigandGroup"]], "mmcalign (class in ampal.align)": [[2, "ampal.align.MMCAlign"]], "malformedpdbwarning": [[2, "ampal.ampal_warnings.MalformedPDBWarning"]], "monomer (class in ampal.base_ampal)": [[2, "ampal.base_ampal.Monomer"]], "noncovalentinteraction (class in ampal.interactions)": [[2, "ampal.interactions.NonCovalentInteraction"]], "noncanonicalwarning": [[2, "ampal.ampal_warnings.NoncanonicalWarning"]], "notparameterisedwarning": [[2, "ampal.ampal_warnings.NotParameterisedWarning"]], "nucleotide (class in ampal.nucleic_acid)": [[2, "ampal.nucleic_acid.Nucleotide"]], "pdbparser (class in ampal.pdb_parser)": [[2, "ampal.pdb_parser.PdbParser"]], "polymer (class in ampal.base_ampal)": [[2, "ampal.base_ampal.Polymer"]], "polynucleotide (class in ampal.nucleic_acid)": [[2, "ampal.nucleic_acid.Polynucleotide"]], "polypeptide (class in ampal.protein)": [[2, "ampal.protein.Polypeptide"]], "primitive (class in ampal.pseudo_atoms)": [[2, "ampal.pseudo_atoms.Primitive"]], "pseudoatom (class in ampal.pseudo_atoms)": [[2, "ampal.pseudo_atoms.PseudoAtom"]], "pseudogroup (class in ampal.pseudo_atoms)": [[2, "ampal.pseudo_atoms.PseudoGroup"]], "pseudomonomer (class in ampal.pseudo_atoms)": [[2, "ampal.pseudo_atoms.PseudoMonomer"]], "quaternion (class in ampal.geometry)": [[2, "ampal.geometry.Quaternion"]], "residue (class in ampal.protein)": [[2, "ampal.protein.Residue"]], "a (ampal.interactions.covalentbond property)": [[2, "ampal.interactions.CovalentBond.a"]], "a (ampal.interactions.interaction attribute)": [[2, "ampal.interactions.Interaction.a"]], "acceptor (ampal.interactions.hydrogenbond attribute)": [[2, "ampal.interactions.HydrogenBond.acceptor"]], "acceptor (ampal.interactions.noncovalentinteraction attribute)": [[2, "ampal.interactions.NonCovalentInteraction.acceptor"]], "acceptor (ampal.interactions.noncovalentinteraction property)": [[2, "id0"]], "acceptor_monomer (ampal.interactions.hydrogenbond property)": [[2, "ampal.interactions.HydrogenBond.acceptor_monomer"]], "active_state (ampal.base_ampal.monomer property)": [[2, "ampal.base_ampal.Monomer.active_state"]], "add_amino_acid_to_json() (in module ampal.amino_acids)": [[2, "ampal.amino_acids.add_amino_acid_to_json"]], "align() (in module ampal.protein)": [[2, "ampal.protein.align"]], "align_backbones() (in module ampal.align)": [[2, "ampal.align.align_backbones"]], "alpha (ampal.geometry.helicalcurve property)": [[2, "ampal.geometry.HelicalCurve.alpha"]], "alpha_and_pitch() (ampal.geometry.helicalcurve class method)": [[2, "ampal.geometry.HelicalCurve.alpha_and_pitch"]], "alpha_and_radius() (ampal.geometry.helicalcurve class method)": [[2, "ampal.geometry.HelicalCurve.alpha_and_radius"]], "alpha_angles() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.alpha_angles"]], "ampal": [[2, "module-ampal"]], "ampal.align": [[2, "module-ampal.align"]], "ampal.amino_acids": [[2, "module-ampal.amino_acids"]], "ampal.ampal_warnings": [[2, "module-ampal.ampal_warnings"]], "ampal.analyse_protein": [[2, "module-ampal.analyse_protein"]], "ampal.assembly": [[2, "module-ampal.assembly"]], "ampal.base_ampal": [[2, "module-ampal.base_ampal"]], "ampal.data": [[2, "module-ampal.data"]], "ampal.dssp": [[2, "module-ampal.dssp"]], "ampal.geometry": [[2, "module-ampal.geometry"]], "ampal.interactions": [[2, "module-ampal.interactions"]], "ampal.ligands": [[2, "module-ampal.ligands"]], "ampal.nucleic_acid": [[2, "module-ampal.nucleic_acid"]], "ampal.pdb_parser": [[2, "module-ampal.pdb_parser"]], "ampal.protein": [[2, "module-ampal.protein"]], "ampal.pseudo_atoms": [[2, "module-ampal.pseudo_atoms"]], "ang_a (ampal.interactions.hydrogenbond attribute)": [[2, "ampal.interactions.HydrogenBond.ang_a"]], "ang_d (ampal.interactions.hydrogenbond attribute)": [[2, "ampal.interactions.HydrogenBond.ang_d"]], "angle_and_axis() (ampal.geometry.quaternion class method)": [[2, "ampal.geometry.Quaternion.angle_and_axis"]], "angle_between_vectors() (in module ampal.geometry)": [[2, "ampal.geometry.angle_between_vectors"]], "append() (ampal.assembly.ampalcontainer method)": [[2, "ampal.assembly.AmpalContainer.append"]], "append() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.append"]], "append() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.append"]], "apply_conjugation() (ampal.geometry.quaternion method)": [[2, "ampal.geometry.Quaternion.apply_conjugation"]], "apply_unit_quaternion() (in module ampal.geometry)": [[2, "ampal.geometry.apply_unit_quaternion"]], "arc_length() (ampal.geometry.axis method)": [[2, "ampal.geometry.Axis.arc_length"]], "arc_length() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.arc_length"]], "array (ampal.base_ampal.atom property)": [[2, "ampal.base_ampal.Atom.array"]], "as_array (ampal.geometry.quaternion property)": [[2, "ampal.geometry.Quaternion.as_array"]], "as_rotation() (ampal.geometry.quaternion method)": [[2, "ampal.geometry.Quaternion.as_rotation"]], "atoms (ampal.base_ampal.monomer property)": [[2, "ampal.base_ampal.Monomer.atoms"]], "atoms (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.atoms"]], "axis (ampal.geometry.helicalcurve property)": [[2, "ampal.geometry.HelicalCurve.axis"]], "b (ampal.interactions.covalentbond property)": [[2, "ampal.interactions.CovalentBond.b"]], "b (ampal.interactions.interaction attribute)": [[2, "ampal.interactions.Interaction.b"]], "backbone (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.backbone"]], "backbone (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.backbone"]], "backbone (ampal.protein.residue property)": [[2, "ampal.protein.Residue.backbone"]], "backbone_bond_angles (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.backbone_bond_angles"]], "backbone_bond_lengths (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.backbone_bond_lengths"]], "c_join() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.c_join"]], "cap() (in module ampal.base_ampal)": [[2, "ampal.base_ampal.cap"]], "cartesian_to_cylindrical() (in module ampal.geometry)": [[2, "ampal.geometry.cartesian_to_cylindrical"]], "cartesian_to_spherical() (in module ampal.geometry)": [[2, "ampal.geometry.cartesian_to_spherical"]], "cc_to_local_params() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.cc_to_local_params"]], "centre_of_atoms() (in module ampal.base_ampal)": [[2, "ampal.base_ampal.centre_of_atoms"]], "centre_of_mass (ampal.base_ampal.baseampal property)": [[2, "ampal.base_ampal.BaseAmpal.centre_of_mass"]], "centre_of_mass() (in module ampal.geometry)": [[2, "ampal.geometry.centre_of_mass"]], "centroid (ampal.protein.residue property)": [[2, "ampal.protein.Residue.centroid"]], "change_state() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.change_state"]], "charge_series() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.charge_series"]], "check_availability() (in module ampal.ampal_warnings)": [[2, "ampal.ampal_warnings.check_availability"]], "check_for_non_canonical() (ampal.pdb_parser.pdbparser static method)": [[2, "ampal.pdb_parser.PdbParser.check_for_non_canonical"]], "check_move() (ampal.align.mmcalign static method)": [[2, "ampal.align.MMCAlign.check_move"]], "classify_angle_as_rotamer() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.classify_angle_as_rotamer"]], "close_monomers() (ampal.base_ampal.monomer method)": [[2, "ampal.base_ampal.Monomer.close_monomers"]], "closest_distance() (in module ampal.geometry)": [[2, "ampal.geometry.closest_distance"]], "closest_points_on_lines() (in module ampal.geometry)": [[2, "ampal.geometry.closest_points_on_lines"]], "conjugate (ampal.geometry.quaternion property)": [[2, "ampal.geometry.Quaternion.conjugate"]], "coordinates (ampal.pseudo_atoms.primitive property)": [[2, "ampal.pseudo_atoms.Primitive.coordinates"]], "covalent_bonds() (in module ampal.interactions)": [[2, "ampal.interactions.covalent_bonds"]], "crick_angles() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.crick_angles"]], "current_line (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.current_line"]], "curvature() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.curvature"]], "cylindrical_to_cartesian() (in module ampal.geometry)": [[2, "ampal.geometry.cylindrical_to_cartesian"]], "dihedral() (in module ampal.geometry)": [[2, "ampal.geometry.dihedral"]], "direction_vector (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.direction_vector"]], "dist (ampal.interactions.hydrogenbond attribute)": [[2, "ampal.interactions.HydrogenBond.dist"]], "dist (ampal.interactions.interaction attribute)": [[2, "ampal.interactions.Interaction.dist"]], "dist (ampal.interactions.noncovalentinteraction attribute)": [[2, "ampal.interactions.NonCovalentInteraction.dist"]], "distance() (ampal.geometry.quaternion method)": [[2, "ampal.geometry.Quaternion.distance"]], "distance() (in module ampal.geometry)": [[2, "ampal.geometry.distance"]], "distance_to_point() (ampal.geometry.axis method)": [[2, "ampal.geometry.Axis.distance_to_point"]], "donor (ampal.interactions.hydrogenbond attribute)": [[2, "ampal.interactions.HydrogenBond.donor"]], "donor (ampal.interactions.noncovalentinteraction attribute)": [[2, "ampal.interactions.NonCovalentInteraction.donor"]], "donor (ampal.interactions.noncovalentinteraction property)": [[2, "id7"]], "donor_monomer (ampal.interactions.hydrogenbond property)": [[2, "ampal.interactions.HydrogenBond.donor_monomer"]], "dssp_available() (in module ampal.dssp)": [[2, "ampal.dssp.dssp_available"]], "element (ampal.base_ampal.atom attribute)": [[2, "ampal.base_ampal.Atom.element"]], "element (ampal.pseudo_atoms.pseudoatom attribute)": [[2, "ampal.pseudo_atoms.PseudoAtom.element"]], "end() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.end"]], "extend() (ampal.assembly.ampalcontainer method)": [[2, "ampal.assembly.AmpalContainer.extend"]], "extend() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.extend"]], "extend() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.extend"]], "extract_all_ss_dssp() (in module ampal.dssp)": [[2, "ampal.dssp.extract_all_ss_dssp"]], "extract_angle_and_axis() (ampal.geometry.quaternion method)": [[2, "ampal.geometry.Quaternion.extract_angle_and_axis"]], "fasta (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.fasta"]], "fasta (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.fasta"]], "filter_mol_types() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.filter_mol_types"]], "find_atoms_within_distance() (in module ampal.base_ampal)": [[2, "ampal.base_ampal.find_atoms_within_distance"]], "find_covalent_bonds() (in module ampal.interactions)": [[2, "ampal.interactions.find_covalent_bonds"]], "find_foot() (in module ampal.geometry)": [[2, "ampal.geometry.find_foot"]], "find_foot_on_plane() (in module ampal.geometry)": [[2, "ampal.geometry.find_foot_on_plane"]], "find_limits() (in module ampal.geometry)": [[2, "ampal.geometry.find_limits"]], "find_ss_regions() (in module ampal.dssp)": [[2, "ampal.dssp.find_ss_regions"]], "find_transformations() (in module ampal.geometry)": [[2, "ampal.geometry.find_transformations"]], "flat_list_to_dummy_chain() (in module ampal.protein)": [[2, "ampal.protein.flat_list_to_dummy_chain"]], "flat_list_to_polymer() (in module ampal.protein)": [[2, "ampal.protein.flat_list_to_polymer"]], "flip_reference_axis_if_antiparallel() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.flip_reference_axis_if_antiparallel"]], "float_f() (ampal.align.mmcalign static method)": [[2, "ampal.align.MMCAlign.float_f"]], "from_coordinates() (ampal.pseudo_atoms.primitive class method)": [[2, "ampal.pseudo_atoms.Primitive.from_coordinates"]], "gen_sectors() (in module ampal.geometry)": [[2, "ampal.geometry.gen_sectors"]], "gen_states() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.gen_states"]], "generate_bond_subgraphs_from_break() (in module ampal.interactions)": [[2, "ampal.interactions.generate_bond_subgraphs_from_break"]], "generate_covalent_bond_graph() (in module ampal.interactions)": [[2, "ampal.interactions.generate_covalent_bond_graph"]], "get_aa_code() (in module ampal.amino_acids)": [[2, "ampal.amino_acids.get_aa_code"]], "get_aa_info() (in module ampal.amino_acids)": [[2, "ampal.amino_acids.get_aa_info"]], "get_aa_letter() (in module ampal.amino_acids)": [[2, "ampal.amino_acids.get_aa_letter"]], "get_atoms() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.get_atoms"]], "get_atoms() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.get_atoms"]], "get_atoms() (ampal.base_ampal.monomer method)": [[2, "ampal.base_ampal.Monomer.get_atoms"]], "get_atoms() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.get_atoms"]], "get_coords() (ampal.geometry.axis method)": [[2, "ampal.geometry.Axis.get_coords"]], "get_coords() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.get_coords"]], "get_ligands() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.get_ligands"]], "get_monomers() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.get_monomers"]], "get_monomers() (ampal.base_ampal.monomer method)": [[2, "ampal.base_ampal.Monomer.get_monomers"]], "get_monomers() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.get_monomers"]], "get_reference_coords() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.get_reference_coords"]], "get_slice_from_res_id() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.get_slice_from_res_id"]], "get_ss_regions() (in module ampal.dssp)": [[2, "ampal.dssp.get_ss_regions"]], "id (ampal.assembly.ampalcontainer attribute)": [[2, "ampal.assembly.AmpalContainer.id"]], "id (ampal.base_ampal.atom attribute)": [[2, "ampal.base_ampal.Atom.id"]], "id (ampal.base_ampal.monomer attribute)": [[2, "ampal.base_ampal.Monomer.id"]], "id (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.id"]], "id (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.id"]], "id (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.id"]], "id (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.id"]], "id (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.id"]], "id (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.id"]], "id (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.id"]], "id (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.id"]], "id (ampal.pseudo_atoms.pseudoatom attribute)": [[2, "ampal.pseudo_atoms.PseudoAtom.id"]], "id (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.id"]], "id (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.id"]], "ignore_end (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.ignore_end"]], "insertion_code (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.insertion_code"]], "insertion_code (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.insertion_code"]], "insertion_code (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.insertion_code"]], "insertion_code (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.insertion_code"]], "insertion_code (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.insertion_code"]], "intersection_start_and_end() (in module ampal.geometry)": [[2, "ampal.geometry.intersection_start_and_end"]], "is_acute() (in module ampal.geometry)": [[2, "ampal.geometry.is_acute"]], "is_hetero (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.is_hetero"]], "is_hetero (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.is_hetero"]], "is_hetero (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.is_hetero"]], "is_hetero (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.is_hetero"]], "is_hetero (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.is_hetero"]], "is_within() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.is_within"]], "is_within() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.is_within"]], "isoelectric_point (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.isoelectric_point"]], "isoelectric_point (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.isoelectric_point"]], "length (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.length"]], "ligands (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.ligands"]], "ligands (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.ligands"]], "ligands (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.ligands"]], "ligands (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.ligands"]], "load_pdb() (in module ampal.pdb_parser)": [[2, "ampal.pdb_parser.load_pdb"]], "make_ampal() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.make_ampal"]], "make_pdb() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.make_pdb"]], "make_pdb() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.make_pdb"]], "make_pdb() (ampal.base_ampal.monomer method)": [[2, "ampal.base_ampal.Monomer.make_pdb"]], "make_pdb() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.make_pdb"]], "make_primitive() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.make_primitive"]], "make_primitive_extrapolate_ends() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.make_primitive_extrapolate_ends"]], "make_primitive_smoothed() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.make_primitive_smoothed"]], "measure_sidechain_torsion_angles() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.measure_sidechain_torsion_angles"]], "measure_torsion_angles() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.measure_torsion_angles"]], "midpoint (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.midpoint"]], "minimal_distance_between_lines() (in module ampal.geometry)": [[2, "ampal.geometry.minimal_distance_between_lines"]], "module": [[2, "module-ampal"], [2, "module-ampal.align"], [2, "module-ampal.amino_acids"], [2, "module-ampal.ampal_warnings"], [2, "module-ampal.analyse_protein"], [2, "module-ampal.assembly"], [2, "module-ampal.base_ampal"], [2, "module-ampal.data"], [2, "module-ampal.dssp"], [2, "module-ampal.geometry"], [2, "module-ampal.interactions"], [2, "module-ampal.ligands"], [2, "module-ampal.nucleic_acid"], [2, "module-ampal.pdb_parser"], [2, "module-ampal.protein"], [2, "module-ampal.pseudo_atoms"]], "mol_code (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.mol_code"]], "mol_code (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.mol_code"]], "mol_code (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.mol_code"]], "mol_code (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.mol_code"]], "mol_code (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.mol_code"]], "molar_extinction_280 (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.molar_extinction_280"]], "molar_extinction_280 (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.molar_extinction_280"]], "molecular_weight (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.molecular_weight"]], "molecular_weight (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.molecular_weight"]], "molecule_type (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.molecule_type"]], "molecule_type (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.molecule_type"]], "molecule_type (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.molecule_type"]], "molecule_type (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.molecule_type"]], "n_join() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.n_join"]], "new_labels (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.new_labels"]], "norm (ampal.geometry.quaternion property)": [[2, "ampal.geometry.Quaternion.norm"]], "parent (ampal.base_ampal.atom attribute)": [[2, "ampal.base_ampal.Atom.parent"]], "parent (ampal.base_ampal.monomer attribute)": [[2, "ampal.base_ampal.Monomer.parent"]], "parent (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.parent"]], "parent (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.parent"]], "parent (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.parent"]], "parent (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.parent"]], "parent (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.parent"]], "parent (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.parent"]], "parent (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.parent"]], "parent (ampal.pseudo_atoms.pseudoatom attribute)": [[2, "ampal.pseudo_atoms.PseudoAtom.parent"]], "parent (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.parent"]], "parent (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.parent"]], "parse_pdb_file() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.parse_pdb_file"]], "partial_charge() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.partial_charge"]], "pdb (ampal.assembly.ampalcontainer property)": [[2, "ampal.assembly.AmpalContainer.pdb"]], "pdb (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.pdb"]], "pdb (ampal.base_ampal.baseampal property)": [[2, "ampal.base_ampal.BaseAmpal.pdb"]], "pdb (ampal.pseudo_atoms.pseudomonomer property)": [[2, "ampal.pseudo_atoms.PseudoMonomer.pdb"]], "pdb_lines (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.pdb_lines"]], "pdb_parse_tree (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.pdb_parse_tree"]], "pitch (ampal.geometry.helicalcurve property)": [[2, "ampal.geometry.HelicalCurve.pitch"]], "pitch_and_radius() (ampal.geometry.helicalcurve class method)": [[2, "ampal.geometry.HelicalCurve.pitch_and_radius"]], "point() (ampal.geometry.axis method)": [[2, "ampal.geometry.Axis.point"]], "point() (ampal.geometry.curve method)": [[2, "ampal.geometry.Curve.point"]], "point() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.point"]], "points_on_a_circle() (in module ampal.geometry)": [[2, "ampal.geometry.points_on_a_circle"]], "polymer_to_reference_axis_distances() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.polymer_to_reference_axis_distances"]], "polypeptide_vector() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.polypeptide_vector"]], "primitive (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.primitive"]], "primitives (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.primitives"]], "proc_atom() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.proc_atom"]], "proc_chain() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.proc_chain"]], "proc_functions (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.proc_functions"]], "proc_line_coordinate() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.proc_line_coordinate"]], "proc_monomer() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.proc_monomer"]], "proc_state() (ampal.pdb_parser.pdbparser method)": [[2, "ampal.pdb_parser.PdbParser.proc_state"]], "radii_of_curvature() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.radii_of_curvature"]], "radii_of_curvature() (ampal.pseudo_atoms.primitive method)": [[2, "ampal.pseudo_atoms.Primitive.radii_of_curvature"]], "radius (ampal.geometry.helicalcurve property)": [[2, "ampal.geometry.HelicalCurve.radius"]], "radius_of_circumcircle() (in module ampal.geometry)": [[2, "ampal.geometry.radius_of_circumcircle"]], "real_and_vector() (ampal.geometry.quaternion class method)": [[2, "ampal.geometry.Quaternion.real_and_vector"]], "reference_atom (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.reference_atom"]], "reference_atom (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.reference_atom"]], "reference_atom (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.reference_atom"]], "reference_atom (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.reference_atom"]], "reference_axis_from_chains() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.reference_axis_from_chains"]], "relabel_all() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.relabel_all"]], "relabel_all() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.relabel_all"]], "relabel_atoms() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.relabel_atoms"]], "relabel_atoms() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.relabel_atoms"]], "relabel_monomers() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.relabel_monomers"]], "relabel_monomers() (ampal.base_ampal.polymer method)": [[2, "ampal.base_ampal.Polymer.relabel_monomers"]], "relabel_polymers() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.relabel_polymers"]], "res_label (ampal.base_ampal.atom attribute)": [[2, "ampal.base_ampal.Atom.res_label"]], "res_label (ampal.pseudo_atoms.pseudoatom attribute)": [[2, "ampal.pseudo_atoms.PseudoAtom.res_label"]], "residues_per_turn() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.residues_per_turn"]], "rise_per_residue() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.rise_per_residue"]], "rise_per_residue() (ampal.pseudo_atoms.primitive method)": [[2, "ampal.pseudo_atoms.Primitive.rise_per_residue"]], "rmsd() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.rmsd"]], "rmsd() (in module ampal.geometry)": [[2, "ampal.geometry.rmsd"]], "rotate() (ampal.base_ampal.atom method)": [[2, "ampal.base_ampal.Atom.rotate"]], "rotate() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.rotate"]], "rotate_vector() (ampal.geometry.quaternion method)": [[2, "ampal.geometry.Quaternion.rotate_vector"]], "rotation_matrix() (in module ampal.geometry)": [[2, "ampal.geometry.rotation_matrix"]], "run_dssp() (in module ampal.dssp)": [[2, "ampal.dssp.run_dssp"]], "sequence (ampal.nucleic_acid.polynucleotide property)": [[2, "ampal.nucleic_acid.Polynucleotide.sequence"]], "sequence (ampal.protein.polypeptide property)": [[2, "ampal.protein.Polypeptide.sequence"]], "sequence_charge() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.sequence_charge"]], "sequence_isoelectric_point() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.sequence_isoelectric_point"]], "sequence_molar_extinction_280() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.sequence_molar_extinction_280"]], "sequence_molecular_weight() (in module ampal.analyse_protein)": [[2, "ampal.analyse_protein.sequence_molecular_weight"]], "sequences (ampal.assembly.assembly property)": [[2, "ampal.assembly.Assembly.sequences"]], "side_chain (ampal.protein.residue property)": [[2, "ampal.protein.Residue.side_chain"]], "sl (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.sl"]], "sl (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.sl"]], "sl (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.sl"]], "sl (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.sl"]], "sort_by_tag() (ampal.assembly.ampalcontainer method)": [[2, "ampal.assembly.AmpalContainer.sort_by_tag"]], "spherical_to_cartesian() (in module ampal.geometry)": [[2, "ampal.geometry.spherical_to_cartesian"]], "start_optimisation() (ampal.align.mmcalign method)": [[2, "ampal.align.MMCAlign.start_optimisation"]], "state (ampal.pdb_parser.pdbparser attribute)": [[2, "ampal.pdb_parser.PdbParser.state"]], "states (ampal.base_ampal.monomer attribute)": [[2, "ampal.base_ampal.Monomer.states"]], "states (ampal.ligands.ligand.self attribute)": [[2, "ampal.ligands.Ligand.self.states"]], "states (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.states"]], "states (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.states"]], "states (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.states"]], "states (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.states"]], "t_from_arc_length() (ampal.geometry.axis method)": [[2, "ampal.geometry.Axis.t_from_arc_length"]], "t_from_arc_length() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.t_from_arc_length"]], "tag_ca_geometry() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.tag_ca_geometry"]], "tag_ca_geometry() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.tag_ca_geometry"]], "tag_dssp_data() (in module ampal.dssp)": [[2, "ampal.dssp.tag_dssp_data"]], "tag_sidechain_dihedrals() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.tag_sidechain_dihedrals"]], "tag_torsion_angles() (ampal.assembly.assembly method)": [[2, "ampal.assembly.Assembly.tag_torsion_angles"]], "tag_torsion_angles() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.tag_torsion_angles"]], "tags (ampal.base_ampal.atom attribute)": [[2, "ampal.base_ampal.Atom.tags"]], "tags (ampal.base_ampal.monomer attribute)": [[2, "ampal.base_ampal.Monomer.tags"]], "tags (ampal.base_ampal.polymer attribute)": [[2, "ampal.base_ampal.Polymer.tags"]], "tags (ampal.ligands.ligand attribute)": [[2, "ampal.ligands.Ligand.tags"]], "tags (ampal.nucleic_acid.nucleotide attribute)": [[2, "ampal.nucleic_acid.Nucleotide.tags"]], "tags (ampal.nucleic_acid.polynucleotide attribute)": [[2, "ampal.nucleic_acid.Polynucleotide.tags"]], "tags (ampal.protein.polypeptide attribute)": [[2, "ampal.protein.Polypeptide.tags"]], "tags (ampal.protein.residue attribute)": [[2, "ampal.protein.Residue.tags"]], "tags (ampal.pseudo_atoms.primitive attribute)": [[2, "ampal.pseudo_atoms.Primitive.tags"]], "tags (ampal.pseudo_atoms.pseudoatom attribute)": [[2, "ampal.pseudo_atoms.PseudoAtom.tags"]], "tags (ampal.pseudo_atoms.pseudogroup attribute)": [[2, "ampal.pseudo_atoms.PseudoGroup.tags"]], "tags (ampal.pseudo_atoms.pseudomonomer attribute)": [[2, "ampal.pseudo_atoms.PseudoMonomer.tags"]], "torsion() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.torsion"]], "translate() (ampal.base_ampal.atom method)": [[2, "ampal.base_ampal.Atom.translate"]], "translate() (ampal.base_ampal.baseampal method)": [[2, "ampal.base_ampal.BaseAmpal.translate"]], "unique_id (ampal.base_ampal.atom property)": [[2, "ampal.base_ampal.Atom.unique_id"]], "unique_id (ampal.protein.residue property)": [[2, "ampal.protein.Residue.unique_id"]], "unit_binormal (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.unit_binormal"]], "unit_binormal() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.unit_binormal"]], "unit_normal (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.unit_normal"]], "unit_normal() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.unit_normal"]], "unit_tangent (ampal.geometry.axis property)": [[2, "ampal.geometry.Axis.unit_tangent"]], "unit_tangent() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.unit_tangent"]], "unit_vector() (in module ampal.geometry)": [[2, "ampal.geometry.unit_vector"]], "valid_backbone_bond_angles() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.valid_backbone_bond_angles"]], "valid_backbone_bond_lengths() (ampal.protein.polypeptide method)": [[2, "ampal.protein.Polypeptide.valid_backbone_bond_lengths"]], "vector (ampal.geometry.quaternion property)": [[2, "ampal.geometry.Quaternion.vector"]], "write_pdb() (in module ampal.base_ampal)": [[2, "ampal.base_ampal.write_pdb"]], "x (ampal.base_ampal.atom property)": [[2, "ampal.base_ampal.Atom.x"]], "x() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.x"]], "y (ampal.base_ampal.atom property)": [[2, "ampal.base_ampal.Atom.y"]], "y() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.y"]], "z (ampal.base_ampal.atom property)": [[2, "ampal.base_ampal.Atom.z"]], "z() (ampal.geometry.helicalcurve method)": [[2, "ampal.geometry.HelicalCurve.z"]]}})